---
title: "Multivariate Data Analysis: Making Sense of High-Dimensional Data"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Multivariate Data Analysis: Making Sense of High-Dimensional Data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---



```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup, eval = T}
library(compbio4all)
```


## Preliminaries

### Packages
```{r}
library(vegan)
#library(ggpubr)
```


### Data sets

```{r}
data(BCI)
data(BCI.env)
```


#### Vocab / Functions

grep(), gsub(), multivariate data, high-dimensional data, PCA, principle components


## High-dimensional Data in Biology

### Morphology: How do species vary

Morphology: numeric data

#### 3 Iris species

```{r}
summary(iris$Species)
```

4 morphological measurements; each is a dimenion of data

```{r}
summary(iris[,-5])
names(iris) <- gsub("Sepal","S",names(iris))
names(iris) <- gsub("Petal","P",names(iris))
names(iris) <- gsub("Length","len",names(iris))
names(iris) <- gsub("Width","wid",names(iris))
```

#### PCA

```{r}
par(mfrow = c(1,1))
# PC doesn't like categorical variables
iris.pca <- vegan::rda(iris[,-5], scale = TRUE)
biplot(iris.pca,
       display = c("sites", 
                   "species"),
       type = c("text",
                "points"), 
       xlab = "PC1: Major sources of variation",
       ylab = "PC2: Secondary sources of variation")
ordihull(iris.pca,
         group = iris$Species,
         col = c(1,2,3))

```


#### Un-packing PCA

```{r}
par(mfrow = c(2,2), mar = c(4,4,1,0.5))
biplot(iris.pca,
       display = c("sites", 
                   "species"),
       type = c("text",
                "points"), 
       xlab = "PC1: Major sources of variation",
       ylab = "PC2: Secondary sources of variation")
ordihull(iris.pca,
         group = iris$Species,
         col = c(1,2,3))

ordihull(iris.pca,
         group = iris$Species,
         col = c(1,2,3))

boxplot(P.len ~ Species, 
        data = iris,
        col = c(1,2,3))

boxplot(S.wid ~ Species, 
        data = iris,
        col = c(1,2,3))
plot(P.len ~ P.wid, data = iris, col = as.numeric(iris$Species))
```


### Ecology: Do trees prefer certain habitats

* 50 plots in a forest, each tree identified and counted.
* Each tree species is a dimensin of data (column)


```{r}
# you should be familiar with all of this code

# 50 plots
nrow(BCI)

# 225 different species
ncol(BCI)

head(BCI)[5:10]

# 3 major habitat types
summary(BCI.env$Habitat)
```

Set minor habitats to NA
```{r}
# you should be able to understand what this code is doing

BCI.env$Habitat[grep("Swamp",BCI.env$Habitat)] <- NA
BCI.env$Habitat[grep("Young",BCI.env$Habitat)] <- NA
```

#### Do certain habitats have the same species?

```{r, echo = F}
#BCI_t <- t(BCI)
example_NMDS <- metaMDS(BCI, # Our community-by-species matrix
                     k=2,
                     trymax= 10) # The number of reduced dimensions

pl <- ordiplot(example_NMDS,
         type="none",
          display = "sites")
points(pl, "sites",  
       col = as.numeric(BCI.env$Habitat),
       pch =  as.numeric(BCI.env$Habitat))

ordihull(example_NMDS,
         groups=BCI.env$Habitat,
         draw="polygon",
         label=T)

```


### Phylogenetics


```{r, echo = F}
library(phangorn)
data(Laurasiatherian)
DNA.dat <- rbind(as.character(Laurasiatherian$Platypus),
      as.character(Laurasiatherian$Armadillo),
      as.character(Laurasiatherian$Alpaca),
     as.character( Laurasiatherian$Vole))

DNA.dat <- gsub("1","A",DNA.dat)
DNA.dat <- gsub("2","T",DNA.dat)
DNA.dat <- gsub("3","C",DNA.dat)
DNA.dat <- gsub("4","G",DNA.dat)
```

Each loci in an alignment is a dimension 
```{r}
DNA.dat[,1:10]
```

Clustering methods can summarize sequences as phylogenetic tree
```{r}
d_hamming <- dist.hamming(Laurasiatherian)
tree <- upgma(d_hamming)
plot(tree)
```


```{r}
par(mfrow = c(1,1))
```

### Transcriptomics

* Calculate relative level of gene expression in different genes.
* Each gene is a dimension


## Exploratory versus Hypothesis testing

* Hypothesis testing - generates statistical values (p-values, etc)
* Exploratory - uses algorithms to generates graphs

```{r}
myTree <- ape::read.tree(text='(Exploratory, Hypothesis-testing);')
plot(myTree, direction = "rightwards", cex = 3,
     main = "Types of Data Analysis: ")
```

