<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Cleaning: Filtering focal rows with dplyr • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data Cleaning: Filtering focal rows with dplyr">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="data_cleaning-dplyr_filter_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data Cleaning: Filtering focal rows with dplyr</h1>
            
      
      
      <div class="hidden name"><code>data_cleaning-dplyr_filter.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>This vignette walks through the process of isolating a subset of a dataframe using the dplyr command filter(). In particular we’ll use filter() to isolate rows of data that correspond to partiular species of birds from a very large dataset. To do this, we’ll have to learn a bit about row the row indices that identify which row data occur in within a dataframe.</p>
</div>
<div id="learning-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h2>
<ul>
<li>Determine row indices using which()</li>
<li>Practice using the package dplyr
<ul>
<li>use dplyr filter</li>
</ul>
</li>
</ul>
</div>
<div id="packages" class="section level2">
<h2 class="hasAnchor">
<a href="#packages" class="anchor"></a>Packages</h2>
<ul>
<li>dplyr</li>
<li>wildlifeR</li>
</ul>
</div>
<div id="r-code" class="section level2">
<h2 class="hasAnchor">
<a href="#r-code" class="anchor"></a>R code</h2>
<div id="load-libraries" class="section level3">
<h3 class="hasAnchor">
<a href="#load-libraries" class="anchor"></a>Load Libraries</h3>
<p>The codes used by the American Ornithological Union (AOU) and BBS data for Pennsylvania are in the wildlifeR package. We’ll use the dplyr package to “reshape” into the proper format for analysis.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brouwern/wildlifeR">wildlifeR</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'wildlifeR'</span>
<span class="co">#&gt; The following objects are masked from 'package:compbio4all':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine_NLCD_classes, expand_BBS_routes, make_my_data2L,</span>
<span class="co">#&gt;     NA_to_zero, sample_BBS_routes, sample_dove_routes</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="data" class="section level3">
<h3 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h3>
<p>We’ll use 2 datasets:</p>
<ol style="list-style-type: decimal">
<li>AOU codes for each species of bird in North American</li>
<li>Bird counts from the BBS data for PA, and</li>
</ol>
<p>You can learn more about these data sets using the help command ?BBS_PA and ?AOU_species_codes</p>
<p>Load both sets of data</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## AOU species codes</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">AOU_species_codes</span><span class="op">)</span>

<span class="co">## BBS data for PA</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="bbs-bird-count-data" class="section level2">
<h2 class="hasAnchor">
<a href="#bbs-bird-count-data" class="anchor"></a>BBS bird count data</h2>
<p>Look at the BBS_PA data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt;   countrynum statenum Route RPID Year  Aou StopTotal SpeciesTotal</span>
<span class="co">#&gt; 1        840       72     1  101 1970 1940         2            2</span>
<span class="co">#&gt; 2        840       72     1  101 1970 2730         1            1</span>
<span class="co">#&gt; 3        840       72     1  101 1970 3091         2            2</span>
<span class="co">#&gt; 4        840       72     1  101 1970 3131         1            1</span>
<span class="co">#&gt; 5        840       72     1  101 1970 3160        21           30</span>
<span class="co">#&gt; 6        840       72     1  101 1970 3250         1            2</span></code></pre></div>
<p>This contains information on every bird species on every BBS route in Pennsylvania since the 1960s.</p>
<p>Using the dim() command we can see that this is a VERY big file</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt; [1] 244185      8</span></code></pre></div>
<p>Almost 1/4 million rows! This would be a very very big spreadsheet. When data files get this big we often do things to save space both in terms of file size and also on the screen when viewing it. Note that even though this datafile is about birds, there are no bird names (nor names of countries or states). Everything is coded using numbers.</p>
<p>To identify birds in the BBS dataset the USGS uses codes designated by the Aou (American Ornithological Union). These are typically 4 number codes unique to each species.</p>
</div>
<div id="aou-species-code-dataframe" class="section level2">
<h2 class="hasAnchor">
<a href="#aou-species-code-dataframe" class="anchor"></a>AOU Species code dataframe</h2>
<p>The dataset we loaded above with the command data(AOU_species_codes) loaded dataset with all of the Aou numeric codes and the corresponding common and latin nmaes of the speices.</p>
<p>Using dim() we can see that there are &gt;1000 different species identified in this dataset</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">AOU_species_codes</span><span class="op">)</span>
<span class="co">#&gt; [1] 1175    5</span></code></pre></div>
<p>Look at the data themselves</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">AOU_species_codes</span><span class="op">)</span>
<span class="co">#&gt;   spp.num alpha.code                  name                       spp order</span>
<span class="co">#&gt; 1      10       WEGR         Western Grebe Aechmophorus occidentalis    96</span>
<span class="co">#&gt; 2      11       CLGR         Clark's Grebe      Aechmophorus clarkii    98</span>
<span class="co">#&gt; 3      12       WCGR Western/Clark's Grebe                    hybrid    97</span>
<span class="co">#&gt; 4      20       RNGR      Red-necked Grebe        Podiceps grisegena    94</span>
<span class="co">#&gt; 5      30       HOGR          Horned Grebe          Podiceps auritus    93</span>
<span class="co">#&gt; 6      40       EAGR           Eared Grebe      Podiceps nigricollis    95</span></code></pre></div>
<p>This dataframe tells us the following thigns</p>
<ul>
<li>spp.num: a numeric code for each species</li>
<li>alapha.code: a 4-letter alphabetic code for each species</li>
<li>name: the common name for each species</li>
<li>spp: scientific name</li>
<li>order: taxonomic order</li>
</ul>
<p>We want to use the alphabetic code and/or the common name to indentify the numeric species code. We can then use this numeric code to isoalted data from the BBS dataset.</p>
<div id="determine-the-code-for-the-european-starling" class="section level3">
<h3 class="hasAnchor">
<a href="#determine-the-code-for-the-european-starling" class="anchor"></a>Determine the code for the European starling</h3>
<p>If we know the 4-letter alphabetic code we can use the which() command to locate the numeric code.</p>
<p>For example, the code for the European starling is “EUST”. The follwoing code determine which ROW NUMBER the European startling occur in within the AOU_species_codes dataframe (and saves it to an object “EUST.row.number”)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">EUST.row.number</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">AOU_species_codes</span><span class="op">$</span><span class="va">alpha.code</span> <span class="op">==</span> <span class="st">"EUST"</span><span class="op">)</span></code></pre></div>
<p>If you google “AOU species codes” you can usually locate a PDF or website with the codes.</p>
<p>What we have retrived with which() is a number that is the ROW that has “EUST” in it is</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">EUST.row.number</span>
<span class="co">#&gt; [1] 643</span></code></pre></div>
<p>The number happens to be ‘r EUST.row.number’</p>
<p>We can look at what is in row ‘r EUST.row.number’ like this, but giving R the dataframe and using square brackets [ ]. Note the comma after “EUST.row.number”.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">AOU_species_codes</span><span class="op">[</span><span class="va">EUST.row.number</span>, <span class="op">]</span>
<span class="co">#&gt;     spp.num alpha.code              name              spp order</span>
<span class="co">#&gt; 717    4930       EUST European Starling Sturnus vulgaris   864</span></code></pre></div>
<p>Or like this. Again , note the comma after “643”.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">AOU_species_codes</span><span class="op">[</span><span class="fl">643</span>, <span class="op">]</span>
<span class="co">#&gt;     spp.num alpha.code              name              spp order</span>
<span class="co">#&gt; 717    4930       EUST European Starling Sturnus vulgaris   864</span></code></pre></div>
<p>This tells us that the in row 643 there is data on the European starling, Sturnus vulgaris. In particular, it has a species number of 4930. This number is what we need to extract European starling data from the BBS_PA dataframe.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">EUST.numeric.code</span> <span class="op">&lt;-</span> <span class="fl">4940</span></code></pre></div>
</div>
</div>
<div id="dplyr-and-filter" class="section level2">
<h2 class="hasAnchor">
<a href="#dplyr-and-filter" class="anchor"></a>dplyr and filter()</h2>
<p>which() is the classic way of doing this in R. You can do the exact same thing with the dplyr function filter(). Note the use of the “pipe” “%&gt;%” which connects the dataset AOU_species_codes with the filter command.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">AOU_species_codes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">alpha.code</span> <span class="op">==</span> <span class="st">"EUST"</span><span class="op">)</span>
<span class="co">#&gt;   spp.num alpha.code              name              spp order</span>
<span class="co">#&gt; 1    4930       EUST European Starling Sturnus vulgaris   864</span></code></pre></div>
<p>More on dplyr and filter() below.</p>
<div id="optionaladvanced-advanced-searching-using-grep" class="section level3">
<h3 class="hasAnchor">
<a href="#optionaladvanced-advanced-searching-using-grep" class="anchor"></a>OPTIONAL/ADVANCED: Advanced searching using grep</h3>
<p>The above approach requires that you know the alphabetic species code, which we fed to the which(). R has a flexible search tool called grep() that allows you to search for text, parts of words, and parts of sentences. grep() allows us to search using words like this within the “$name” column.</p>
<p>We can get the same result as above like this</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"European starling"</span>, 
     <span class="va">AOU_species_codes</span><span class="op">$</span><span class="va">name</span>,
     ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; [1] 643</span></code></pre></div>
<p>If we use just the word starling, though, we can more than one number. Each of these is a row number where the word “starling” occurs.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"starling"</span>, 
     <span class="va">AOU_species_codes</span><span class="op">$</span><span class="va">name</span>,
     ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; [1]  643 1129 1130 1138</span></code></pre></div>
<p>We can save thse numbers to an R object then see what they pull up in the AOU names dataframe</p>
<p>Store the row numbers</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">i.starling</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"starling"</span>, 
     <span class="va">AOU_species_codes</span><span class="op">$</span><span class="va">name</span>,
     ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>See what they correspond to</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">AOU_species_codes</span><span class="op">[</span><span class="va">i.starling</span>, <span class="op">]</span>
<span class="co">#&gt;      spp.num alpha.code                 name               spp order</span>
<span class="co">#&gt; 717     4930       EUST    European Starling  Sturnus vulgaris   864</span>
<span class="co">#&gt; 1235    8640       SAST      Samoan Starling Aplonis atrifusca   863</span>
<span class="co">#&gt; 1236    8650       POST  Polynesian Starling Aplonis tabuensis   862</span>
<span class="co">#&gt; 1244    8720       MIST Micronesian Starling     Aplonis opaca   861</span></code></pre></div>
<p>There are 4 species of starling in the AOU_species_codes dataframe.</p>
<p>We could also use “European” with grep. There are 4 birds with this word in thei common name</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#store the row numbers</span>
<span class="va">i.European</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"European"</span>, 
     <span class="va">AOU_species_codes</span><span class="op">$</span><span class="va">name</span>,
     ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co">#see what they match</span>
<span class="va">AOU_species_codes</span><span class="op">[</span><span class="va">i.European</span>, <span class="op">]</span>
<span class="co">#&gt;     spp.num alpha.code                       name                 spp order</span>
<span class="co">#&gt; 195    1380       EGWT European Green-winged Teal  Anas crecca crecca    57</span>
<span class="co">#&gt; 373    2710       EUGP     European Golden-Plover Pluvialis apricaria   289</span>
<span class="co">#&gt; 717    4930       EUST          European Starling    Sturnus vulgaris   864</span>
<span class="co">#&gt; 781    5261       EUGO         European Goldfinch Carduelis carduelis  1154</span></code></pre></div>
<p>grep() is very flexible and can take parts of works. Let’s try just “Euor”</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#store the row numbers</span>
<span class="va">i.Euro</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"Euro"</span>, 
     <span class="va">AOU_species_codes</span><span class="op">$</span><span class="va">name</span>,
     ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co">#see what they match</span>
<span class="va">AOU_species_codes</span><span class="op">[</span><span class="va">i.Euro</span>, <span class="op">]</span>
<span class="co">#&gt;     spp.num alpha.code                       name                 spp order</span>
<span class="co">#&gt; 195    1380       EGWT European Green-winged Teal  Anas crecca crecca    57</span>
<span class="co">#&gt; 373    2710       EUGP     European Golden-Plover Pluvialis apricaria   289</span>
<span class="co">#&gt; 717    4930       EUST          European Starling    Sturnus vulgaris   864</span>
<span class="co">#&gt; 781    5261       EUGO         European Goldfinch Carduelis carduelis  1154</span></code></pre></div>
<p>It would work also wiht “pean”, “star”, or any other set of letters.</p>
</div>
</div>
<div id="isolate-rows-from-a-large-dataframe-using-dplyrs-filter-function" class="section level2">
<h2 class="hasAnchor">
<a href="#isolate-rows-from-a-large-dataframe-using-dplyrs-filter-function" class="anchor"></a>Isolate rows from a large dataframe using dplyr’s filter function</h2>
<p>Above we used the which() and dplyr::filter() functiosn to figure out the Aou code for our focal bird. Now we’ll use that infromation to create a more manageable subset of BBS data.</p>
<p>First, load dplyr if you haven’t. THen use the big BBS_PA dataset in conjuction with a pipe and filter() to subset just the European starling data. Note that use the “&lt;-” operation to send all the output to a new dataframe</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load dplyr</span>
 <span class="co">#library(dplyr)</span>

<span class="co">#filter just the Euro starling</span>
<span class="va">BBS_PA_EUST</span> <span class="op">&lt;-</span> <span class="va">BBS_PA</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Aou</span> <span class="op">==</span> <span class="fl">4940</span><span class="op">)</span></code></pre></div>
<p>Here I gave filter the Aou code 4940. I could also have given it the ojbect I made above that stored the number</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#object with the code</span>
<span class="va">EUST.numeric.code</span>
<span class="co">#&gt; [1] 4940</span></code></pre></div>
<p>Use the object instead of the number 4940</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_EUST</span> <span class="op">&lt;-</span> <span class="va">BBS_PA</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Aou</span> <span class="op">==</span> <span class="va">EUST.numeric.code</span><span class="op">)</span></code></pre></div>
<p>The BBS_PA_EUST contains just rows from BBS_PA that contain the numeric code 4940 in the Aou column.</p>
<p>Let’s look at what we’ve done. Call dim() on the original BBS_PA data and the subset BBS_PA_EUST and see how the See how the dataframe has change.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt; [1] 244185      8</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA_EUST</span><span class="op">)</span>
<span class="co">#&gt; [1] 1615    8</span></code></pre></div>
<p>We’ve gone froma 1/4 million row dataframe BBS_PA to a 1600 row dataframe BBS_PA_EUST.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA_EUST</span><span class="op">)</span>
<span class="co">#&gt;    countrynum     statenum      Route             RPID          Year     </span>
<span class="co">#&gt;  Min.   :840   Min.   :72   Min.   :  1.00   Min.   :101   Min.   :1966  </span>
<span class="co">#&gt;  1st Qu.:840   1st Qu.:72   1st Qu.: 16.00   1st Qu.:101   1st Qu.:1984  </span>
<span class="co">#&gt;  Median :840   Median :72   Median : 35.00   Median :101   Median :1997  </span>
<span class="co">#&gt;  Mean   :840   Mean   :72   Mean   : 82.77   Mean   :101   Mean   :1995  </span>
<span class="co">#&gt;  3rd Qu.:840   3rd Qu.:72   3rd Qu.: 62.00   3rd Qu.:101   3rd Qu.:2006  </span>
<span class="co">#&gt;  Max.   :840   Max.   :72   Max.   :907.00   Max.   :101   Max.   :2016  </span>
<span class="co">#&gt;       Aou         StopTotal      SpeciesTotal  </span>
<span class="co">#&gt;  Min.   :4940   Min.   : 1.00   Min.   : 1.00  </span>
<span class="co">#&gt;  1st Qu.:4940   1st Qu.: 1.00   1st Qu.: 2.00  </span>
<span class="co">#&gt;  Median :4940   Median : 3.00   Median : 6.00  </span>
<span class="co">#&gt;  Mean   :4940   Mean   : 4.21   Mean   :10.75  </span>
<span class="co">#&gt;  3rd Qu.:4940   3rd Qu.: 6.00   3rd Qu.:14.00  </span>
<span class="co">#&gt;  Max.   :4940   Max.   :20.00   Max.   :78.00</span></code></pre></div>
<div id="what-next" class="section level3">
<h3 class="hasAnchor">
<a href="#what-next" class="anchor"></a>What next?</h3>
<p>This vignette shows how to take a really big dataframe and isolate just the subset of data that is required. In the wildlifeR two analayzes with these data are outlined: analysis of change over time, and analysis of abundance versus landcover.</p>
<ol style="list-style-type: decimal">
<li>If you want to examine change in a spcies over time, see the vignette … which uses the wildlifeR function sample_BBS_routes() to create a simple dataframe for looking at change over time.</li>
<li>If you want to examine change in abundance versus a landcover variable see “Data Cleaning: Merging 2 large data sets with dplyr”, which is an extended vignette which reviews what we’ve done here and extends it to access and merge-in landcover data.</li>
</ol>
</div>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="books" class="section level3">
<h3 class="hasAnchor">
<a href="#books" class="anchor"></a>Books</h3>
<p>A good introduction to using dplyr is:</p>
<p>Beckerman et al’s book “Getting start with R: An introduction for biologists” 2nd ed.</p>
</div>
<div id="websites" class="section level3">
<h3 class="hasAnchor">
<a href="#websites" class="anchor"></a>Websites</h3>
<p>“Selecting columns and renaming are so easy with dplyr” <a href="https://blog.exploratory.io/selecting-columns-809bdd1ef615" class="uri">https://blog.exploratory.io/selecting-columns-809bdd1ef615</a></p>
</div>
</div>
<div id="references-1" class="section level2">
<h2 class="hasAnchor">
<a href="#references-1" class="anchor"></a>References</h2>
<div id="books-1" class="section level3">
<h3 class="hasAnchor">
<a href="#books-1" class="anchor"></a>Books</h3>
<p>A good introduction to using dplyr is:</p>
<p>Beckerman et al’s book “Getting start with R: An introduction for biologists” 2nd ed.</p>
</div>
<div id="websites-1" class="section level3">
<h3 class="hasAnchor">
<a href="#websites-1" class="anchor"></a>Websites</h3>
<p>“Selecting columns and renaming are so easy with dplyr” <a href="https://blog.exploratory.io/selecting-columns-809bdd1ef615" class="uri">https://blog.exploratory.io/selecting-columns-809bdd1ef615</a></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
