<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data cleaning: Merging 2 large data sets with dplyr • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data cleaning: Merging 2 large data sets with dplyr">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="data_cleaning-merge_BBS_with_landcover_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data cleaning: Merging 2 large data sets with dplyr</h1>
            
      
      
      <div class="hidden name"><code>data_cleaning-merge_BBS_with_landcover.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>This vignette walks through the process of taking 2 complimentary sets of data and “merging” them using the powerful function full_join() in the package dplyr. We’ll take data on 1) the number of birds observed along USGS Breeding Bird Survey (BBS) routes and merge that ecological data with 2) geographic information on the types of landcover along those routes. In addition to this primary tasks of merging datasets, we’ll also do several other things to clean up the data to make it easy to use.</p>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="books" class="section level3">
<h3 class="hasAnchor">
<a href="#books" class="anchor"></a>Books</h3>
<p>A good introduction to using dplyr is:</p>
<p>Beckerman et al’s book “Getting start with R: An introduction for biologists” 2nd ed.</p>
</div>
<div id="websites" class="section level3">
<h3 class="hasAnchor">
<a href="#websites" class="anchor"></a>Websites</h3>
<p>“Selecting columns and renaming are so easy with dplyr” <a href="https://blog.exploratory.io/selecting-columns-809bdd1ef615" class="uri">https://blog.exploratory.io/selecting-columns-809bdd1ef615</a></p>
</div>
</div>
<div id="learning-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h2>
<ul>
<li>Practice using the package dplyr
<ul>
<li>use dplyr pipes: %&gt;%</li>
</ul>
</li>
<li>General dataframe cleaning with dplyr
<ul>
<li>Select focal columns using select()</li>
<li>Select focal rows using filter()</li>
<li>Change columns names with rename()</li>
</ul>
</li>
<li>Merge 2 dataframes using full_join()</li>
<li>Plotting data using the ggpubr extension of ggplot2</li>
</ul>
</div>
<div id="packages" class="section level2">
<h2 class="hasAnchor">
<a href="#packages" class="anchor"></a>Packages</h2>
<ul>
<li>dplry</li>
<li>ggplot2</li>
<li>ggpubr</li>
<li>wildlifeR</li>
</ul>
</div>
<div id="r-code" class="section level2">
<h2 class="hasAnchor">
<a href="#r-code" class="anchor"></a>R code</h2>
<div id="load-libraries" class="section level3">
<h3 class="hasAnchor">
<a href="#load-libraries" class="anchor"></a>Load Libraries</h3>
<p>BBS and landcover data for Pennsylvania are in the wildlifeR package, and we’ll use the dplyr package to “reshape” into the proper format for analysis.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brouwern/wildlifeR">wildlifeR</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="data" class="section level3">
<h3 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h3>
<p>We’ll use 2 datasets:</p>
<ol style="list-style-type: decimal">
<li>Bird counts from the BBS data for PA, and</li>
<li>USGS landcover classifications (eg urban, deciduous forest, etc) for a “buffer” 1 km around each BBS route in PA.</li>
</ol>
<p>You can learn more about these data sets using the help command ?BBS_PA and “BBS_PA_landcover_1km”.</p>
<p>Load both sets of data</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## BBS data for PA</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>

<span class="co">##Landcover data for BBS routes in PA</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km</span><span class="op">)</span></code></pre></div>
</div>
<div id="look-at-the-bbs_pa-dataframes" class="section level3">
<h3 class="hasAnchor">
<a href="#look-at-the-bbs_pa-dataframes" class="anchor"></a>Look at the BBS_PA dataframes</h3>
<div id="definitions-of" class="section level4">
<h4 class="hasAnchor">
<a href="#definitions-of" class="anchor"></a>Definitions of</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt;   countrynum statenum Route RPID Year  Aou StopTotal SpeciesTotal</span>
<span class="co">#&gt; 1        840       72     1  101 1970 1940         2            2</span>
<span class="co">#&gt; 2        840       72     1  101 1970 2730         1            1</span>
<span class="co">#&gt; 3        840       72     1  101 1970 3091         2            2</span>
<span class="co">#&gt; 4        840       72     1  101 1970 3131         1            1</span>
<span class="co">#&gt; 5        840       72     1  101 1970 3160        21           30</span>
<span class="co">#&gt; 6        840       72     1  101 1970 3250         1            2</span></code></pre></div>
<p>For our work, the important columns are</p>
<ul>
<li>Route: a categorical variables that defines the unique ID of each route in the State of PA</li>
<li>Year: a numeric varibles that defines the year the data were collected. The minmum is 1966. This is a potential focal predictor variable for examining change over time.</li>
<li>Aou: a code that designates each species, as defined by the American Ornithological Union (AOU). (Note that ths A is uppercase while the rest is lowercase)</li>
<li>SpeciesTotal: the total number of individuals of a given species that were observed on a given route. This will be our focal response variable.</li>
</ul>
<p>(NOTE: We will <strong>not</strong> be using the “StopTotal” column. A BBS routes is composed of 50 point counts, called “stops.” StopTotal is the total number of stops out of 50 on which a given species in a given year was observed. Its max value is 50, or 100% of the 50 stops on the route)</p>
<p>see ?BBS_PA for more information on these variables</p>
</div>
<div id="size-of-bbs_pa-dataframe" class="section level4">
<h4 class="hasAnchor">
<a href="#size-of-bbs_pa-dataframe" class="anchor"></a>Size of BBS_PA dataframe</h4>
<p>Thes BBS data is really really big dataset!:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt; [1] 244185      8</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt;    countrynum     statenum      Route            RPID            Year     </span>
<span class="co">#&gt;  Min.   :840   Min.   :72   Min.   :  1.0   Min.   :101.0   Min.   :1966  </span>
<span class="co">#&gt;  1st Qu.:840   1st Qu.:72   1st Qu.: 31.0   1st Qu.:101.0   1st Qu.:1984  </span>
<span class="co">#&gt;  Median :840   Median :72   Median : 58.0   Median :101.0   Median :1997  </span>
<span class="co">#&gt;  Mean   :840   Mean   :72   Mean   :107.5   Mean   :101.1   Mean   :1995  </span>
<span class="co">#&gt;  3rd Qu.:840   3rd Qu.:72   3rd Qu.: 90.0   3rd Qu.:101.0   3rd Qu.:2007  </span>
<span class="co">#&gt;  Max.   :840   Max.   :72   Max.   :911.0   Max.   :203.0   Max.   :2016  </span>
<span class="co">#&gt;       Aou          StopTotal       SpeciesTotal   </span>
<span class="co">#&gt;  Min.   :   60   Min.   : 1.000   Min.   :  1.00  </span>
<span class="co">#&gt;  1st Qu.: 4610   1st Qu.: 2.000   1st Qu.:  2.00  </span>
<span class="co">#&gt;  Median : 5630   Median : 4.000   Median :  5.00  </span>
<span class="co">#&gt;  Mean   : 5493   Mean   : 7.239   Mean   : 13.78  </span>
<span class="co">#&gt;  3rd Qu.: 6670   3rd Qu.:10.000   3rd Qu.: 15.00  </span>
<span class="co">#&gt;  Max.   :22860   Max.   :50.000   Max.   :992.00</span></code></pre></div>
<p>There are so many rows b/c</p>
<ul>
<li>There are 50 years of data.</li>
<li>There are 202 species of birds observed during the BBS in Pennsylvania</li>
<li>There are 137 routes in Pennsylvania</li>
</ul>
<p>Note, however, that if you multiply years<em>species</em>routes you get an even <strong>bigger</strong> number.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fl">50</span><span class="op">*</span><span class="fl">202</span><span class="op">*</span><span class="fl">137</span>
<span class="co">#&gt; [1] 1383700</span></code></pre></div>
<p>This is because</p>
<ul>
<li>The number of routes has changed over time</li>
<li>Data are only given for a species if it was obseved on a route</li>
</ul>
<p>Any important consequence of this is that there are only data on a species when it is actually obseved. So if a bird is see on route 99 in 2015 but not 2016, there is a row of data for it in 2015 but not 2016. Therefore, there are <strong>no zeros</strong> in the StopTotal column, which gives the number of of each species seen on a route in a given year.</p>
<p>We can see using summary() that the minimum value in the StopTotal column is indeed 1; not a single zero.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">$</span><span class="va">StopTotal</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;   1.000   2.000   4.000   7.239  10.000  50.000</span></code></pre></div>
</div>
</div>
</div>
<div id="isolate-focal-columns" class="section level2">
<h2 class="hasAnchor">
<a href="#isolate-focal-columns" class="anchor"></a>Isolate focal columns</h2>
<ul>
<li>For our analyses we don’t need every column in the dataset</li>
<li>Let’s look at the Year, Aou species code, Route number, and SpeciesTotal columns
<ul>
<li>All the data are from the state of Pennsyvlvania in the USA, so we don’t need these columns.</li>
</ul>
</li>
<li>We can use the select() function from the dplyr package to isolate the columns we want.</li>
<li>We tell dplyr the dataframe we want to work on, BBS_PA, add a “pipe” %&gt;%, and then use the select() command to tell it the columns within the dataframe we want.</li>
<li>We’ll save them into a new object called BBS_PA2</li>
<li>Note that the “A” in “Aou”" is uppercase while the rest are lowercase)</li>
<li>See page 60-61 in Beckerman et al’s book “Getting start with R: An introduction for biologists” for more info on select()</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load dplyr if you haven't already</span>
<span class="co">#library(dplyr)</span>

<span class="co">#look at the names of the full dataframe</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt; [1] "countrynum"   "statenum"     "Route"        "RPID"         "Year"        </span>
<span class="co">#&gt; [6] "Aou"          "StopTotal"    "SpeciesTotal"</span>


<span class="co">#use select() to isolate focal columns</span>
<span class="co">## and put into a new dataframe</span>
<span class="co">##(Note that ths A in Aou is uppercase </span>
<span class="co">## while the rest of the letters are lowercase)</span>

<span class="co">#BBS_PA2 &lt;-  select(.data = BBS_PA,</span>
<span class="co">#                   Year, Aou, Route, SpeciesTotal)</span>

<span class="co">#dplyr was not working for me so I did this another way of doing</span>
<span class="co">#this</span>

<span class="va">BBS_PA2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Year"</span>, <span class="st">"Aou"</span>, <span class="st">"Route"</span>, <span class="st">"SpeciesTotal"</span><span class="op">)</span><span class="op">]</span>

<span class="co">#look at columns in new dataframe</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">BBS_PA2</span><span class="op">)</span>
<span class="co">#&gt; [1] "Year"         "Aou"          "Route"        "SpeciesTotal"</span></code></pre></div>
<div id="isolate-focal-species-using-filter" class="section level3">
<h3 class="hasAnchor">
<a href="#isolate-focal-species-using-filter" class="anchor"></a>Isolate focal species using filter()</h3>
<ul>
<li>Each row is data on a different combination of: a species, a year, and a route</li>
<li>We want to isolate rows that just relate to our focal species; we’ll use the Scarlet Tanager</li>
<li>The AOU (American ornithological Union) code for the Scarlet Tanager (SCTA) is 6080.<br>
</li>
<li>We can tell R to just give us data on SCTA using the fliter() command in the dplyr package.</li>
<li>See page 62-65 in Beckerman et al’s book “Getting start with R: An introduction for biologists” for more info on filter()</li>
</ul>
<div id="an-example-bbs-dataframe" class="section level4">
<h4 class="hasAnchor">
<a href="#an-example-bbs-dataframe" class="anchor"></a>An example BBS dataframe</h4>
<ul>
<li>The following dataframe is a mock up of the general structure of the BBS data.</li>
<li>In the “Aou” column are listed several numbers: 5980, 6080, 5950, 6110, and several others
<ul>
<li>Each one of these numbers represents a different species</li>
<li>We want to isolate just the codes 6080, which represent the scarlet tanager</li>
<li>We want to discard the rest</li>
</ul>
</li>
</ul>
<pre><code>#&gt;      Year  Aou Route SpeciesTotal                 
#&gt; 2910 2011 5980     2           10                 
#&gt; 2974 2012 5980     2            9                 
#&gt; 3047 2013 5980     2            6                 
#&gt; 2911 2011 6080     2            3 we want this row
#&gt; 2909 2011 5950     2            4                 
#&gt; 2973 2012 5950     2            1                 
#&gt; 3046 2013 5950     2            5                 
#&gt; 2975 2012 6080     2            3 we want this row
#&gt; 2912 2011 6110     2            1                 
#&gt; 2976 2012 6110     2            3                 
#&gt; 3049 2013 6130     2           23                 
#&gt; 3048 2013 6080     2            3 we want this row
#&gt; 2913 2011 6120     2            4                 
#&gt; 2977 2012 6120     2            2                 
#&gt; 3050 2013 6140     2            7</code></pre>
<p>That is, we want this</p>
<pre><code>#&gt;   Year  Aou Route SpeciesTotal                 
#&gt; 1 2011 6080     2            3 we want this row
#&gt; 2 2012 6080     2            3 we want this row
#&gt; 3 2013 6080     2            3 we want this row</code></pre>
</div>
</div>
<div id="select-just-focal-rows-with-dplyrfilter" class="section level3">
<h3 class="hasAnchor">
<a href="#select-just-focal-rows-with-dplyrfilter" class="anchor"></a>Select just focal rows with dplyr::filter()</h3>
<div id="focal-bird-rows-6080-the-scarlet-tanager" class="section level4">
<h4 class="hasAnchor">
<a href="#focal-bird-rows-6080-the-scarlet-tanager" class="anchor"></a>Focal bird rows: 6080, the scarlet tanager</h4>
<ul>
<li>Again, the “%&gt;%” is called a pipe that connects the dataframe BBS_PA2 with the filter() command</li>
<li>“==” in the filter() commands means “exactly equals”</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="va">BBS_PA_SCTA</span> <span class="op">&lt;-</span> <span class="va">BBS_PA2</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Aou</span> <span class="op">==</span> <span class="fl">6080</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-filtered-bbs-data" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-filtered-bbs-data" class="anchor"></a>Plot filtered BBS data</h4>
<p>We can see how many birds were observed on all the routes each year using ggscatter() from the ggpubr() package.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span>

<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_SCTA</span>,
          y <span class="op">=</span> <span class="st">"SpeciesTotal"</span>,
          x <span class="op">=</span> <span class="st">"Year"</span><span class="op">)</span></code></pre></div>
<p><img src="data_cleaning-merge_BBS_with_landcover_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Note however that as discussed above this dataframe only contains data from routes where SCTA was observed; when SCTA wasn’t observed, nothing is recored at all, hence no zeros. We can see this with summary()</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA</span><span class="op">)</span>
<span class="co">#&gt;       Year           Aou           Route        SpeciesTotal   </span>
<span class="co">#&gt;  Min.   :1966   Min.   :6080   Min.   :  1.0   Min.   : 1.000  </span>
<span class="co">#&gt;  1st Qu.:1983   1st Qu.:6080   1st Qu.: 29.0   1st Qu.: 3.000  </span>
<span class="co">#&gt;  Median :1996   Median :6080   Median : 56.0   Median : 7.000  </span>
<span class="co">#&gt;  Mean   :1995   Mean   :6080   Mean   :113.3   Mean   : 8.566  </span>
<span class="co">#&gt;  3rd Qu.:2007   3rd Qu.:6080   3rd Qu.: 88.5   3rd Qu.:12.000  </span>
<span class="co">#&gt;  Max.   :2016   Max.   :6080   Max.   :911.0   Max.   :44.000</span></code></pre></div>
<p>Or by looking at a histogram</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/gghistogram.html">gghistogram</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA</span>,
            x <span class="op">=</span> <span class="st">"SpeciesTotal"</span><span class="op">)</span></code></pre></div>
<p><img src="data_cleaning-merge_BBS_with_landcover_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div id="focal-year-rows" class="section level4">
<h4 class="hasAnchor">
<a href="#focal-year-rows" class="anchor"></a>Focal year rows</h4>
<p>For this exercise we are only going to consider data from 2006, since that is the year that the landcover data we’ll be merging with comes from. (notes that “Year” has a capital “Y” while the rest is lowercase).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_SCTA</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Year</span> <span class="op">==</span> <span class="fl">2006</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_SCTA</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Year</span> <span class="op">==</span> <span class="fl">2006</span><span class="op">)</span></code></pre></div>
<p>We can see that we’ve greatly reduced the size of the original dataframe</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA</span><span class="op">)</span>
<span class="co">#&gt; [1] 244185      8</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA</span><span class="op">)</span>
<span class="co">#&gt; [1] 3535    4</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_2</span><span class="op">)</span>
<span class="co">#&gt; [1] 82  4</span></code></pre></div>
</div>
<div id="plot-filtered-scta-data" class="section level4">
<h4 class="hasAnchor">
<a href="#plot-filtered-scta-data" class="anchor"></a>Plot filtered SCTA data</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/gghistogram.html">gghistogram</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_SCTA_2</span>,
          x <span class="op">=</span> <span class="st">"SpeciesTotal"</span><span class="op">)</span></code></pre></div>
<p><img src="data_cleaning-merge_BBS_with_landcover_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
</div>
</div>
<div id="clean-landcover-data" class="section level3">
<h3 class="hasAnchor">
<a href="#clean-landcover-data" class="anchor"></a>Clean Landcover data</h3>
<ul>
<li>We’ll step back from the bird data now and switch to the data on the type of landcover surrounding the BBS routes.</li>
<li>There are a large number of columns of data in the landcover dataframe</li>
<li>Note that a couple of the columns names aren’t very good: “RT..NAME” and “RT..LENG.”</li>
<li>Columns that start with “NLCD” have data on the number of pixels of a certain land cover class surrounding a rounte. There are about 15 or so of these, for everything from open water to tundra.</li>
<li>See ?BBS_PA_landcover_1km for infromation on each of these</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Route"     "BCR"       "State"     "RT..NAME"  "RT..LENG." "NLCD.0"   </span>
<span class="co">#&gt;  [7] "NLCD.11"   "NLCD.12"   "NLCD.21"   "NLCD.22"   "NLCD.23"   "NLCD.24"  </span>
<span class="co">#&gt; [13] "NLCD.31"   "NLCD.41"   "NLCD.42"   "NLCD.43"   "NLCD.52"   "NLCD.71"  </span>
<span class="co">#&gt; [19] "NLCD.81"   "NLCD.82"   "NLCD.90"   "NLCD.95"   "SUM"</span></code></pre></div>
<div id="isoalte-focal-columnsforest-cover" class="section level4">
<h4 class="hasAnchor">
<a href="#isoalte-focal-columnsforest-cover" class="anchor"></a>Isoalte focal columns:forest cover</h4>
<p>Let’s focus on the columns that pertain to forest cover, which are contained in columns NLCD41, NLCD42, NLCD43. These represent three different types of forest, eg deciduous, coniferous, mixed. You find out information on them from the help file for the dataset using ?BBS_PA_landcover_1km.</p>
<p>We can subset these columns as we did for the BBS bird count data using using select(). Note that there is a period between “NLCD” and the number, and that “SUM” is in all caps.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">Route</span>, <span class="va">NLCD.41</span>, <span class="va">NLCD.42</span>, <span class="va">NLCD.43</span>, <span class="va">SUM</span><span class="op">)</span></code></pre></div>
</div>
<div id="add-up-total-forest-cover" class="section level4">
<h4 class="hasAnchor">
<a href="#add-up-total-forest-cover" class="anchor"></a>Add up total forest cover</h4>
<p>Let’s add up the NLCD.41, NLCD.42, and NLCD.43 columns</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">forest.total</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km_2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NLCD.41"</span>,
                                 <span class="st">"NLCD.42"</span>,
                                  <span class="st">"NLCD.43"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p>We can add this new column to the dataframe like this</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">forest.total</span> <span class="op">&lt;-</span> <span class="va">forest.total</span></code></pre></div>
<p>And see that it’s there using head()</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km_2</span><span class="op">)</span>
<span class="co">#&gt;   Route NLCD.41 NLCD.42 NLCD.43   SUM forest.total</span>
<span class="co">#&gt; 1     1   44953     563     797 98591        46313</span>
<span class="co">#&gt; 2     2   25119    1130      48 82345        26297</span>
<span class="co">#&gt; 3     3   34154     521      78 86328        34753</span>
<span class="co">#&gt; 4     4   39813    1695      60 88526        41568</span>
<span class="co">#&gt; 5     5   32244    5892    3518 97708        41654</span>
<span class="co">#&gt; 6     6   50529    4674   13320 87503        68523</span></code></pre></div>
<p>The “SUM” column tells us the total number of GIS pixels were actually used to determine all of the landcover in the BBS_PA_landcover_1km_2 dataframe. This varies a bit so we will actually want to use the percentage of forest cover, not the raw total. We can calcautle the percentage like this</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">forest.percent</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">forest.total</span> <span class="op">/</span> <span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">SUM</span></code></pre></div>
<p>And add it to the dataframe like this</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">forest.percent</span> <span class="op">&lt;-</span> <span class="va">forest.percent</span></code></pre></div>
<p>Note that we can actually do the math and add the new data to the dataframe in one step like this:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">forest.percent</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">forest.total</span> <span class="op">/</span> <span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">SUM</span></code></pre></div>
<p>We don’t need the individual columns for each of the three cover classes anymore (NCLD.41, 42, 43) so we can use select() to focus just on our new forest.percent and forest.total columns</p>
<p>We’ll call this new dataframe BBS_PA_landcover_1km_3</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_3</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">Route</span>, <span class="va">forest.total</span>, <span class="va">SUM</span>, <span class="va">forest.percent</span>,
         <span class="va">NLCD.41</span>,<span class="va">NLCD.42</span>,<span class="va">NLCD.43</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="merge-bbs-data-and-landcover-data" class="section level3">
<h3 class="hasAnchor">
<a href="#merge-bbs-data-and-landcover-data" class="anchor"></a>Merge BBS data and landcover data</h3>
<p>We have reduced the size and made changes two dataframe: BBS_PA, with counts of birds, and BBS_PA_landcover_1km, which has information on the habitats around BBS routes in PA.</p>
<p>Now we’ll make a new dataframe that <em>combines</em> these two sperate data sets. This is one of the most powerful features of R - taking big sets of data and with a few lines of codes merging them into a new data set.</p>
<p>We’ll do this using thhe full_join() command from the dplyr package. All we need to do is 1) tell full_joing the names of the two data sets and 2) tell the function what column is shared between the data setst.</p>
<p>The fact that the “Route” column occurs in both data sets allows them to be matached up.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_2</span> ,
                       <span class="va">BBS_PA_landcover_1km_3</span>, 
                       by <span class="op">=</span> <span class="st">"Route"</span><span class="op">)</span></code></pre></div>
<p>Look at the dataframe</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">)</span>
<span class="co">#&gt;   Year  Aou Route SpeciesTotal forest.total   SUM forest.percent NLCD.41</span>
<span class="co">#&gt; 1 2006 6080     2            6        26297 82345      0.3193515   25119</span>
<span class="co">#&gt; 2 2006 6080     3            6        34753 86328      0.4025693   34154</span>
<span class="co">#&gt; 3 2006 6080     4            3        41568 88526      0.4695570   39813</span>
<span class="co">#&gt; 4 2006 6080     5            1        41654 97708      0.4263110   32244</span>
<span class="co">#&gt; 5 2006 6080     6            8        68523 87503      0.7830932   50529</span>
<span class="co">#&gt; 6 2006 6080     8            5        56098 87248      0.6429718   49029</span>
<span class="co">#&gt;   NLCD.42 NLCD.43</span>
<span class="co">#&gt; 1    1130      48</span>
<span class="co">#&gt; 2     521      78</span>
<span class="co">#&gt; 3    1695      60</span>
<span class="co">#&gt; 4    5892    3518</span>
<span class="co">#&gt; 5    4674   13320</span>
<span class="co">#&gt; 6    4304    2765</span></code></pre></div>
<p>Compare the size of this dataframe and the original landcover dataframe.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#the BBS data that were merged</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_2</span><span class="op">)</span>
<span class="co">#&gt; [1] 82  4</span>

<span class="co">#the landcover data that were merged</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km_3</span><span class="op">)</span>
<span class="co">#&gt; [1] 137   7</span>

<span class="co">#the final merged dataframe</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">)</span>
<span class="co">#&gt; [1] 137  10</span></code></pre></div>
</div>
<div id="dealing-with-nas" class="section level3">
<h3 class="hasAnchor">
<a href="#dealing-with-nas" class="anchor"></a>Dealing with NAs</h3>
<p>If we look at our new dataframe we see that some of our rows now contains NAs</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">)</span>
<span class="co">#&gt;       Year           Aou           Route        SpeciesTotal     forest.total  </span>
<span class="co">#&gt;  Min.   :2006   Min.   :6080   Min.   :  1.0   Min.   : 1.000   Min.   : 1563  </span>
<span class="co">#&gt;  1st Qu.:2006   1st Qu.:6080   1st Qu.: 35.0   1st Qu.: 3.000   1st Qu.:35308  </span>
<span class="co">#&gt;  Median :2006   Median :6080   Median : 69.0   Median : 6.000   Median :51222  </span>
<span class="co">#&gt;  Mean   :2006   Mean   :6080   Mean   :137.2   Mean   : 8.341   Mean   :47706  </span>
<span class="co">#&gt;  3rd Qu.:2006   3rd Qu.:6080   3rd Qu.:132.0   3rd Qu.:10.000   3rd Qu.:62998  </span>
<span class="co">#&gt;  Max.   :2006   Max.   :6080   Max.   :911.0   Max.   :42.000   Max.   :84033  </span>
<span class="co">#&gt;  NA's   :55     NA's   :55                     NA's   :55                      </span>
<span class="co">#&gt;       SUM         forest.percent       NLCD.41         NLCD.42     </span>
<span class="co">#&gt;  Min.   : 63828   Min.   :0.01881   Min.   : 1377   Min.   :   59  </span>
<span class="co">#&gt;  1st Qu.: 84435   1st Qu.:0.40257   1st Qu.:27990   1st Qu.:  572  </span>
<span class="co">#&gt;  Median : 87008   Median :0.59019   Median :40555   Median : 1707  </span>
<span class="co">#&gt;  Mean   : 86994   Mean   :0.55450   Mean   :39383   Mean   : 2891  </span>
<span class="co">#&gt;  3rd Qu.: 89201   3rd Qu.:0.71135   3rd Qu.:50529   3rd Qu.: 4871  </span>
<span class="co">#&gt;  Max.   :110663   Max.   :0.96461   Max.   :73426   Max.   :11389  </span>
<span class="co">#&gt;                                                                    </span>
<span class="co">#&gt;     NLCD.43     </span>
<span class="co">#&gt;  Min.   :    0  </span>
<span class="co">#&gt;  1st Qu.:  531  </span>
<span class="co">#&gt;  Median : 1579  </span>
<span class="co">#&gt;  Mean   : 5432  </span>
<span class="co">#&gt;  3rd Qu.: 8346  </span>
<span class="co">#&gt;  Max.   :39534  </span>
<span class="co">#&gt; </span></code></pre></div>
<p>This occured b/c the original BBS data only contains data when a species is observed - if it isn’t seen, nothing is entered. So each “NA” in the new dataframe we made represents a route for which, in 2006, the SCTA wasn’t observed.</p>
<p>Its easy to fix the Year and Aou columns because the all have the same values. All of the years = 2006, and all of the Aou columns = 6080, for Scarlet tanager. The following code will fill in any of the missing values</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">Year</span> <span class="op">&lt;-</span> <span class="fl">2006</span>
<span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">Aou</span> <span class="op">&lt;-</span> <span class="fl">6080</span></code></pre></div>
<p>Actualy, since the code “6080” isn’t very meaningful, let’s add the letters “SCTA” to a column to make it easy to remembe what we are looking at. Let’s make a new column called “name” and put “SCTA” in it.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"SCTA"</span></code></pre></div>
<p>We’ll make this a factor variable</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></code></pre></div>
<p>Summary will show us what we’ver done: now there are no NAs in Year or Aou and there’s a new column calld name</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">)</span>
<span class="co">#&gt;       Year           Aou           Route        SpeciesTotal     forest.total  </span>
<span class="co">#&gt;  Min.   :2006   Min.   :6080   Min.   :  1.0   Min.   : 1.000   Min.   : 1563  </span>
<span class="co">#&gt;  1st Qu.:2006   1st Qu.:6080   1st Qu.: 35.0   1st Qu.: 3.000   1st Qu.:35308  </span>
<span class="co">#&gt;  Median :2006   Median :6080   Median : 69.0   Median : 6.000   Median :51222  </span>
<span class="co">#&gt;  Mean   :2006   Mean   :6080   Mean   :137.2   Mean   : 8.341   Mean   :47706  </span>
<span class="co">#&gt;  3rd Qu.:2006   3rd Qu.:6080   3rd Qu.:132.0   3rd Qu.:10.000   3rd Qu.:62998  </span>
<span class="co">#&gt;  Max.   :2006   Max.   :6080   Max.   :911.0   Max.   :42.000   Max.   :84033  </span>
<span class="co">#&gt;                                                NA's   :55                      </span>
<span class="co">#&gt;       SUM         forest.percent       NLCD.41         NLCD.42     </span>
<span class="co">#&gt;  Min.   : 63828   Min.   :0.01881   Min.   : 1377   Min.   :   59  </span>
<span class="co">#&gt;  1st Qu.: 84435   1st Qu.:0.40257   1st Qu.:27990   1st Qu.:  572  </span>
<span class="co">#&gt;  Median : 87008   Median :0.59019   Median :40555   Median : 1707  </span>
<span class="co">#&gt;  Mean   : 86994   Mean   :0.55450   Mean   :39383   Mean   : 2891  </span>
<span class="co">#&gt;  3rd Qu.: 89201   3rd Qu.:0.71135   3rd Qu.:50529   3rd Qu.: 4871  </span>
<span class="co">#&gt;  Max.   :110663   Max.   :0.96461   Max.   :73426   Max.   :11389  </span>
<span class="co">#&gt;                                                                    </span>
<span class="co">#&gt;     NLCD.43        name    </span>
<span class="co">#&gt;  Min.   :    0   SCTA:137  </span>
<span class="co">#&gt;  1st Qu.:  531             </span>
<span class="co">#&gt;  Median : 1579             </span>
<span class="co">#&gt;  Mean   : 5432             </span>
<span class="co">#&gt;  3rd Qu.: 8346             </span>
<span class="co">#&gt;  Max.   :39534             </span>
<span class="co">#&gt; </span></code></pre></div>
<p>To fill in the NAs for the SpeciesTotal column (the counts of the number of birds) requires a new function: is.na(). is.na() determiens if a row in a column has NA or it doesn’t. is.na() returns “TRUE” whenever there is an NA</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">SpeciesTotal</span><span class="op">)</span>
<span class="co">#&gt;   [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE</span>
<span class="co">#&gt;  [85]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt;  [97]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt; [109]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt; [121]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt; [133]  TRUE  TRUE  TRUE  TRUE  TRUE</span></code></pre></div>
<p>We can use is.na with a function from dplyr called mutate() to change these NAs to 0s.</p>
<p><strong>This code is pretty complex, actually, so don’t worry if you don’t get it the 1st try</strong> Actually, since its pretty trick for beginers, I’ve made a new function that does it more simply (see next chunk of code)</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NA_to_zero.html">NA_to_zero</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">BBS_PA_SCTA_3</span>,column <span class="op">=</span> <span class="st">"SpeciesTotal"</span><span class="op">)</span></code></pre></div>
<p>If you compare the dataset BBS_PA_SCTA_3 to BBS_PA_SCTA_4 that was made with NA_to_zero() you can see that the NAs were removed</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#with NAs</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_3</span><span class="op">$</span><span class="va">SpeciesTotal</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's </span>
<span class="co">#&gt;   1.000   3.000   6.000   8.341  10.000  42.000      55</span>

<span class="co">#with NAs removed by NA_to_zero()</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_4</span><span class="op">$</span><span class="va">SpeciesTotal</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;   0.000   0.000   2.000   4.993   7.000  42.000</span></code></pre></div>
<p>Since the above code is rather long, the following function from the wildlifeR package will do the same thing</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NA_to_zero.html">NA_to_zero</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">BBS_PA_SCTA_3</span>,
           column <span class="op">=</span> <span class="st">"SpeciesTotal"</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-relationship-between-scta-and-forest-cover" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-relationship-between-scta-and-forest-cover" class="anchor"></a>PLot relationship between SCTA and forest cover</h3>
<p>We can now finally check out the relationship between the Scarlet tanger and forest cover</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_SCTA_3</span>,
          y <span class="op">=</span> <span class="st">"SpeciesTotal"</span>,
          x <span class="op">=</span> <span class="st">"forest.percent"</span><span class="op">)</span></code></pre></div>
<p><img src="data_cleaning-merge_BBS_with_landcover_files/figure-html/unnamed-chunk-42-1.png" width="700"></p>
</div>
<div id="re-orgainzing-rows" class="section level3">
<h3 class="hasAnchor">
<a href="#re-orgainzing-rows" class="anchor"></a>Re-orgainzing rows</h3>
<ul>
<li>Dataframes typically put the information columns to the left and the data columns to the right</li>
<li>We can reorder our columns in BBS_PA_SCTA_3 using select()</li>
<li>We can also rename columns using the selection function using the format “new.name = old.name”</li>
<li>Its generally good to use a consistent format for all of your column names</li>
<li>I also think its good to avoid uppercase letters</li>
<li>Using using standard abbreviations is also good b/c shorter names are easier to type; here I’ll use “spp” instead of “speices” , “for” instead of forest, and “tot” instead of “total”</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_4</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_SCTA_3</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>year <span class="op">=</span> <span class="va">Year</span>, 
                              aou <span class="op">=</span> <span class="va">Aou</span>, 
                              route <span class="op">=</span> <span class="va">Route</span>, 
                              name <span class="op">=</span> <span class="va">name</span>,
                              spp.tot <span class="op">=</span> <span class="va">SpeciesTotal</span>,
                              for.tot <span class="op">=</span> <span class="va">forest.total</span>,
                              NLCD.sum <span class="op">=</span> <span class="va">SUM</span>,
                              for.percent <span class="op">=</span> <span class="va">forest.percent</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_SCTA_4</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_SCTA_3</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>year <span class="op">=</span> <span class="va">Year</span>, 
                              aou <span class="op">=</span> <span class="va">Aou</span>, 
                              route <span class="op">=</span> <span class="va">Route</span>, 
                              name <span class="op">=</span> <span class="va">name</span>,
                              spp.tot <span class="op">=</span> <span class="va">SpeciesTotal</span>,
                              for.tot <span class="op">=</span> <span class="va">forest.total</span>,
                              NLCD.sum <span class="op">=</span> <span class="va">SUM</span>,
                              for.percent <span class="op">=</span> <span class="va">forest.percent</span><span class="op">)</span></code></pre></div>
</div>
<div id="saving-to--csv" class="section level3">
<h3 class="hasAnchor">
<a href="#saving-to--csv" class="anchor"></a>Saving to .csv</h3>
<p>When we have a dataset prepared its a good idea to save it to a spreadsheet as a .csv file for safe keeping. This is done with the write.csv() function</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.csv</a></span><span class="op">(</span><span class="va">BBS_PA_SCTA_4</span>, file <span class="op">=</span> <span class="st">"SCTA_vs_forest_cover.csv"</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
