<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Selecting different species and landovers from the wildlifeR package • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Selecting different species and landovers from the wildlifeR package">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="data_cleaning_selecting_different_spp_and_landcovers_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Selecting different species and landovers from the wildlifeR package</h1>
            
      
      
      <div class="hidden name"><code>data_cleaning_selecting_different_spp_and_landcovers.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Most examples associated with the BBS data from the wildlifeR package relate to forest birds and how their abundance changes with variation in forest cover (eg the Scarlet Tanager and the Pileated Woodpecker). In this example we’ll look at urban and forest habitats.</p>
<div id="installing-widlifer" class="section level3">
<h3 class="hasAnchor">
<a href="#installing-widlifer" class="anchor"></a>Installing widlifeR</h3>
<p>The current version of wildlifeR can be downloaded like this:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load devtools</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: usethis</span>

<span class="co">#download wildlifeR from github</span>
<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"brouwern/wildlifeR"</span><span class="op">)</span>
<span class="co">#&gt; Skipping install of 'wildlifeR' from a github remote, the SHA1 (29f3ae31) has not changed since last install.</span>
<span class="co">#&gt;   Use `force = TRUE` to force installation</span>

<span class="co">#load wildlifeR into your current R sessions</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brouwern/wildlifeR">wildlifeR</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'wildlifeR'</span>
<span class="co">#&gt; The following objects are masked from 'package:compbio4all':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine_NLCD_classes, expand_BBS_routes, make_my_data2L,</span>
<span class="co">#&gt;     NA_to_zero, sample_BBS_routes, sample_dove_routes</span></code></pre></div>
</div>
<div id="selecting-a-focal-bird-species" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-a-focal-bird-species" class="anchor"></a>Selecting a focal bird species</h3>
<p>We’ll look at the Brown Headed Cowbirds. Its alphabetic code is “BHCO”.</p>
</div>
<div id="determing-the-numeric-species-code" class="section level3">
<h3 class="hasAnchor">
<a href="#determing-the-numeric-species-code" class="anchor"></a>Determing the numeric species code</h3>
<p>In their database the USGS uses AOU numeric codes, <strong>not</strong> alphabetic codes. We can look up numeric codes using the dataset AOU_species_codes included in wildlifeR.</p>
<p>First, load the AOU code data</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">AOU_species_codes</span><span class="op">)</span></code></pre></div>
<p>Then make sure the dplyr package is loaded</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span></code></pre></div>
<p>We can see the entry for BHCO and its numeric code in the “spp.num” code using the dplyr function filter()</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">AOU_species_codes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">alpha.code</span> <span class="op">==</span> <span class="st">"BHCO"</span><span class="op">)</span>
<span class="co">#&gt;   spp.num alpha.code                 name            spp order</span>
<span class="co">#&gt; 1    4950       BHCO Brown-headed Cowbird Molothrus ater  1089</span></code></pre></div>
<p>A more detailed walk through of this process can be found in the vignette “Data Cleaning: Filtering focal rows with dplyr”. This vignette can most easily be accessed at the wildlifeR website <a href="https://brouwern.github.io/wildlifeR/" class="uri">https://brouwern.github.io/wildlifeR/</a> .</p>
<p>The filter() code above tells us that the the numeric code is 4950 We can use this number to extract just the BHCO data. We’ll save this data to an object called “BBA_PA_BHCO”. (Note that the species code is <strong>not</strong> the same row number of the dataframe).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_BHCO</span> <span class="op">&lt;-</span> <span class="va">BBS_PA</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Aou</span> <span class="op">==</span> <span class="st">"4950"</span><span class="op">)</span></code></pre></div>
</div>
<div id="selecting-a-focal-analysis-question" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-a-focal-analysis-question" class="anchor"></a>Selecting a focal analysis question</h3>
<p>We’ll be examining this data:</p>
<ul>
<li>How does the abundance of a focal bird species in Pennsylvania vary with the type of habitat in the landscape.</li>
</ul>
</div>
<div id="formatting-data-for-habitat-analyses" class="section level3">
<h3 class="hasAnchor">
<a href="#formatting-data-for-habitat-analyses" class="anchor"></a>Formatting data for habitat analyses</h3>
<div id="subsetting-bird-data" class="section level4">
<h4 class="hasAnchor">
<a href="#subsetting-bird-data" class="anchor"></a>Subsetting bird data</h4>
<p>The habitat data used wildlifeR was collected in 2006. We will therefore use the filter() command again to isolate just that year of data. (Note that “Year” has a capital “Y” while the rest is lowercase).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_BHCO_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_BHCO</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Year</span> <span class="op">==</span> <span class="fl">2006</span><span class="op">)</span></code></pre></div>
</div>
<div id="preparing-habitat-data" class="section level4">
<h4 class="hasAnchor">
<a href="#preparing-habitat-data" class="anchor"></a>Preparing habitat data</h4>
<p>The habitat data are found in the dataframe BBS_PA_landcover_1km. For each BBS route the number of pixels of different habitat was determined using a GIS. These data are from the National Land Cover Dataset (NCLD) and each column is a different type of cover.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km</span><span class="op">)</span></code></pre></div>
<p>Urban cover classes are numbers 21 through 24. These represent variosu “developed” landcovers with different intensities of develoement and impervious surfaces. See the help file using ?BBS_PA_landcover_1km for more information. Forest land covers are numbers 41, 42, 43.</p>
<p>W’ll isolate both the urban and forestlandcovers and the Route number in a new dataframe called BBS_PA_landcover_1km_2. We’ll also take the “SUM” column, which is the total number of pixels in the buffer around the BBS route (this varies somewhat between routes and so landcover needs to be expressed as a proportions; see below)</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">Route</span>, 
         <span class="va">NLCD.21</span>, <span class="co"># Developed Open Space</span>
         <span class="va">NLCD.22</span>, <span class="co"># ... Low Intensity</span>
         <span class="va">NLCD.23</span>, <span class="co"># ... Medium Intensity</span>
         <span class="va">NLCD.24</span>, <span class="co"># ... High Intensity</span>
         <span class="va">NLCD.41</span>, <span class="co"># Deciduous forest</span>
         <span class="va">NLCD.42</span>, <span class="co"># Evergreen</span>
         <span class="va">NLCD.43</span>, <span class="co"># Mixed</span>
         <span class="va">SUM</span><span class="op">)</span></code></pre></div>
</div>
<div id="combining-landcovers" class="section level4">
<h4 class="hasAnchor">
<a href="#combining-landcovers" class="anchor"></a>Combining landcovers</h4>
<div id="combining-urban-landcover-classes" class="section level5">
<h5 class="hasAnchor">
<a href="#combining-urban-landcover-classes" class="anchor"></a>Combining urban landcover classes</h5>
<p>For the sake of simplicity let’s combine all four of these urban landcovers into a single “total.urban” land cover class. We’ll add up the four NCLD columns put the result into a new columns “total.urban.”</p>
<p>We’ll use dplyr’s mutate() command to accheive this. We “pipe” the BBS_PA_landcover_1km_2 dataframe to mutate(); within mutate we say we want a new column “total.urban”, and that this new column will be built by adding up for columns from within BBS_PA_landcover_1km_2.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>total.urban <span class="op">=</span>  <span class="va">NLCD.21</span><span class="op">+</span><span class="va">NLCD.22</span><span class="op">+</span><span class="va">NLCD.23</span><span class="op">+</span><span class="va">NLCD.24</span><span class="op">)</span></code></pre></div>
<p>The results looks like this:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">BBS_PA_landcover_1km_2</span><span class="op">)</span>
<span class="co">#&gt;   Route NLCD.21 NLCD.22 NLCD.23 NLCD.24 NLCD.41 NLCD.42 NLCD.43   SUM</span>
<span class="co">#&gt; 1     1    9025    4861    1085     230   44953     563     797 98591</span>
<span class="co">#&gt; 2     2   11340    3425     142       6   25119    1130      48 82345</span>
<span class="co">#&gt; 3     3    5638    2078     344      34   34154     521      78 86328</span>
<span class="co">#&gt; 4     4    4757    1753     250      32   39813    1695      60 88526</span>
<span class="co">#&gt; 5     5    5273    1238     269      15   32244    5892    3518 97708</span>
<span class="co">#&gt; 6     6    3444     384      71       7   50529    4674   13320 87503</span>
<span class="co">#&gt;   total.urban</span>
<span class="co">#&gt; 1       15201</span>
<span class="co">#&gt; 2       14913</span>
<span class="co">#&gt; 3        8094</span>
<span class="co">#&gt; 4        6792</span>
<span class="co">#&gt; 5        6795</span>
<span class="co">#&gt; 6        3906</span></code></pre></div>
</div>
<div id="combining-forst-land-cover-classes" class="section level5">
<h5 class="hasAnchor">
<a href="#combining-forst-land-cover-classes" class="anchor"></a>Combining forst land cover classes</h5>
<p>We’ll do the same thing for the forest landcovers</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>total.forest <span class="op">=</span>  <span class="va">NLCD.41</span><span class="op">+</span><span class="va">NLCD.42</span><span class="op">+</span><span class="va">NLCD.43</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="calculating-proportionages-of-habitat" class="section level4">
<h4 class="hasAnchor">
<a href="#calculating-proportionages-of-habitat" class="anchor"></a>Calculating proportionages of habitat</h4>
<p>The raw habitat data are actual counts of the number of pixels classified as a given habitat by a GIS. The number of pixels varies a little bit between routes, so to make everything comparable we need to calculate the proportionage of each forest cover. This can be done by dividing each habitat class by the SUM column</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">urban.proportion</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">total.urban</span><span class="op">/</span><span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">SUM</span>

<span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">forest.proportion</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">total.forest</span><span class="op">/</span><span class="va">BBS_PA_landcover_1km_2</span><span class="op">$</span><span class="va">SUM</span></code></pre></div>
</div>
<div id="merging-bird-counts-and-habitat-cover" class="section level4">
<h4 class="hasAnchor">
<a href="#merging-bird-counts-and-habitat-cover" class="anchor"></a>Merging bird counts and habitat cover</h4>
<p>Now we need to line up the data on the number of BHCO observed on each route in our BBA_PA_BHCO_2 dataframe with the habitat data in BBS_PA_landcover_1km_2. This can be done using the full_join() function in dplyr. This joins (combines) the two dataframes together.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_BHCO_3</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">BBS_PA_BHCO_2</span> ,
                       <span class="va">BBS_PA_landcover_1km_2</span>, 
                       by <span class="op">=</span> <span class="st">"Route"</span><span class="op">)</span></code></pre></div>
<p>The habitat data in BBS_PA_landcover_1km_2 contains data from each route in PA, but BHCO wasn’t seen on every route. To line things up dplyr has inserted NA value where data were missing from BBA_PA_BHCO_2. We can get rid of these NAs like this</p>
<p>First, make all the “years” equal to 2006, all the Aou codes equal to 4950, and assign an alphabetic species code for each reference</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_BHCO_3</span><span class="op">$</span><span class="va">Year</span> <span class="op">&lt;-</span> <span class="fl">2006</span>
<span class="va">BBS_PA_BHCO_3</span><span class="op">$</span><span class="va">Aou</span> <span class="op">&lt;-</span> <span class="fl">4950</span>
<span class="va">BBS_PA_BHCO_3</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"BHCO"</span></code></pre></div>
<p>There are NA values in the “StopTotal” column, which has our counts of the number of birds. We want these to be zero. We can fix this, using the function NA_to_zero() from the wildlifeR package.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_BHCO_4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NA_to_zero.html">NA_to_zero</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">BBS_PA_BHCO_3</span>,
           column <span class="op">=</span> <span class="st">"SpeciesTotal"</span><span class="op">)</span></code></pre></div>
<p>The data are now read for analysis. To save the data for future use use the write.csv() command.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.csv</a></span><span class="op">(</span><span class="va">BBS_PA_BHCO_4</span>, file <span class="op">=</span> <span class="st">"BHCO_vs_forest_cover.csv"</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="plotting-data" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-data" class="anchor"></a>Plotting Data</h2>
<div id="exploratory-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#exploratory-analysis" class="anchor"></a>Exploratory analysis</h3>
<p>THis dataset contains an extreme outlier where &gt;75 cowbirds were seen. This is suspect to me. We can visualize this using a boxplot and a histogram</p>
<div id="boxplot" class="section level4">
<h4 class="hasAnchor">
<a href="#boxplot" class="anchor"></a>Boxplot</h4>
<p>The function ggboxplot() in the ggpubr package make boxplots.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span>

<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggboxplot.html">ggboxplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_BHCO_4</span>,
         y <span class="op">=</span> <span class="st">"SpeciesTotal"</span>,
         ylab <span class="op">=</span> <span class="st">""</span>,
         xlab <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p><img src="data_cleaning_selecting_different_spp_and_landcovers_files/figure-html/unnamed-chunk-18-1.png" width="192"></p>
</div>
<div id="histograms" class="section level4">
<h4 class="hasAnchor">
<a href="#histograms" class="anchor"></a>Histograms</h4>
<p>gghistogram() makes histograms. Note that we use the arguement x = “SpeciesTotal”.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/gghistogram.html">gghistogram</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_BHCO_4</span>,
         x <span class="op">=</span> <span class="st">"SpeciesTotal"</span><span class="op">)</span>
<span class="co">#&gt; Warning: Using `bins = 30` by default. Pick better value with the argument</span>
<span class="co">#&gt; `bins`.</span></code></pre></div>
<p><img src="data_cleaning_selecting_different_spp_and_landcovers_files/figure-html/unnamed-chunk-19-1.png" width="192"></p>
</div>
</div>
</div>
<div id="removing-an-outlier3" class="section level2">
<h2 class="hasAnchor">
<a href="#removing-an-outlier3" class="anchor"></a>Removing an outlier3</h2>
<p>This observation is very very odd so I am going to remove it. Normally I would do a very thorough investigation of this datapoint and try to go back to the raw data if possible to determine what is going on. For the sake of this exercise, I will make the very strong assumption that its a complete error (eg typo during data entry) and remove it. This should only be done to your own data with extreme care, thorough documentation, and total tranparency.</p>
<p>I can use a logical arguement using the “&gt;” operator to identify the row that the outlier is in.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">BBS_PA_BHCO_4</span><span class="op">$</span><span class="va">SpeciesTotal</span> <span class="op">&gt;</span> <span class="fl">75</span><span class="op">)</span>
<span class="co">#&gt; [1] 76</span></code></pre></div>
<p>I will save this row number to an object called “outlier.index” then set the value of SpeciesTotal in that row to NA</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outlier.index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">BBS_PA_BHCO_4</span><span class="op">$</span><span class="va">SpeciesTotal</span> <span class="op">&gt;</span> <span class="fl">75</span><span class="op">)</span>

<span class="va">BBS_PA_BHCO_4</span><span class="op">$</span><span class="va">SpeciesTotal</span><span class="op">[</span><span class="va">outlier.index</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></code></pre></div>
<p>Again, I would normally only do this after very thorough investigation as to the origin of this extreme value.</p>
</div>
<div id="plotting-polished-graphs" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-polished-graphs" class="anchor"></a>Plotting polished graphs</h2>
<p>We can visualize the relationship between the Brown-headed cowbird and forest cover using a scatterplot using the ggpubr extentsion to ggplot2. Since we’re examining two variables we’ll make a two-panel plot with total urban habitat (as a proportion) on the left and total forest habitat on the right.</p>
<p>To set up this graph we’ll use the plotgrid() function from the cowplot library.</p>
<p>First, we need to make to plots and assign them to R objects. We’ll call them urban.plot and forest.plot. Note that to make these plots look nice I do the following things</p>
<ul>
<li>Add a regression line using add = “reg.line”</li>
<li>Add a confidence interval to the line using conf.int = TRUE,</li>
<li>Label the x and y axes using xlab and ylab</li>
<li>Change the shape of the points in the 2nd graph using shape = 23</li>
<li>Change the fill color using fill = “geen”</li>
</ul>
<p>Note that you can only change hte fill color for certain shape numbers (21 through 25).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">urban.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_BHCO_4</span>,
          y <span class="op">=</span> <span class="st">"SpeciesTotal"</span>,
          x <span class="op">=</span> <span class="st">"urban.proportion"</span>,
          add <span class="op">=</span> <span class="st">"reg.line"</span>,
          conf.int <span class="op">=</span> <span class="cn">TRUE</span>,
          xlab <span class="op">=</span> <span class="st">"Proportion urban habitat"</span>,
          ylab <span class="op">=</span> <span class="st">"Number of cowbirds"</span><span class="op">)</span>

<span class="va">forest.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">BBS_PA_BHCO_4</span>,
          y <span class="op">=</span> <span class="st">"SpeciesTotal"</span>,
          x <span class="op">=</span> <span class="st">"forest.proportion"</span>,
          add <span class="op">=</span> <span class="st">"reg.line"</span>,
          conf.int <span class="op">=</span> <span class="cn">TRUE</span>,
          xlab <span class="op">=</span> <span class="st">"Proportion forest habitat"</span>,
          ylab <span class="op">=</span> <span class="st">"Number of cowbirds"</span>,
          shape <span class="op">=</span> <span class="fl">23</span>,
          fill <span class="op">=</span> <span class="st">"green"</span><span class="op">)</span></code></pre></div>
<p>Now I’ll use plot_grid() to layout the two panels</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'cowplot'</span>
<span class="co">#&gt; The following object is masked from 'package:ggpubr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     get_legend</span>

<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">urban.plot</span>, 
          <span class="va">forest.plot</span>,
          labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; `geom_smooth()` using formula 'y ~ x'</span>
<span class="co">#&gt; Warning: Removed 1 rows containing non-finite values (stat_smooth).</span>
<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span>
<span class="co">#&gt; `geom_smooth()` using formula 'y ~ x'</span>
<span class="co">#&gt; Warning: Removed 1 rows containing non-finite values (stat_smooth).</span>

<span class="co">#&gt; Warning: Removed 1 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="data_cleaning_selecting_different_spp_and_landcovers_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
</div>
<div id="data-modeling" class="section level2">
<h2 class="hasAnchor">
<a href="#data-modeling" class="anchor"></a>Data modeling</h2>
<div id="modeling-habitat" class="section level3">
<h3 class="hasAnchor">
<a href="#modeling-habitat" class="anchor"></a>Modeling habitat</h3>
<p>Three models</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m.null</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal</span> <span class="op">~</span> <span class="fl">1</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span>

<span class="va">m.urban</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal</span> <span class="op">~</span> <span class="va">urban.proportion</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span>

<span class="va">m.forest</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal</span> <span class="op">~</span> <span class="va">forest.proportion</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span></code></pre></div>
</div>
<div id="present-results" class="section level3">
<h3 class="hasAnchor">
<a href="#present-results" class="anchor"></a>Present results</h3>
<div id="tidy-results" class="section level4">
<h4 class="hasAnchor">
<a href="#tidy-results" class="anchor"></a>Tidy results</h4>
<p>Look at “tidy” output using the tidy() function from the broom package. This is brief but doesn’t contain everything. For more info on broom see <a href="https://cran.r-project.org/web/packages/broom/vignettes/broom.html" class="uri">https://cran.r-project.org/web/packages/broom/vignettes/broom.html</a></p>
<p>We’ll also set R to round things off for us a bit using the option options(digits=4)</p>
<p>Note that the output of tidy() will often be in scientific notation, where “e-x” is used to indicate what is often written as “10^-x”. Eg, “5e-2” equals 5*10^-2, which equals 0.05. 5e-3 equals 0.005.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://broom.tidymodels.org/">broom</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html">tidy</a></span><span class="op">(</span><span class="va">m.null</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 5</span>
<span class="co">#&gt;   term        estimate std.error statistic  p.value</span>
<span class="co">#&gt;   &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;</span>
<span class="co">#&gt; 1 (Intercept)     3.74     0.451      8.29 1.01e-13</span>

<span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html">tidy</a></span><span class="op">(</span><span class="va">m.urban</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 2 x 5</span>
<span class="co">#&gt;   term             estimate std.error statistic    p.value</span>
<span class="co">#&gt;   &lt;chr&gt;               &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;      &lt;dbl&gt;</span>
<span class="co">#&gt; 1 (Intercept)          3.39     0.728     4.65  0.00000787</span>
<span class="co">#&gt; 2 urban.proportion     3.17     5.06      0.626 0.532</span>

<span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html">tidy</a></span><span class="op">(</span><span class="va">m.forest</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 2 x 5</span>
<span class="co">#&gt;   term              estimate std.error statistic   p.value</span>
<span class="co">#&gt;   &lt;chr&gt;                &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt; 1 (Intercept)           5.32      1.16      4.57 0.0000110</span>
<span class="co">#&gt; 2 forest.proportion    -2.83      1.93     -1.47 0.145</span></code></pre></div>
</div>
</div>
<div id="full-results" class="section level3">
<h3 class="hasAnchor">
<a href="#full-results" class="anchor"></a>Full results</h3>
<p>Look at full output using summary()</p>
<div id="full-results-null-model" class="section level4">
<h4 class="hasAnchor">
<a href="#full-results-null-model" class="anchor"></a>Full results null model</h4>
<p>The null model has only an interecept. Its summary() output is not particularly interesting. The intecept here is highly significant, but this really isn’t very useful to us.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">m.null</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; lm(formula = SpeciesTotal ~ 1, data = BBS_PA_BHCO_4)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residuals:</span>
<span class="co">#&gt;    Min     1Q Median     3Q    Max </span>
<span class="co">#&gt;  -3.74  -3.74  -1.74   2.26  26.26 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="co">#&gt; (Intercept)    3.743      0.451    8.29    1e-13 ***</span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 5.26 on 135 degrees of freedom</span>
<span class="co">#&gt;   (1 observation deleted due to missingness)</span></code></pre></div>
<div style="page-break-after: always;"></div>
</div>
<div id="full-results-urban-landcover-model" class="section level4">
<h4 class="hasAnchor">
<a href="#full-results-urban-landcover-model" class="anchor"></a>Full results urban landcover model</h4>
<p>The urban landcover model has two parameters: intercept and the slope. Note that R doesn’t provide names that are necessarily easy to understand. It also provides things printed out to lots of decimal places, which can be distracting, hence why we set options(digits=4).</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">m.urban</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; lm(formula = SpeciesTotal ~ urban.proportion, data = BBS_PA_BHCO_4)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residuals:</span>
<span class="co">#&gt;    Min     1Q Median     3Q    Max </span>
<span class="co">#&gt;  -4.95  -3.66  -1.61   2.23  26.13 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;                  Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="co">#&gt; (Intercept)         3.385      0.728    4.65  7.9e-06 ***</span>
<span class="co">#&gt; urban.proportion    3.170      5.061    0.63     0.53    </span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 5.28 on 134 degrees of freedom</span>
<span class="co">#&gt;   (1 observation deleted due to missingness)</span>
<span class="co">#&gt; Multiple R-squared:  0.00292,    Adjusted R-squared:  -0.00452 </span>
<span class="co">#&gt; F-statistic: 0.392 on 1 and 134 DF,  p-value: 0.532</span></code></pre></div>
</div>
<div id="full-results-forest-model" class="section level4">
<h4 class="hasAnchor">
<a href="#full-results-forest-model" class="anchor"></a>Full results forest model</h4>
<p>The mixed forest model has the same basic format, though the values are different.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">m.forest</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; lm(formula = SpeciesTotal ~ forest.proportion, data = BBS_PA_BHCO_4)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residuals:</span>
<span class="co">#&gt;    Min     1Q Median     3Q    Max </span>
<span class="co">#&gt;  -5.26  -3.65  -1.57   2.36  25.63 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt;                   Estimate Std. Error t value Pr(&gt;|t|)    </span>
<span class="co">#&gt; (Intercept)           5.32       1.16    4.57  1.1e-05 ***</span>
<span class="co">#&gt; forest.proportion    -2.83       1.93   -1.47     0.14    </span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 5.24 on 134 degrees of freedom</span>
<span class="co">#&gt;   (1 observation deleted due to missingness)</span>
<span class="co">#&gt; Multiple R-squared:  0.0158, Adjusted R-squared:  0.00844 </span>
<span class="co">#&gt; F-statistic: 2.15 on 1 and 134 DF,  p-value: 0.145</span></code></pre></div>
</div>
</div>
<div id="hypothesis-testing" class="section level3">
<h3 class="hasAnchor">
<a href="#hypothesis-testing" class="anchor"></a>Hypothesis testing</h3>
<p>P-values from the summary() command are not to be used for final analyses. To compare to models and get a p-value use the anova() command.</p>
<div id="deciduous-versus-null" class="section level4">
<h4 class="hasAnchor">
<a href="#deciduous-versus-null" class="anchor"></a>Deciduous versus null</h4>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">m.null</span>, 
      <span class="va">m.urban</span><span class="op">)</span>
<span class="co">#&gt; Analysis of Variance Table</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Model 1: SpeciesTotal ~ 1</span>
<span class="co">#&gt; Model 2: SpeciesTotal ~ urban.proportion</span>
<span class="co">#&gt;   Res.Df  RSS Df Sum of Sq    F Pr(&gt;F)</span>
<span class="co">#&gt; 1    135 3742                         </span>
<span class="co">#&gt; 2    134 3731  1      10.9 0.39   0.53</span></code></pre></div>
</div>
<div id="mixed-versus-null" class="section level4">
<h4 class="hasAnchor">
<a href="#mixed-versus-null" class="anchor"></a>Mixed versus null</h4>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">m.null</span>, 
      <span class="va">m.forest</span><span class="op">)</span>
<span class="co">#&gt; Analysis of Variance Table</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Model 1: SpeciesTotal ~ 1</span>
<span class="co">#&gt; Model 2: SpeciesTotal ~ forest.proportion</span>
<span class="co">#&gt;   Res.Df  RSS Df Sum of Sq    F Pr(&gt;F)</span>
<span class="co">#&gt; 1    135 3742                         </span>
<span class="co">#&gt; 2    134 3683  1      59.1 2.15   0.14</span></code></pre></div>
</div>
<div id="model-comparison" class="section level4">
<h4 class="hasAnchor">
<a href="#model-comparison" class="anchor"></a>Model comparison</h4>
<p>Compare models w/AIC</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/bbolker/bbmle">bbmle</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'bbmle'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     slice</span>
<span class="fu"><a href="https://rdrr.io/pkg/bbmle/man/ICtab.html">AICtab</a></span><span class="op">(</span><span class="va">m.null</span>,
       <span class="va">m.urban</span>,
       <span class="va">m.forest</span><span class="op">)</span>
<span class="co">#&gt;          dAIC df</span>
<span class="co">#&gt; m.forest 0.0  3 </span>
<span class="co">#&gt; m.null   0.2  2 </span>
<span class="co">#&gt; m.urban  1.8  3</span></code></pre></div>
</div>
</div>
</div>
<div id="transformations" class="section level2">
<h2 class="hasAnchor">
<a href="#transformations" class="anchor"></a>Transformations</h2>
<div id="carry-out-log-transformation" class="section level3">
<h3 class="hasAnchor">
<a href="#carry-out-log-transformation" class="anchor"></a>Carry out log transformation</h3>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BBS_PA_BHCO_4</span> <span class="op">&lt;-</span> <span class="va">BBS_PA_BHCO_4</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>SpeciesTotal.log <span class="op">=</span> 
                                             <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="va">SpeciesTotal</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="build-new-models" class="section level3">
<h3 class="hasAnchor">
<a href="#build-new-models" class="anchor"></a>Build new models</h3>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m.null.log</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal.log</span> <span class="op">~</span> <span class="fl">1</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span>

<span class="va">m.urban.log</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal.log</span> <span class="op">~</span> <span class="va">urban.proportion</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span>

<span class="va">m.forest.log</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal.log</span> <span class="op">~</span> <span class="va">forest.proportion</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span></code></pre></div>
</div>
<div id="compare-with-anova" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-with-anova" class="anchor"></a>Compare with anova()</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">m.null.log</span>,
        <span class="va">m.urban.log</span><span class="op">)</span>
<span class="co">#&gt; Analysis of Variance Table</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Model 1: SpeciesTotal.log ~ 1</span>
<span class="co">#&gt; Model 2: SpeciesTotal.log ~ urban.proportion</span>
<span class="co">#&gt;   Res.Df RSS Df Sum of Sq    F Pr(&gt;F)</span>
<span class="co">#&gt; 1    135 138                         </span>
<span class="co">#&gt; 2    134 137  1     0.907 0.89   0.35</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">m.null.log</span>,
        <span class="va">m.forest.log</span><span class="op">)</span>
<span class="co">#&gt; Analysis of Variance Table</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Model 1: SpeciesTotal.log ~ 1</span>
<span class="co">#&gt; Model 2: SpeciesTotal.log ~ forest.proportion</span>
<span class="co">#&gt;   Res.Df RSS Df Sum of Sq    F Pr(&gt;F)</span>
<span class="co">#&gt; 1    135 138                         </span>
<span class="co">#&gt; 2    134 137  1     0.385 0.38   0.54</span></code></pre></div>
</div>
<div id="compare-with-aic" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-with-aic" class="anchor"></a>COmpare with AIC</h3>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/bbmle/man/ICtab.html">AICtab</a></span><span class="op">(</span><span class="va">m.null.log</span>,
       <span class="va">m.urban.log</span>,
       <span class="va">m.forest.log</span><span class="op">)</span>
<span class="co">#&gt;              dAIC df</span>
<span class="co">#&gt; m.null.log   0.0  2 </span>
<span class="co">#&gt; m.urban.log  1.1  3 </span>
<span class="co">#&gt; m.forest.log 1.6  3</span></code></pre></div>
</div>
<div id="model-curved-relationship" class="section level3">
<h3 class="hasAnchor">
<a href="#model-curved-relationship" class="anchor"></a>Model curved relationship</h3>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m.forest.log2</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">SpeciesTotal.log</span> <span class="op">~</span> <span class="va">urban.proportion</span> <span class="op">+</span>
                      <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">forest.proportion</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span>, <span class="va">BBS_PA_BHCO_4</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
