<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>h) Introduction to biological sequences databases • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="h) Introduction to biological sequences databases">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ddd-ac-h-NCBI_by_GUI2_AC03-1Q_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>h) Introduction to biological sequences databases</h1>
            
      
      
      <div class="hidden name"><code>ddd-ac-h-NCBI_by_GUI2_AC03-1Q.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<p><strong>By</strong>: Avril Coghlan.</p>
<p><strong>Adapted, edited and expanded</strong>: Nathan Brouwer under the Creative Commons 3.0 Attribution License <a href="https://creativecommons.org/licenses/by/3.0/">(CC BY 3.0)</a>.</p>
<div id="to-do" class="section level2">
<h2 class="hasAnchor">
<a href="#to-do" class="anchor"></a>To do</h2>
<ul>
<li>replicate these searches in R using rentrez and make into a separate chapter?</li>
<li>expand background info?</li>
<li>make graph of increase in amount of sequence info on NCBI</li>
<li>check relationship between NCBI, entrez, GenBank, etc</li>
</ul>
<p><strong>NCBI</strong> is the National Center for Biotechnology Information. The <a href="www.ncbi.nlm.nih.gov/">NCBI Webiste</a> is the entry point to a large number of databases giving access to <strong>biological sequences</strong> (DNA, RNA, protein) and biology-related publications.</p>
<p>When scientists sequence DNA, RNA and proteins they typically publish their data via databases with the NCBI. Each is given a unique identification number known as an <strong>accession number</strong>. For example, each time a unique human genome sequence is produced it is uploaded to the relevant databases, assigned a unique <strong>accession</strong>, and a website created to access it. Sequence are also cross-referenced to related papers, so you can start with a sequence and find out what scientific paper it was used in, or start with a paper and see if any sequences are associated with it.</p>
<p>This chapter provides an introduction to the general search features of the NCBI databases via the interface on the website, including how to locate sequences using accession numbers and other search parameters, such as specific authors or papers. Subsequent chapters will introduce advanced search features, and how to carry out searches using R commands.</p>
<p>One consequence of the explosion of biological sequences used in publications is that the system of databases has become fairly complex. There are databases for different types of data, different types of molecules, data from individual experiments showing <strong>genetic variation</strong> and also the <strong>consensus sequence</strong> for a given molecule. Luckily, if you know the accession number of the sequence you are looking for – which will be our starting point throughout this book – its fairly straight forward. There are numerous other books on bioinformatics and genomics that provide all details if you need to do more complex searches.</p>
<p>In this chapter we’ll typically refer generically to “NCBI data” and “NCBI databases.” This is a simplification, since NCBI is the name of the organization and the databases and search engines often have specific names.</p>
</div>
<div id="biological-sequence-databases" class="section level2">
<h2 class="hasAnchor">
<a href="#biological-sequence-databases" class="anchor"></a>Biological sequence databases</h2>
<p>Almost published biological sequences are available online, as it is a requirement of every scientific journal that any published DNA or RNA or protein sequence must be deposited in a public database. The main resources for storing and distributing sequence data are three large databases:</p>
<ol style="list-style-type: decimal">
<li>USA: <strong><a href="www.ncbi.nlm.nih.gov/">NCBI database</a></strong> (www.ncbi.nlm.nih.gov/)</li>
<li>Europe: <strong>European Molecular Biology Laboratory (EMBL)</strong> database (<a href="https://www.ebi.ac.uk/ena" class="uri">https://www.ebi.ac.uk/ena</a>)</li>
<li>Japan: <strong>DNA Database of Japan (DDBJ)</strong> database (www.ddbj.nig.ac.jp/). These databases collect all publicly available DNA, RNA and protein sequence data and make it available for free. They exchange data nightly, so contain essentially the same data. The redundancy among the databases allows them to serve different communities (e.g. native languages), provide different additioal services such as tutorials, and assure that the world’s scientists have their data backed up in different physical locations – a key component of good data management!</li>
</ol>
</div>
<div id="the-ncbi-sequence-database" class="section level2">
<h2 class="hasAnchor">
<a href="#the-ncbi-sequence-database" class="anchor"></a>The NCBI Sequence Database</h2>
<p>In this chapter we will explore the <strong>NCBI sequence database</strong> using the accession number NC_001477, which is for the complete DEN-1 Dengue virus genome sequence. The accession number is reported in scientific papers originally describing the sequence, and also in subsequent papers that use that particular sequence.</p>
<p>In addition to the sequence itself, for each sequence the NCBI database also stores some additional <strong>annotation</strong> data, such as the name of the species it comes from, references to publications describing that sequence, information on the structure of the proteins coded by the sequence, etc. Some of this annotation data was added by the person who sequenced a sequence and submitted it to the NCBI database, while some may have been added later by a human curator working for NCBI.</p>
</div>
<div id="the-ncbi-sub-databases" class="section level2">
<h2 class="hasAnchor">
<a href="#the-ncbi-sub-databases" class="anchor"></a>The NCBI Sub-Databases</h2>
<p>TODO: Set this up</p>
<p>The NCBI database contains several sub-databases, the most important of which are:</p>
<ul>
<li>
<strong>Nucleotide database</strong>: contains DNA and RNA sequences</li>
<li>
<strong>Protein database</strong>: contains protein sequences</li>
<li>
<strong>EST database</strong>: contains ESTs (expressed sequence tags), which are short sequences derived from mRNAs. (This terminology is likely to be unfamiliar because it is not often used in introductory biology courses. The “Expressed” of EST comes from the fact that mRNA is the result of gene expression.)</li>
<li>
<strong>Genome database</strong>: contains the DNA sequences for entire genomes</li>
<li>
<strong>PubMed</strong>: contains data on scientific publications</li>
</ul>
<p>From the main NCBI website you can initiate a search and it will look for hits accross all the databases. You can narrow your search by selecting a particular database.</p>
</div>
<div id="ncbi-genbank-record-format" class="section level2">
<h2 class="hasAnchor">
<a href="#ncbi-genbank-record-format" class="anchor"></a>NCBI GenBank Record Format</h2>
<p>As mentioned above, for each sequence the NCBI database stores some extra information such as the species that it came from, publications describing the sequence, etc. This information is stored in the GenBank entry (aka GenBank Record) for the sequence. The GenBank entry for a sequence can be viewed by searching the NCBI database for the accession number for that sequence.</p>
<p>To view the GenBank entry for the DEN-1 Dengue virus, follow these steps:</p>
<ol style="list-style-type: decimal">
<li>Go to the <a href="www.ncbi.nlm.nih.gov">NCBI website</a> (www.ncbi.nlm.nih.gov).</li>
<li>Search for the accession number NC_001477.</li>
<li>Since we searched for a particular accession we are only returned a single main result which is titled “NUCLEOTIDE SEQUENCE: Dengue virus 1, complete genome.”</li>
<li>Click on “Dengue virus 1, complete genome” to go to the GenBank entry.</li>
</ol>
<p>TODO: Set this up</p>
<p>The GenBank entry for an accession contains a LOT of information about the sequence, such as papers describing it, features in the sequence, etc. The <strong>DEFINITION</strong> field gives a short description for the sequence. The <strong>ORGANISM</strong> field in the NCBI entry identifies the species that the sequence came from. The <strong>REFERENCE</strong> field contains scientific publications describing the sequence. The <strong>FEATURES</strong> field contains information about the location of features of interest inside the sequence, such as regulatory sequences or genes that lie inside the sequence. The <strong>ORIGIN</strong> field gives the sequence itself.</p>
</div>
<div id="the-fasta-file-format" class="section level2">
<h2 class="hasAnchor">
<a href="#the-fasta-file-format" class="anchor"></a>The FASTA file format</h2>
<p>The <strong>FASTA</strong> file format is a simple file format commonly used to store and share sequence information. When you download sequences from databases such as NCBI you usually want FASTA files.</p>
<p>The first line of a FASTA file starts with the “greater than” character (&gt;) followed by a name and/or description for the sequence. Subsequent lines contain the sequence itself. A short FASTA file might contain just something like this:</p>
<pre><code>#&gt; &gt;mysequence1
#&gt; ACATGAGACAGACAGACCCCCAGAGACAGACCCCTAGACACAGAGAGAG
#&gt; TATGCAGGACAGGGTTTTTGCCCAGGGTGGCAGTATG</code></pre>
<p>A FASTA file can contain the sequence for a single, an entire genome, or more than one sequence. If a FASTA file contains many sequences, then for each sequence it will have a <strong>header line</strong> starting with a greater than character followed by the sequence itself.</p>
<p>This is what a FASTA file with two sequence looks like.</p>
<pre><code>#&gt; &gt;mysequence1
#&gt; ACATGAGACAGACAGACCCCCAGAGACAGACCCCTAGACACAGAGAGAG
#&gt; TATGCAGGACAGGGTTTTTGCCCAGGGTGGCAGTATG
#&gt; 
#&gt; &gt;mysequence2
#&gt; AGGATTGAGGTATGGGTATGTTCCCGATTGAGTAGCCAGTATGAGCCAG
#&gt; AGTTTTTTACAAGTATTTTTCCCAGTAGCCAGAGAGAGAGTCACCCAGT
#&gt; ACAGAGAGC</code></pre>
<p>TODO: Set this up</p>
</div>
<div id="graphics" class="section level2">
<h2 class="hasAnchor">
<a href="#graphics" class="anchor"></a>Graphics</h2>
<p>TODO: Set this up</p>
</div>
<div id="refseq" class="section level2">
<h2 class="hasAnchor">
<a href="#refseq" class="anchor"></a>RefSeq</h2>
<p>When carrying out searches of the NCBI database, it is important to bear in mind that the database may contain <strong>redundant sequences</strong> for the same gene that were sequenced by different laboratories and experimenta. This is bbecause many different labs have sequenced the gene, and submitted their sequences to the NCBI database, and variation exists between individual organisms due to population-level variation due to previous mutations and also potentiall recent spontaneous mutations. There also can be some error in the sequencing process that results in differences between sequences.</p>
<p>There are also many different types of nucleotide sequences and protein sequences in the NCBI database. With respect to nucleotide sequences, some many be entire genomic DNA sequences, some may be mRNAs, and some may be lower quality sequences such as expressed sequence tags (ESTs, which are derived from parts of mRNAs), or DNA sequences of <strong>contigs</strong> from genome projects. That is, you can end up with an entry in the protein database based on sequence derived from a genomic sequence, from sequencing just the gene, and from other routes. Furthermore, some sequences may be <strong>manually curated</strong> by NCBI staff so that the associated entries contain extra information, but the majority of sequences are <strong>uncurated.</strong></p>
<p>Therefore, NCBI databases often contains redundant information for a gene, contains sequences of varying quality, and contains both uncurated and curated data. As a result, NCBI has made a special database called <strong>RefSeq (reference sequence database)</strong>, which is a subset of the NCBI database. The data in RefSeq is manually curated, is high quality sequence data, and is non-redundant; this means that each gene (or <strong>splice-form / isoform</strong> of a gene, in the case of eukaryotes), protein, or genome sequence is only represented once.</p>
<p>The data in RefSeq is curated and is of much higher quality than the rest of the NCBI Sequence Database. However, unfortunately, because of the high level of manual curation required, RefSeq does not cover all species, and is not comprehensive for the species that are covered so far. To speed up searches and simplify the results in to can be very useful to just search RefSeq. However, for detailed and thorough work the full database should probably be searched and the results scrutinized.</p>
<p>You can easily tell that a sequence comes from RefSeq because its accession number starts with particular sequence of letters. That is, accessions of RefSeq sequences corresponding to protein records usually start with <strong>NP_</strong>, and accessions of RefSeq curated complete genome sequences usually start with <strong>NC_</strong> or <strong>NS_</strong>.</p>
</div>
<div id="querying-the-ncbi-database" class="section level2">
<h2 class="hasAnchor">
<a href="#querying-the-ncbi-database" class="anchor"></a>Querying the NCBI Database</h2>
<p>You may need to interrogate the NCBI Database to find particular sequences or a set of sequences matching given criteria, such as:</p>
<ul>
<li>The sequence with accession NC_001477</li>
<li>The sequences published in Nature 460:352-358</li>
<li>All sequences from <em>Chlamydia trachomatis</em>
</li>
<li>Sequences submitted by Caroline Cameron, a syphyllis researcher</li>
<li>Flagellin or fibrinogen sequences</li>
<li>The glutamine synthetase gene from <em>Mycobacteriuma leprae</em>
</li>
<li>Just the upstream control region of the <em>Mycobacterium leprae</em> dnaA gene</li>
<li>The sequence of the <em>Mycobacterium leprae</em> DnaA protein</li>
<li>The genome sequence of syphilis, <em>Treponema pallidum</em> subspp. <em>pallidum</em>
</li>
<li>All human nucleotide sequences associated with malaria</li>
</ul>
<p>There are two main ways that you can query the NCBI database to find these sets of sequences. The first possibility is to carry out searches on the NCBI website. The second possibility is to carry out searches from R using one of several packages that can interface with NCBI. As of October 2019 rentrez seems to be the best package for this..</p>
<p>Below, I will explain how to manually carry out queries on the NCBI database.</p>
</div>
<div id="querying-the-ncbi-database-via-the-ncbi-website-for-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#querying-the-ncbi-database-via-the-ncbi-website-for-reference" class="anchor"></a>Querying the NCBI Database via the NCBI Website (for reference)</h2>
<p><strong>NOTE</strong>: The following section is here for reference; you need to know its possible to refine searches but do not need to know any of these actual tags.</p>
<p>If you are carrying out searches on the NCBI website, to narrow down your searches to specific types of sequences or to specific organisms, you will need to use <strong>“search tags”.</strong></p>
<p>For example, the search tags “[PROP]” and “[ORGN]” let you restrict your search to a specific subset of the NCBI Sequence Database, or to sequences from a particular taxon, respectively. Here is a list of useful search tags, which we will explain how to use below:</p>
<ul>
<li>[AC] NC_001477[AC] With a particular accession number</li>
<li>[ORGN] Fungi[ORGN] From a particular organism or taxon</li>
<li>[PROP] biomol_mRNA[PROP] Of a specific type (eg. mRNA) or from a specific database (eg. RefSeq)</li>
<li>[JOUR] Nature[JOUR] Described in a paper published in a particular journal</li>
<li>[VOL] 531[VOL] Described in a paper published in a particular journal volume</li>
<li>[PAGE] 27[PAGE] Described in a paper with a particular start-page in a journal</li>
<li>[AU] “Smith J”[AU] Described in a paper, or submitted to NCBI, by a particular author</li>
</ul>
<p>To carry out searches of the NCBI database, you first need to go to the NCBI website, and type your search query into the search box at the top. For example, to search for all sequences from Fungi, you would type “Fungi[ORGN]” into the search box on the NCBI website.</p>
<p>You can combine the search tags above by using “AND”, to make more complex searches. For example, to find all mRNA sequences from Fungi, you could type “Fungi[ORGN] AND biomol_mRNA[PROP]” in the search box on the NCBI website.</p>
<p>Likewise, you can also combine search tags by using “OR”, for example, to search for all mRNA sequences from Fungi or Bacteria, you would type “(Fungi[ORGN] OR Bacteria[ORGN]) AND biomol_mRNA[PROP]” in the search box. Note that you need to put brackets around “Fungi[ORGN] OR Bacteria[ORGN]” to specify that the word “OR” refers to these two search tags.</p>
<p>Here are some examples of searches, some of them made by combining search terms using “AND”:</p>
<ul>
<li>NC_001477[AC] - With accession number NC_001477</li>
<li>Nature[JOUR] AND 460[VOL] AND 352[PAGE] - Published in Nature 460:352-358</li>
<li>“Chlamydia trachomatis”[ORGN] - From the bacterium Chlamydia trachomatis</li>
<li>“Berriman M”[AU] - Published in a paper, or submitted to NCBI, by M. Berriman</li>
<li>flagellin OR fibrinogen - Which contain the word “flagellin” or “fibrinogen” in their NCBI record</li>
<li>“Mycobacterium leprae”[ORGN] AND dnaA - Which are from M. leprae, and contain “dnaA” in their NCBI record</li>
<li>“Homo sapiens”[ORGN] AND “colon cancer” - Which are from human, and contain “colon cancer” in their NCBI record</li>
<li>“Homo sapiens”[ORGN] AND malaria - Which are from human, and contain “malaria” in their NCBI record</li>
<li>“Homo sapiens”[ORGN] AND biomol_mrna[PROP] - Which are mRNA sequences from human</li>
<li>“Bacteria”[ORGN] AND srcdb_refseq[PROP] - Which are RefSeq sequences from Bacteria</li>
<li>“colon cancer” AND srcdb_refseq[PROP] - From RefSeq, which contain “colon cancer” in their NCBI record</li>
</ul>
<p>Note that if you are searching for a phrase such as “colon cancer” or “Chlamydia trachomatis”, you need to put the phrase in quotes when typing it into the search box. This is because if you type the phrase in the search box without quotes, the search will be for NCBI records that contain either of the two words “colon” or “cancer” (or either of the two words “Chlamydia” or “trachomatis”), not necessarily both words.</p>
<p>As mentioned above, the NCBI database contains several sub-databases, including the NCBI <strong>Nucleotide database</strong> and the NCBI <strong>Protein database</strong>. If you go to the NCBI website, and type one of the search queries above in the search box at the top of the page, the results page will tell you how many matching NCBI records were found in each of the NCBI sub-databases.</p>
<p>For example, if you search for “Chlamydia trachomatis[ORGN]”, you will get matches to proteins from C. trachomatis in the NCBI Protein database, matches to DNA and RNA sequences from <em>C. trachomatis</em> in the NCBI Nucleotide database, matches to whole genome sequences for C. trachomatis strains in the NCBI Genome database, and so on:</p>
<p>Alternatively, if you know in advance that you want to search a particular sub-database, for example, the NCBI Protein database, when you go to the NCBI website, you can select that sub-database from the drop-down list above the search box, so that you will search that sub-database.</p>
</div>
<div id="example-finding-the-sequences-published-in-nature-460352-358-for-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#example-finding-the-sequences-published-in-nature-460352-358-for-reference" class="anchor"></a>Example: finding the sequences published in Nature 460:352-358 (for reference)</h2>
<p><strong>NOTE</strong>: The following section is here for reference; you need to know its possible to refine searches but do not need to know any of these actual tags.</p>
<p>For example, if you want to find sequences published in Nature 460:352-358, you can use the “[JOUR]”, “[VOL]” and “[PAGE]” search terms. That is, you would go to the NCBI website and type in the search box on the top: “Nature”[JOUR] AND 460[VOL] AND 352[PAGE], where [JOUR] specifies the journal name, [VOL] the volume of the journal the paper is in, and [PAGE] the page number.</p>
<p>This should bring up a results page with “50890” beside the word “Nucleotide”, and “1” beside the word “Genome”, and “25701” beside the word “Protein”, indicating that there were 50890 hits to sequence records in the Nucleotide database, which contains DNA and RNA sequences, and 1 hit to the Genome database, which contains genome sequences, and 25701 hits to the Protein database, which contains protein sequences.</p>
<p>If you click on the word “Nucleotide”, it will bring up a webpage with a list of links to the NCBI sequence records for those 50890 hits. The 50890 hits are all contigs from the schistosome worm <em>Schistosoma mansoni</em>.</p>
<p>Likewise, if you click on the word “Protein”, it will bring up a webpage with a list of links to the NCBI sequence records for the 25701 hits, and you will see that the hits are all predicted proteins for <em>Schistosoma mansoni</em>.</p>
<p>If you click on the word “Genome”, it will bring you to the NCBI record for the <em>Schistosoma mansoni</em> genome sequence, which has NCBI accession NS_00200. Note that the accession starts with “NS_”, which indicates that it is a RefSeq accession.</p>
<p>Therefore, in Nature volume 460, page 352, the <em>Schistosoma mansoni</em> genome sequence was published, along with all the DNA sequence contigs that were sequenced for the genome project, and all the predicted proteins for the gene predictions made in the genome sequence. You can view the original paper on the Nature website at <a href="http://www.nature.com/nature/journal/v460/n7253/abs/nature08160.html" class="uri">http://www.nature.com/nature/journal/v460/n7253/abs/nature08160.html</a>.</p>
<p>Note: <em>Schistmosoma mansoni</em> is a parasitic worm that is responsible for causing <strong>schistosomiasis</strong>, which is classified by the WHO as a <strong>neglected tropical disease</strong>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
