<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Higgs and Atwood student version with updates • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Higgs and Atwood student version with updates">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Higgs and Atwood student version with updates</h1>
                        <h4 class="author">Nathan Brouwer</h4>
            
            <h4 class="date">2021-05-20</h4>
      
      
      <div class="hidden name"><code>Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.Rmd</code></div>

    </div>

    
    
<p>Add your name to “author” above</p>
<p>NOTE 1: When you submit this please remove all “TODO” prompts, NOTES, instructions. etc by me. THe final product should look like a blog post, NOT a homework assignment</p>
<p>NOTE 2: Each code chunk should have a line that indicates what is being done, like an instructional blog post.</p>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>WRITING TODO: Write a 4-5 sentence introduce describing what is done by this script and why we care about it biologically.</p>
</div>
<div id="preliminaries" class="section level2">
<h2 class="hasAnchor">
<a href="#preliminaries" class="anchor"></a>Preliminaries</h2>
<div id="packages" class="section level3">
<h3 class="hasAnchor">
<a href="#packages" class="anchor"></a>Packages</h3>
<ol style="list-style-type: decimal">
<li>CODE-TODO: load ALL packages</li>
<li>COMMENT-TODO: add a comment to indicate what each packages does</li>
<li>WRITING-TODO: describe in 1-2 sentences the relationship between ggpubr R and ggplot2.</li>
</ol>
</div>
<div id="build-the-dataframe" class="section level3">
<h3 class="hasAnchor">
<a href="#build-the-dataframe" class="anchor"></a>Build the dataframe</h3>
<ol style="list-style-type: decimal">
<li>CODE-TODO: build the amino acid dataframe I provided.</li>
<li>WRITING-TODO: write 2-3 sentences about what the data represent.</li>
<li>TODO: read the comments I wrote in the code; you will want to refer to these later</li>
</ol>
<p>These data are derived mostly from Higgs (2009) Table 1. They are similar to Higgs and Attwood 2005 but have extra columns. Nathan Brouwer added additional categorical varibles based on online information.</p>
<p>Make the vectors.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 1 letter code</span>
<span class="va">aa</span>        <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'A'</span>,<span class="st">'C'</span>,<span class="st">'D'</span>,<span class="st">'E'</span>,<span class="st">'F'</span>,<span class="st">'G'</span>,<span class="st">'H'</span>,<span class="st">'I'</span>,<span class="st">'K'</span>,<span class="st">'L'</span>,<span class="st">'M'</span>,<span class="st">'N'</span>,
              <span class="st">'P'</span>,<span class="st">'Q'</span>,<span class="st">'R'</span>,<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'V'</span>,<span class="st">'W'</span>,<span class="st">'Y'</span><span class="op">)</span>


<span class="co">## molecular weight in dalton</span>
<span class="va">MW.da</span>     <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">89</span>,<span class="fl">121</span>,<span class="fl">133</span>,<span class="fl">146</span>,<span class="fl">165</span>,<span class="fl">75</span>,<span class="fl">155</span>,<span class="fl">131</span>,<span class="fl">146</span>,<span class="fl">131</span>,<span class="fl">149</span>,<span class="fl">132</span>,<span class="fl">115</span>,<span class="fl">147</span>,<span class="fl">174</span>,<span class="fl">105</span>,<span class="fl">119</span>,<span class="fl">117</span>,<span class="fl">204</span>,<span class="fl">181</span><span class="op">)</span>

<span class="co">## vol from van der Waals radii </span>
<span class="va">vol</span>    <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">67</span>,<span class="fl">86</span>,<span class="fl">91</span>,<span class="fl">109</span>,<span class="fl">135</span>,<span class="fl">48</span>,<span class="fl">118</span>,<span class="fl">124</span>,<span class="fl">135</span>,<span class="fl">124</span>,<span class="fl">124</span>,<span class="fl">96</span>,<span class="fl">90</span>,
              <span class="fl">114</span>,<span class="fl">148</span>,<span class="fl">73</span>,<span class="fl">93</span>,<span class="fl">105</span>,<span class="fl">163</span>,<span class="fl">141</span><span class="op">)</span>

<span class="co">##  bulk – a measure of the shape of the side chain</span>
<span class="va">bulk</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">11.5</span>,<span class="fl">13.46</span>,<span class="fl">11.68</span>,<span class="fl">13.57</span>,<span class="fl">19.8</span>,<span class="fl">3.4</span>,<span class="fl">13.69</span>,<span class="fl">21.4</span>,
              <span class="fl">15.71</span>,<span class="fl">21.4</span>,<span class="fl">16.25</span>,<span class="fl">12.28</span>,<span class="fl">17.43</span>,
              <span class="fl">14.45</span>,<span class="fl">14.28</span>,<span class="fl">9.47</span>,<span class="fl">15.77</span>,<span class="fl">21.57</span>,<span class="fl">21.67</span>,<span class="fl">18.03</span><span class="op">)</span>

<span class="co">## pol – a measure of the electric field strength around the molecule</span>
<span class="va">pol</span>  <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.48</span>,<span class="fl">49.7</span>,<span class="fl">49.9</span>,<span class="fl">0.35</span>,<span class="fl">0</span>,<span class="fl">51.6</span>,<span class="fl">0.13</span>,<span class="fl">49.5</span>,<span class="fl">0.13</span>,
              <span class="fl">1.43</span>,<span class="fl">3.38</span>,<span class="fl">1.58</span>,<span class="fl">3.53</span>,<span class="fl">52</span>,<span class="fl">1.67</span>,<span class="fl">1.66</span>,<span class="fl">0.13</span>,<span class="fl">2.1</span>,<span class="fl">1.61</span><span class="op">)</span>

<span class="co">## isoelec point</span>
<span class="va">isoelec</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">5.07</span>,<span class="fl">2.77</span>,<span class="fl">3.22</span>,<span class="fl">5.48</span>,<span class="fl">5.97</span>,<span class="fl">7.59</span>,<span class="fl">6.02</span>,<span class="fl">9.74</span>,<span class="fl">5.98</span>,
                   <span class="fl">5.74</span>,<span class="fl">5.41</span>,<span class="fl">6.3</span>,<span class="fl">5.65</span>,<span class="fl">10.76</span>,<span class="fl">5.68</span>,<span class="fl">6.16</span>,<span class="fl">5.96</span>,<span class="fl">5.89</span>,<span class="fl">5.66</span><span class="op">)</span>

<span class="co">## 1st Hydrophobicity scale</span>
<span class="va">H2Opho.34</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1.8</span>,<span class="fl">2.5</span>,<span class="op">-</span><span class="fl">3.5</span>,<span class="op">-</span><span class="fl">3.5</span>,<span class="fl">2.8</span>,<span class="op">-</span><span class="fl">0.4</span>,<span class="op">-</span><span class="fl">3.2</span>,<span class="fl">4.5</span>,<span class="op">-</span><span class="fl">3.9</span>,<span class="fl">3.8</span>,<span class="fl">1.9</span>,
                  <span class="op">-</span><span class="fl">3.5</span>,<span class="op">-</span><span class="fl">1.6</span>,<span class="op">-</span><span class="fl">3.5</span>,<span class="op">-</span><span class="fl">4.5</span>,<span class="op">-</span><span class="fl">0.8</span>,<span class="op">-</span><span class="fl">0.7</span>,<span class="fl">4.2</span>,<span class="op">-</span><span class="fl">0.9</span>,<span class="op">-</span><span class="fl">1.3</span><span class="op">)</span>

<span class="co">## 2nd Hydrophobicity scale</span>
<span class="va">H2Opho.35</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1.6</span>,<span class="fl">2</span>,<span class="op">-</span><span class="fl">9.2</span>,<span class="op">-</span><span class="fl">8.2</span>,<span class="fl">3.7</span>,<span class="fl">1</span>,<span class="op">-</span><span class="fl">3</span>,<span class="fl">3.1</span>,<span class="op">-</span><span class="fl">8.8</span>,<span class="fl">2.8</span>,<span class="fl">3.4</span>,<span class="op">-</span><span class="fl">4.8</span>,
                  <span class="op">-</span><span class="fl">0.2</span>,<span class="op">-</span><span class="fl">4.1</span>,<span class="op">-</span><span class="fl">12.3</span>,<span class="fl">0.6</span>,<span class="fl">1.2</span>,<span class="fl">2.6</span>,<span class="fl">1.9</span>,<span class="op">-</span><span class="fl">0.7</span><span class="op">)</span>

<span class="co">## Surface area accessible to water in an unfolded peptide</span>
<span class="va">saaH2O</span>       <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">113</span>,<span class="fl">140</span>,<span class="fl">151</span>,<span class="fl">183</span>,<span class="fl">218</span>,<span class="fl">85</span>,<span class="fl">194</span>,<span class="fl">182</span>,<span class="fl">211</span>,<span class="fl">180</span>,<span class="fl">204</span>,<span class="fl">158</span>,
                 <span class="fl">143</span>,<span class="fl">189</span>,<span class="fl">241</span>,<span class="fl">122</span>,<span class="fl">146</span>,<span class="fl">160</span>,<span class="fl">259</span>,<span class="fl">229</span><span class="op">)</span>


<span class="co">## Fraction of accessible area lost when a protein folds</span>
<span class="va">faal.fold</span>    <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.74</span>,<span class="fl">0.91</span>,<span class="fl">0.62</span>,<span class="fl">0.62</span>,<span class="fl">0.88</span>,<span class="fl">0.72</span>,<span class="fl">0.78</span>,<span class="fl">0.88</span>,<span class="fl">0.52</span>,
                 <span class="fl">0.85</span>,<span class="fl">0.85</span>,<span class="fl">0.63</span>,<span class="fl">0.64</span>,<span class="fl">0.62</span>,<span class="fl">0.64</span>,<span class="fl">0.66</span>,<span class="fl">0.7</span>,<span class="fl">0.86</span>,<span class="fl">0.85</span>,<span class="fl">0.76</span><span class="op">)</span>



<span class="co"># Polar requirement </span>
<span class="va">polar.req</span>    <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">7</span>,<span class="fl">4.8</span>,<span class="fl">13</span>,<span class="fl">12.5</span>,<span class="fl">5</span>,<span class="fl">7.9</span>,<span class="fl">8.4</span>,<span class="fl">4.9</span>,<span class="fl">10.1</span>,<span class="fl">4.9</span>,<span class="fl">5.3</span>,<span class="fl">10</span>,
                 <span class="fl">6.6</span>,<span class="fl">8.6</span>,<span class="fl">9.1</span>,<span class="fl">7.5</span>,<span class="fl">6.6</span>,<span class="fl">5.6</span>,<span class="fl">5.2</span>,<span class="fl">5.4</span><span class="op">)</span>

<span class="co">## relative frequency of occurance</span>
<span class="co">## "The frequencies column shows the mean </span>
<span class="co">### percentage of each amino acid in the protein sequences </span>
<span class="co">### of modern organisms"</span>
<span class="va">freq</span>        <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">7.8</span>,<span class="fl">1.1</span>,<span class="fl">5.19</span>,<span class="fl">6.72</span>,<span class="fl">4.39</span>,<span class="fl">6.77</span>,<span class="fl">2.03</span>,<span class="fl">6.95</span>,<span class="fl">6.32</span>,
                <span class="fl">10.15</span>,<span class="fl">2.28</span>,<span class="fl">4.37</span>,<span class="fl">4.26</span>,<span class="fl">3.45</span>,<span class="fl">5.23</span>,<span class="fl">6.46</span>,<span class="fl">5.12</span>,<span class="fl">7.01</span>,<span class="fl">1.09</span>,<span class="fl">3.3</span><span class="op">)</span>

<span class="co">##</span>
<span class="co"># charges</span>
<span class="co">## un = Un-charged</span>
<span class="co">## neg = negative</span>
<span class="co">## pos = positive</span>
<span class="va">charge</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'neg'</span>,<span class="st">'neg'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'pos'</span>,<span class="st">'un'</span>,<span class="st">'pos'</span>,
          <span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'pos'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'un'</span>,<span class="st">'un'</span><span class="op">)</span>

<span class="co">## hydropathy</span>
<span class="va">hydropathy</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'hydrophobic'</span>,<span class="st">'hydrophobic'</span>,<span class="st">'hydrophilic'</span>,<span class="st">'
              hydrophilic'</span>,<span class="st">'hydrophobic'</span>,<span class="st">'neutral'</span>,<span class="st">'neutral'</span>,
              <span class="st">'hydrophobic'</span>,<span class="st">'hydrophilic'</span>,<span class="st">'hydrophobic'</span>,<span class="st">'hydrophobic'</span>,
              <span class="st">'hydrophilic'</span>,<span class="st">'neutral'</span>,
              <span class="st">'hydrophilic'</span>,<span class="st">'hydrophilic'</span>,<span class="st">'neutral'</span>,<span class="st">'neutral'</span>,
              <span class="st">'hydrophobic'</span>,<span class="st">'hydrophobic'</span>,<span class="st">'neutral'</span><span class="op">)</span>

<span class="co">## vol</span>
<span class="va">vol.cat</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'verysmall'</span>,<span class="st">'small'</span>,<span class="st">'small'</span>,<span class="st">'medium'</span>,
              <span class="st">'verylarge'</span>,<span class="st">'verysmall'</span>,<span class="st">'medium'</span>,<span class="st">'large'</span>,<span class="st">'large'</span>,
              <span class="st">'large'</span>,<span class="st">'large'</span>,<span class="st">'small'</span>,<span class="st">'small'</span>,<span class="st">'medium'</span>,<span class="st">'large'</span>,
              <span class="st">'verysmall'</span>,
              <span class="st">'small'</span>,<span class="st">'medium'</span>,<span class="st">'verylarge'</span>,<span class="st">'verylarge'</span><span class="op">)</span>

<span class="co">## pol</span>
<span class="va">pol.cat</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'nonpolar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'polar'</span>,<span class="st">'polar'</span>,
                <span class="st">'nonpolar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'polar'</span>,<span class="st">'nonpolar'</span>,
                <span class="st">'polar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'polar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'polar'</span>,
                <span class="st">'polar'</span>,<span class="st">'polar'</span>,<span class="st">'polar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'nonpolar'</span>,<span class="st">'polar'</span><span class="op">)</span>

<span class="co">## chemical</span>
<span class="va">chemical</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'aliphatic'</span>,<span class="st">'sulfur'</span>,<span class="st">'acidic'</span>,<span class="st">'acidic'</span>,<span class="st">'aromatic'</span>,
            <span class="st">'aliphatic'</span>,<span class="st">'basic'</span>,<span class="st">'aliphatic'</span>,<span class="st">'basic'</span>,<span class="st">'aliphatic'</span>,<span class="st">'sulfur'</span>,
            <span class="st">'amide'</span>,<span class="st">'aliphatic'</span>,<span class="st">'amide'</span>,<span class="st">'basic'</span>,<span class="st">'hydroxyl'</span>,<span class="st">'hydroxyl'</span>,
            <span class="st">'aliphatic'</span>,<span class="st">'aromatic'</span>,<span class="st">'aromatic'</span><span class="op">)</span></code></pre></div>
<p>Build the dataframes.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Build dataframe</span>
<span class="va">aa_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">aa</span>, 
                     <span class="va">MW.da</span>,<span class="va">vol</span>,
                     <span class="va">bulk</span>, <span class="va">pol</span>, 
                     <span class="va">isoelec</span>,<span class="va">H2Opho.34</span>, <span class="va">H2Opho.35</span>,
                     <span class="va">saaH2O</span>, <span class="va">faal.fold</span>, <span class="va">polar.req</span>,
                     <span class="va">freq</span>, <span class="va">charge</span>, <span class="va">hydropathy</span>,
                     <span class="va">vol.cat</span>, <span class="va">pol.cat</span>, <span class="va">chemical</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="raw-data-exploration" class="section level2">
<h2 class="hasAnchor">
<a href="#raw-data-exploration" class="anchor"></a>Raw data exploration</h2>
<div id="xxxxxxxx" class="section level3">
<h3 class="hasAnchor">
<a href="#xxxxxxxx" class="anchor"></a>XXXXXXXX</h3>
<p>TODO: Examine the code below. What is it called? Replace the XXXXXXXX above with the name of this thing</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cor_</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">aa_dat</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">13</span><span class="op">:</span><span class="fl">17</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,<span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/diag.html">diag</a></span><span class="op">(</span><span class="va">cor_</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">cor_</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html">upper.tri</a></span><span class="op">(</span><span class="va">cor_</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">cor_</span></code></pre></div>
<p>WRITING-TODO: Describe what the code below is doing and summarzie the next 2 TODOs in 2-3 sentences. TODO: Which variables have the most postive correlation? TODO: What have most negative correlation?</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">cor_</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">cor_</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, arr.ind <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">cor_</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cor_</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">cor_</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cor_</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, arr.ind <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-0-scatterplot-matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-0-scatterplot-matrix" class="anchor"></a>Plot 0: Scatterplot matrix</h3>
<p><strong>Code TODO</strong>: Make a scatterplot matrix of all of the data. Included the following variables:</p>
<ol style="list-style-type: decimal">
<li>MW.da</li>
<li>vol</li>
<li>pol</li>
<li>H2Opho.34</li>
<li>polar.req</li>
</ol>
<p>PLUS 2 others (6 total)</p>
<p>NOTE: Depending on how you do this you may have to make sure the first column doesn’t go into the plotting command</p>
<p><strong>TODO:</strong> Write 1-2 sentences describing what this below code does in your scatter plot matrix below</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">panel.cor</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">digits</span> <span class="op">=</span> <span class="fl">2</span>, <span class="va">prefix</span> <span class="op">=</span> <span class="st">""</span>, <span class="va">cex.cor</span>,<span class="va">...</span><span class="op">)</span>
<span class="op">{</span>
    <span class="va">usr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span><span class="st">"usr"</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/on.exit.html">on.exit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span><span class="va">usr</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>usr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
    <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span>
    <span class="va">txt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">r</span>, <span class="fl">0.123456789</span><span class="op">)</span>, digits <span class="op">=</span> <span class="va">digits</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
    <span class="va">txt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prefix</span>, <span class="va">txt</span><span class="op">)</span>
    <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/missing.html">missing</a></span><span class="op">(</span><span class="va">cex.cor</span><span class="op">)</span><span class="op">)</span> <span class="va">cex.cor</span> <span class="op">&lt;-</span> <span class="fl">0.8</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/graphics/strwidth.html">strwidth</a></span><span class="op">(</span><span class="va">txt</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="va">txt</span>, cex <span class="op">=</span> <span class="va">cex.cor</span> <span class="op">*</span> <span class="va">r</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>TOD: WRITE 2-3 sentences describing the relationships between the 6 variables you plotted; which have strong correlations which ones weak, which ones have non-linear relationships.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">aa_dat</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MW.da"</span>,<span class="st">"vol"</span>,<span class="st">"pol"</span>,
               <span class="st">"H2Opho.34"</span>,<span class="st">"polar.req"</span>,
               <span class="st">"saaH2O"</span>,<span class="st">"faal.fold"</span><span class="op">)</span><span class="op">]</span>,
     panel <span class="op">=</span> <span class="va">panel.smooth</span>,
      upper.panel <span class="op">=</span><span class="va">panel.cor</span> ,
     main <span class="op">=</span> <span class="st">"Example"</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-1-replication-higgs-and-attwood-figure-2-8" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-1-replication-higgs-and-attwood-figure-2-8" class="anchor"></a>Plot 1: Replication Higgs and Attwood Figure 2.8</h3>
<ol style="list-style-type: decimal">
<li>CODE-TODO: make a plot that matches Figure 2.8 on page 26 of the Higgs and Atwood chapter 2. Make sure you have plotted the correct axes</li>
<li>CODE-TODO: label the x and y axes.</li>
<li>COMMENT-TODO: indicate what is doig the main plotting and what is adding the text lables.</li>
<li>WRITING-TODO: Write 1-2 sentences describing what the plot represents</li>
</ol>
</div>
<div id="figure-2-ha-figure-2-8-with-ggpubr" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-2-ha-figure-2-8-with-ggpubr" class="anchor"></a>Figure 2: HA Figure 2.8 with ggpubr</h3>
<ol style="list-style-type: decimal">
<li>TODO: Make a ggpubr version of HA Figure 2.8. Additionally, use both COLOR and SHAPE to represent 2 additional NUMERIC (NOT categorical) variables of your choice.</li>
<li>TODO: lable the axes with the full name of the variables</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"polar.req"</span>,
          x <span class="op">=</span> <span class="st">"freq"</span>,
          size <span class="op">=</span> <span class="st">"polar.req"</span>,
          color <span class="op">=</span> <span class="st">"polar.req"</span>,
          data <span class="op">=</span> <span class="va">aa_dat</span>,
          xlab <span class="op">=</span> <span class="st">""</span>,
          ylab <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
</div>
<div id="figure-3-highly-correlated-data" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-3-highly-correlated-data" class="anchor"></a>Figure 3: Highly correlated data</h3>
<p>CODE:</p>
<ol style="list-style-type: decimal">
<li>TODO: Identify 2 variables that are highly positively correlated and plot them in GGPUBR</li>
<li>TODO: add a regression line</li>
<li>TODO: add a data ellipse</li>
<li>TODO: add a correlation coefficient</li>
<li>TODO: lable the axes with the full name of the variables</li>
</ol>
<p>COMMENTS:</p>
<ol style="list-style-type: decimal">
<li>TODO: add a comment indicating which line of the ggpubr is adding the line</li>
<li>TODO: add a comment indicating which line of the ggpubr is adding the ellipse TODO: add a comment indicating which line of the ggpubr is adding the correlation coefficient</li>
</ol>
<p>WRITING</p>
<p>1.TODO: describe the general mathematical form of the regression line in 1-2 sentences 1.TODO: describe the general meaning of the data ellipse in 1-2 sentences</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"polar.req"</span>,
          x <span class="op">=</span> <span class="st">"freq"</span>,
          size <span class="op">=</span> <span class="st">"polar.req"</span>,
           add <span class="op">=</span> <span class="st">"reg.line"</span>,  <span class="co"># line of best fit</span>
           ellipse <span class="op">=</span> <span class="cn">TRUE</span>,   <span class="co"># data ellipse</span>
          cor.coef <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># correlation coef</span>
         <span class="co"># color = "freq",</span>
          data <span class="op">=</span> <span class="va">aa_dat</span>,
         xlab <span class="op">=</span> <span class="st">""</span>,
         ylab <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
</div>
<div id="figure-4-non-linear-relationship" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-4-non-linear-relationship" class="anchor"></a>Figure 4: Non-linear relationship</h3>
<p>CODE:</p>
<ol style="list-style-type: decimal">
<li>TODO: Identify 2 variables that have a non-linear relationship between them (scatter plot has a curved shape).</li>
<li>TODO: Use ggpubr to plot a scatterplot with a <strong>loess smoother</strong>. Use google or the ggubr help files to figure out how to do this</li>
<li>TODO: also vary the size and color of the data by 2 other NUMERIC variables</li>
<li>TODO: label the axes with the full name of the variables</li>
</ol>
</div>
<div id="figure-5-non-linear-relationship-on-log-scale" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-5-non-linear-relationship-on-log-scale" class="anchor"></a>Figure 5: Non-linear relationship on LOG scale</h3>
<p>CODE:</p>
<ol style="list-style-type: decimal">
<li>TODO: Log-transform both of your the variables you used in Figure 4 make the plot again</li>
<li>TODO: a regression line instead of a loess smoother</li>
<li>TODO: also vary the size and color of the data by 2 other NUMERIC variables</li>
</ol>
<p>WRITING:</p>
<ol style="list-style-type: decimal">
<li>TODO: describe what the log transformation does in 1 to 2 sentences.</li>
</ol>
</div>
<div id="figure-6-3d-scatterplot" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-6-3d-scatterplot" class="anchor"></a>Figure 6: 3D Scatterplot</h3>
<ol style="list-style-type: decimal">
<li>CODE TODO: Make a 3D scatterplot of 3 variables that have relatively strong correlations.</li>
<li>CODE TODO: use highlight.3d = TRUE to colored code</li>
<li>CODE CODE TODO: Use type=“h”</li>
<li>WRITING TODO: Summarize the relationships and correlations between the variables you chose in 2-4 sentences.</li>
</ol>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">scatterplot3d</span><span class="op">(</span>x <span class="op">=</span> <span class="va">aa_dat</span><span class="op">$</span><span class="va">MW.da</span>,
              y <span class="op">=</span> <span class="va">aa_dat</span><span class="op">$</span><span class="va">vol</span>,
              z <span class="op">=</span> <span class="va">aa_dat</span><span class="op">$</span><span class="va">bulk</span>,
              highlight.3d <span class="op">=</span> <span class="cn">TRUE</span>,
              type<span class="op">=</span><span class="st">"h"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="principal-component-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal component analysis</h2>
<p>TODO: Write 3-4 sentences describing the basic use and interpretation of PCA as we have used it in this class. Refer to Higgs and Attwood, and provide a useful quote that explains PCA. Cite the quote as (Higgs and Attwood 2005, pg XX).</p>
<p>TODO: Write 3-4 sentences describing the 2 basic ways we made PCA plots in this class.</p>
<div id="data-prep" class="section level3">
<h3 class="hasAnchor">
<a href="#data-prep" class="anchor"></a>Data prep</h3>
<p>TODO: To make life easier, make a new version of your dataset which drops any categorical variables. Call it aa_dat2 TODO: Additionally: compare Table 2.2 and the data provided. I give you ~11 numeric variables, while Table 2.2 has only 8. Read the information carefully and remove the extra variables</p>
<p>TODO: Write 1-2 sentences describing the approach you used to make this new dataframe.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aa_dat2</span> <span class="op">&lt;-</span> <span class="va">aa_dat</span><span class="op">[</span>,<span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,
                      <span class="fl">2</span>,<span class="fl">11</span>,<span class="fl">12</span>,
                      <span class="fl">13</span>,<span class="fl">14</span>,<span class="fl">15</span>,<span class="fl">16</span>,<span class="fl">17</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="figure-7-principal-components-analysis---base-r" class="section level3">
<h3 class="hasAnchor">
<a href="#figure-7-principal-components-analysis---base-r" class="anchor"></a>Figure 7: Principal components analysis - base R</h3>
<p>Run a PCA and create a PCA plot using base R command.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pca.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html">prcomp</a></span><span class="op">(</span><span class="va">aa_dat2</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Plot the output</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/biplot.html">biplot</a></span><span class="op">(</span><span class="va">pca.out</span><span class="op">)</span></code></pre></div>
<p>We can carry out the numeric analysis using the rda() function from the vegan package.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rda.out</span> <span class="op">&lt;-</span> <span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/cca.html">rda</a></span><span class="op">(</span><span class="va">aa_dat2</span>, 
                      scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>We then extract what are known as the PCA score - don’t worry about what this means right now, we’ll dig into it in a moment. Basically, we’re going from 15 columns of data to just 2, which we pull out of rda.out with the scores() function.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rda_scores</span> <span class="op">&lt;-</span> <span class="fu">scores</span><span class="op">(</span><span class="va">rda.out</span><span class="op">)</span></code></pre></div>
</div>
<div id="principal-components-analysis-pca" class="section level3">
<h3 class="hasAnchor">
<a href="#principal-components-analysis-pca" class="anchor"></a>Principal Components Analysis (PCA)</h3>
<p>We can reduce the <strong>dimensionality</strong> of complicated data sets in order to make them easier to visualize. In the case of our amino acid dataset we can go from 15 different variables to just to.</p>
<p>Principal Components Analysis (PCA) is common form of <strong>dimension reduction</strong>. We’ll first produce the standard output of a PCA analysis – a <strong>biplot</strong> – before digging into the theory behind what is happening.</p>
<p>We can carry out the numeric analysis using the rda() function from the vegan package.</p>
<p><strong>TASK</strong>: Call the rda() function on the aa_dat2 object and save it to the rda.out object provided below.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># COMPLETE THIS CODE:</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>rda.out <span class="ot">&lt;-</span> <span class="do">## YOUR CODE GOES HERE ##</span></span></code></pre></div>
<p>We then extract what are known as the PCA score - don’t worry about what this means right now, we’ll dig into it in a moment. Basically, we’re going from 15 columns of data to just 2, which we pull out of rda.out with the scores() function.</p>
<p><strong>TASK</strong>: Call the scores() function on the rda.out object produced above; save the output to the rda_score object provided</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># COMPLETE THIS CODE:</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>rda_scores <span class="ot">&lt;-</span> <span class="co"># COMPLETE THIS CODE:</span></span></code></pre></div>
<p>Now we’ll make a <strong>biplot</strong>. This is similar in some ways to a scatter plot. We’ll also add some additional annotation to the plot to help us see meaninful grouping of our amino acids. First, we’ll label each amino acids with its 1-letter code. Then we’ll delineate all amino acids within pre-specified groups to see if these groups map to the entire set of columns.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Build plot</span>
<span class="fu"><a href="https://rdrr.io/r/stats/biplot.html">biplot</a></span><span class="op">(</span><span class="va">rda.out</span>, display <span class="op">=</span> <span class="st">"sites"</span>, col <span class="op">=</span> <span class="fl">0</span>,
         main <span class="op">=</span> <span class="st">"Major clusters of amino acids based on chemistry"</span>,
       xlab <span class="op">=</span> <span class="st">"PCA Axis 1: Primary axis of variation"</span>,
       ylab <span class="op">=</span> <span class="st">"PCA Axis 1: Secondary axis of variation"</span><span class="op">)</span>

<span class="co"># add point</span>
<span class="fu">orditorp</span><span class="op">(</span><span class="va">rda.out</span>, display <span class="op">=</span> <span class="st">"sites"</span>, labels <span class="op">=</span> <span class="va">aa_dat</span><span class="op">$</span><span class="va">aa</span>, cex <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span>

<span class="co"># add groupings</span>
<span class="fu">ordihull</span><span class="op">(</span><span class="va">rda.out</span>,
         group <span class="op">=</span> <span class="va">aa_dat</span><span class="op">$</span><span class="va">hydropathy</span>,
         col <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">7</span>,
         lty <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">7</span>,
         lwd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="cluster-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-analysis" class="anchor"></a>Cluster analysis</h2>
<p>TODO: Write 2-3 sentences describing the purpose of cluster analysis. TODO: Write 3-5 sentences describing how UPGMA works.</p>
<div id="xxxxxxxxxxxxxxx-cluster-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#xxxxxxxxxxxxxxx-cluster-analysis" class="anchor"></a>XXXXXXXXXXXXXXX cluster analysis</h3>
<p>TODO: What type of cluster analysis is UPGMA? Change the XXXXXXX above to this. NOTE: use the aa_dat2 data we used above for all of this.</p>
<p>Add row names. Don’t worry about what this code is doing</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">aa_dat2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">aa_dat</span><span class="op">$</span><span class="va">aa</span>,
                            <span class="co">#aa_dat$hydropathy,</span>
                            sep <span class="op">=</span> <span class="st">"."</span><span class="op">)</span></code></pre></div>
<p>CODE-TODO: Create a distance matrix using the dist() function. Set the distance to be euclidean using method = “euclidean”. Call the output dist_euc WRITING-TODO: Write 2-3 sentences describing what a distance matrix is. WRITING-TODO: write 2-3 sentences describing what Euclidean distance is.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dist_euc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span><span class="op">(</span><span class="va">aa_dat2</span>, 
                 method <span class="op">=</span> <span class="st">"euclidean"</span><span class="op">)</span></code></pre></div>
<p>CODE-TODO: Use the hclust() function to do cluster analysis. Use the UPGMA algorithm (see the help file for the proper setting)</p>
<p>COMMENT-TODO: Add a comment that that indicates what part of the code is setting it to use UPGMA.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clust_euc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span><span class="op">(</span><span class="va">dist_euc</span>,
                    method <span class="op">=</span> <span class="st">"average"</span><span class="op">)</span></code></pre></div>
<p>CODE-TODO: Plot the dendrogram (Optional - convert to a dengrogram as I did) WRITING-TODO: Write 2-5 sentences describing how your cluster diagram compares to the cluster diagram produced by Higgs and Atwood (2005) in Plate 2.2. (This is a cluster diagram that also includes a heat map)</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dendro_euc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html">as.dendrogram</a></span><span class="op">(</span><span class="va">clust_euc</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">dendro_euc</span>,horiz <span class="op">=</span> <span class="cn">T</span>,
     nodePar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>pch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="cn">NA</span><span class="op">)</span>, 
                  cex <span class="op">=</span> <span class="fl">0.5</span>, 
                   lab.cex <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
