<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>UPGMA in R • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="UPGMA in R">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="lll-UPGMA_in_R_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>UPGMA in R</h1>
            
      
      
      <div class="hidden name"><code>lll-UPGMA_in_R.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>UPGMA is a method for constructing phylogenetic trees using <strong>genetic distances</strong>. It is an old, outmoded method that is rarely ever used any more building final versions of trees. Its relative simplicity makes it a useful starting point for thinking about how to construct phylogenetic trees and code up algorithms to build them.</p>
<p>A good, no-math overview of the general ideas of UPGMA is at <a href="https://medium.com/@sharma.ravit/upgma-method-designing-a-phylogenetic-tree-9a708de18419" class="uri">https://medium.com/@sharma.ravit/upgma-method-designing-a-phylogenetic-tree-9a708de18419</a></p>
<p>Wikipedia provides a good overview with math that isn’t too daunting (though it could be improved) <a href="https://en.wikipedia.org/wiki/UPGMA" class="uri">https://en.wikipedia.org/wiki/UPGMA</a></p>
<p>An excellent worked example is at <a href="http://www.slimsuite.unsw.edu.au/teaching/upgma/" class="uri">http://www.slimsuite.unsw.edu.au/teaching/upgma/</a> This example breaks down the math into basic steps, but there’s a lot of taxa so its still complicated.</p>
<p>Some useful but terse notes are here: <a href="https://www.sequentix.de/gelquest/help/upgma_method.htm" class="uri">https://www.sequentix.de/gelquest/help/upgma_method.htm</a></p>
<p>If you want to understand what the <strong>ultrametricity</strong> assumption is, check this out (I don’t get hung up on this) <a href="https://www.sequentix.de/gelquest/help/explanation_of_the_term_ultrametric.htm" class="uri">https://www.sequentix.de/gelquest/help/explanation_of_the_term_ultrametric.htm</a></p>
<p>A worked example showing the limits of UPGMA is at <a href="https://www.icp.ucl.ac.be/~opperd/private/upgma.html" class="uri">https://www.icp.ucl.ac.be/~opperd/private/upgma.html</a></p>
<p>Another, dense-looking example is here <a href="http://www.nmsr.org/upgma.htm" class="uri">http://www.nmsr.org/upgma.htm</a></p>
<p>Anothere worked example is <a href="https://www.mun.ca/biology/scarr/2900_UPGMA.htm" class="uri">https://www.mun.ca/biology/scarr/2900_UPGMA.htm</a></p>
<p>Explantion of the convoluted meaning of “Un-weighted” is here <a href="https://www.mun.ca/biology/scarr/UPGMA_vs_WPGMA.html" class="uri">https://www.mun.ca/biology/scarr/UPGMA_vs_WPGMA.html</a></p>
</div>
<div id="preliminaries" class="section level2">
<h2 class="hasAnchor">
<a href="#preliminaries" class="anchor"></a>Preliminaries</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#install.packages("phangorn")</span>
<span class="co">#install.packages("ape")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn">phangorn</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ape</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="sequence-data" class="section level2">
<h2 class="hasAnchor">
<a href="#sequence-data" class="anchor"></a>Sequence data</h2>
<p>Let’s consider three hypothetical DNA sequences.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">human</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>,<span class="st">"t"</span>,<span class="st">"c"</span>,<span class="st">"g"</span>,<span class="st">"a"</span>,<span class="st">"t"</span>,<span class="st">"c"</span>,<span class="st">"g"</span><span class="op">)</span>
<span class="va">chimp</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>,<span class="st">"t"</span>,<span class="st">"c"</span>,<span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"t"</span>,<span class="st">"c"</span>,<span class="st">"a"</span><span class="op">)</span>
<span class="va">gorilla</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"a"</span>,<span class="st">"a"</span><span class="op">)</span></code></pre></div>
<p>We can examine each pair of sequences and count up the number of differences between each one. This gives us a preliminary estimate of how diverged the sequences are.</p>
<p>Humans and chimps have 2 differences</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">human</span>, <span class="va">chimp</span><span class="op">)</span>
<span class="co">#&gt;       [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8]</span>
<span class="co">#&gt; human "a"  "t"  "c"  "g"  "a"  "t"  "c"  "g" </span>
<span class="co">#&gt; chimp "a"  "t"  "c"  "a"  "a"  "t"  "c"  "a"</span></code></pre></div>
<p>Chimps and gorillas have 4 differences</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">chimp</span>, <span class="va">gorilla</span><span class="op">)</span>
<span class="co">#&gt;         [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8]</span>
<span class="co">#&gt; chimp   "a"  "t"  "c"  "a"  "a"  "t"  "c"  "a" </span>
<span class="co">#&gt; gorilla "a"  "a"  "a"  "a"  "a"  "a"  "a"  "a"</span></code></pre></div>
<p>Humans and gorillas have 6 differences</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">human</span>, <span class="va">gorilla</span><span class="op">)</span>
<span class="co">#&gt;         [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8]</span>
<span class="co">#&gt; human   "a"  "t"  "c"  "g"  "a"  "t"  "c"  "g" </span>
<span class="co">#&gt; gorilla "a"  "a"  "a"  "a"  "a"  "a"  "a"  "a"</span></code></pre></div>
<p>We could represent this data as a genetic distance matrix like this</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my.dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="cn">NA</span>, <span class="cn">NA</span>,
                            <span class="fl">2</span>, <span class="cn">NA</span>,<span class="cn">NA</span>,
                            <span class="fl">6</span>, <span class="fl">4</span>, <span class="cn">NA</span><span class="op">)</span>,
                  nrow <span class="op">=</span> <span class="fl">3</span>, 
                  byrow <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">my.dist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"H"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">my.dist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"H"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>
<span class="va">my.dist</span>
<span class="co">#&gt;    H  C  G</span>
<span class="co">#&gt; H NA NA NA</span>
<span class="co">#&gt; C  2 NA NA</span>
<span class="co">#&gt; G  6  4 NA</span></code></pre></div>
<p>Alternatively, we could make each row of the matrix into a vector and then stack them with rbind</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">human.row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="cn">NA</span>,<span class="cn">NA</span><span class="op">)</span>
<span class="va">chimp.row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="fl">2</span>,<span class="cn">NA</span>,<span class="cn">NA</span><span class="op">)</span>
<span class="va">gorilla.row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">4</span>,<span class="cn">NA</span><span class="op">)</span>

<span class="va">my.dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">human.row</span>, <span class="va">chimp.row</span>, <span class="va">gorilla.row</span><span class="op">)</span>

<span class="va">spp.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"human"</span>,<span class="st">"chimp"</span>,<span class="st">"gorilla"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">my.dist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">spp.names</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">my.dist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">spp.names</span>
<span class="va">my.dist</span>
<span class="co">#&gt;         human chimp gorilla</span>
<span class="co">#&gt; human      NA    NA      NA</span>
<span class="co">#&gt; chimp       2    NA      NA</span>
<span class="co">#&gt; gorilla     6     4      NA</span></code></pre></div>
<p>Note that we leave the values on the diagonal as NA because the genetic distance between humans and humans, or chimps and chimps, is 0 and isn’t helpful. A matrix like this is also symmetrical, where the lower triangle portion is the same as the upper portion We could build the matrix like the one below if we wanted, but the convention is to use the lower.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">human.row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="fl">2</span>,<span class="fl">6</span><span class="op">)</span>
<span class="va">chimp.row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="cn">NA</span>,<span class="cn">NA</span>,<span class="fl">4</span><span class="op">)</span>
<span class="va">gorilla.row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="cn">NA</span>,<span class="cn">NA</span><span class="op">)</span>

<span class="va">my.dist.upper</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">human.row</span>, <span class="va">chimp.row</span>, <span class="va">gorilla.row</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">my.dist.upper</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">spp.names</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">my.dist.upper</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">spp.names</span></code></pre></div>
<p>Compare them</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my.dist</span>
<span class="co">#&gt;         human chimp gorilla</span>
<span class="co">#&gt; human      NA    NA      NA</span>
<span class="co">#&gt; chimp       2    NA      NA</span>
<span class="co">#&gt; gorilla     6     4      NA</span>
<span class="va">my.dist.upper</span>
<span class="co">#&gt;         human chimp gorilla</span>
<span class="co">#&gt; human      NA     2       6</span>
<span class="co">#&gt; chimp      NA    NA       4</span>
<span class="co">#&gt; gorilla    NA    NA      NA</span></code></pre></div>
<p>In both matrixes the distance from humans to gorillas is 6, humans to chimps is 2, and chimps to gorillas is 4. Make sure you understand why this is true.</p>
<p>Again, the lower triangular for is the form usually used. Because “NA” occurs in the entire top row and last column, people often omit it</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my.dist</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">3</span><span class="op">]</span>
<span class="co">#&gt;         human chimp</span>
<span class="co">#&gt; chimp       2    NA</span>
<span class="co">#&gt; gorilla     6     4</span></code></pre></div>
<p>These calculations imply that humans and chimps are more closely related. Phylogenetic algorithms would therefore cluster them into a clade. In Newick Format this would be (H,C).</p>
<p>Since we have just three taxa, we add G outside the first clade ((H,C),G)</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">str1</span><span class="op">&lt;-</span> <span class="st">"(G,(C, H) );"</span>
<span class="va">tree1</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>text<span class="op">=</span><span class="va">str1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">tree1</span>,edge.width<span class="op">=</span><span class="fl">2</span>, main <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p><img src="lll-UPGMA_in_R_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Newick format does not represent genetic distance/branch lengths, only the clade and structure. We can calculate branch lengths using the UPGMA algorithm.</p>
</div>
<div id="distance-matrix-in-r-from-raw-sequences" class="section level2">
<h2 class="hasAnchor">
<a href="#distance-matrix-in-r-from-raw-sequences" class="anchor"></a>Distance matrix in R from raw sequences</h2>
<p>To make a distance matrix in Rfrom the original sequences we first need to put our original sequences into a matrix</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my.seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">human</span>, 
                 <span class="va">chimp</span>, 
                 <span class="va">gorilla</span><span class="op">)</span></code></pre></div>
<p>We then need to do some fancy processing (this has to do with how the packages we’re going to use works, which is picky about formats because its optimized to accommodate real sequences which can be very very long). We’ll use the as.DNAbin() function.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">my.seqs.pro</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.alignment.html">as.DNAbin</a></span><span class="op">(</span><span class="va">my.seqs</span><span class="op">)</span></code></pre></div>
<p>We then can calculate the <strong>distance</strong> between each sequence in terms of the number (N) of mutations. This uses the dist.dna() function. We set the arguement model = “N”. This is the most basic and naive model of mutation and ignores the possibility of back mutations.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/dist.dna.html">dist.dna</a></span><span class="op">(</span><span class="va">my.seqs.pro</span>, model <span class="op">=</span> <span class="st">"N"</span><span class="op">)</span>

<span class="va">seq.dist0</span>
<span class="co">#&gt;         human chimp</span>
<span class="co">#&gt; chimp       2      </span>
<span class="co">#&gt; gorilla     6     4</span></code></pre></div>
</div>
<div id="distance-matrix-in-r-from-normal-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#distance-matrix-in-r-from-normal-matrix" class="anchor"></a>Distance matrix in R from normal matrix</h2>
<p>Sometimes we already have a distance matrix computed. In this case we can convert our normal matrix into a properly formatted distance matrix. This requires the as.dist() function. I’ll call this object “seq.dist.from.mat” to indicate that it came “from a matrix”. Note that I’m going to give it the full 3 x x matrix, but</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist.from.mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span><span class="op">(</span><span class="va">my.dist</span><span class="op">)</span></code></pre></div>
<p>I’ll now compare the two ways of making the matrix: seq.dist0, which was build from the original sequences, and seq.dist.from.mat, which was built from the hand-entered matrix.</p>
<p>Firt, i’ll just look at them</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist0</span>
<span class="co">#&gt;         human chimp</span>
<span class="co">#&gt; chimp       2      </span>
<span class="co">#&gt; gorilla     6     4</span>
<span class="va">seq.dist.from.mat</span>
<span class="co">#&gt;         human chimp</span>
<span class="co">#&gt; chimp       2      </span>
<span class="co">#&gt; gorilla     6     4</span></code></pre></div>
<p>Then check them out with is() and class()</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">is</span><span class="op">(</span><span class="va">seq.dist0</span><span class="op">)</span> <span class="op">==</span> <span class="fu">is</span><span class="op">(</span><span class="va">seq.dist.from.mat</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">seq.dist0</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">seq.dist.from.mat</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>Now I’ll confirm that that have the same content (Note that distance matrices are not indexed internall by rows and columns, but by the order that the values appear. The following code therefores returns only 3 responses, not 4.)</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist0</span> <span class="op">==</span> <span class="va">seq.dist.from.mat</span>
<span class="co">#&gt; [1] TRUE TRUE TRUE</span></code></pre></div>
<p>Cool, so we can make a matrix either from the raw sequences or from a hand-entered matrix.</p>
</div>
<div id="upgma-in-r" class="section level2">
<h2 class="hasAnchor">
<a href="#upgma-in-r" class="anchor"></a>UPGMA in R</h2>
<p>Once you have a distance matrix, UPGMA is easy in R with the upgam() function</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">seq.upgma0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phangorn/man/upgma.html">upgma</a></span><span class="op">(</span><span class="va">seq.dist0</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">seq.upgma0</span>, main <span class="op">=</span> <span class="st">"orig"</span><span class="op">)</span></code></pre></div>
<p><img src="lll-UPGMA_in_R_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>This doesn’t look too different than the phylogeny produced from Newick. Let’s play with the distance matrix to see what happens.</p>
<p>First, make some more copies of the distance matrix</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist1</span> <span class="op">&lt;-</span> <span class="va">seq.dist0</span>
<span class="va">seq.dist2</span> <span class="op">&lt;-</span> <span class="va">seq.dist0</span>
<span class="va">seq.dist3</span> <span class="op">&lt;-</span> <span class="va">seq.dist0</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist1</span>
<span class="co">#&gt;         human chimp</span>
<span class="co">#&gt; chimp       2      </span>
<span class="co">#&gt; gorilla     6     4</span></code></pre></div>
<p>The relationship between humans and chimps in in the upper left hand cell of the matrix. Internally, R is calling this the first cell of the matrix</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist1</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; [1] 2</span></code></pre></div>
<p>Let’s increase and decrease this value</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist1</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">20</span> <span class="co">#change from 2 to 3.75</span>
<span class="va">seq.dist2</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0.5</span> <span class="co">#change from 2 to 0.5</span>
<span class="va">seq.dist3</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0.05</span> <span class="co">#change from 2 to 0.05</span></code></pre></div>
<p>Now we’ll plot the original and the other ones</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">seq.upgma0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phangorn/man/upgma.html">upgma</a></span><span class="op">(</span><span class="va">seq.dist0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">seq.upgma0</span>, main <span class="op">=</span> <span class="st">"orig"</span><span class="op">)</span>


<span class="va">seq.upgma1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phangorn/man/upgma.html">upgma</a></span><span class="op">(</span><span class="va">seq.dist1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">seq.upgma1</span>, main <span class="op">=</span> <span class="st">"longer"</span><span class="op">)</span>

<span class="va">seq.upgma2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phangorn/man/upgma.html">upgma</a></span><span class="op">(</span><span class="va">seq.dist2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">seq.upgma2</span>, main <span class="op">=</span> <span class="st">"shorter"</span><span class="op">)</span>


<span class="va">seq.upgma3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phangorn/man/upgma.html">upgma</a></span><span class="op">(</span><span class="va">seq.dist3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">seq.upgma3</span>, main <span class="op">=</span> <span class="st">"much shorter"</span><span class="op">)</span></code></pre></div>
<p><img src="lll-UPGMA_in_R_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
</div>
<div id="upgma-by-hand" class="section level2">
<h2 class="hasAnchor">
<a href="#upgma-by-hand" class="anchor"></a>UPGMA by hand</h2>
<p>The following is a brief overview of how branch lenths are calculated.</p>
<p>Human and chimps are closest</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.dist0</span>
<span class="co">#&gt;         human chimp</span>
<span class="co">#&gt; chimp       2      </span>
<span class="co">#&gt; gorilla     6     4</span>

<span class="va">d.min</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">seq.dist0</span><span class="op">)</span></code></pre></div>
<p>When two individual taxa are grouped into a clade the branch lengths are set at 1/2 the distance between them.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d.min</span><span class="op">/</span><span class="fl">2</span>
<span class="co">#&gt; [1] 1</span></code></pre></div>
<p>We can confirm that this is correct by checking what R calculated via the upgma() function. We can access R’s calculations by looking at the object we saved the upgma output too</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.upgma0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Phylogenetic tree with 3 tips and 2 internal nodes.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Tip labels:</span>
<span class="co">#&gt;   human, chimp, gorilla</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Rooted; includes branch lengths.</span></code></pre></div>
<p>On its own this isn’t helpful. But what we can do is use the str() command to look under the hood of the object</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">seq.upgma0</span><span class="op">)</span>
<span class="co">#&gt; List of 4</span>
<span class="co">#&gt;  $ edge       : int [1:4, 1:2] 5 5 4 4 1 2 3 5</span>
<span class="co">#&gt;  $ edge.length: num [1:4] 1 1 2.5 1.5</span>
<span class="co">#&gt;  $ tip.label  : chr [1:3] "human" "chimp" "gorilla"</span>
<span class="co">#&gt;  $ Nnode      : int 2</span>
<span class="co">#&gt;  - attr(*, "class")= chr "phylo"</span>
<span class="co">#&gt;  - attr(*, "order")= chr "postorder"</span></code></pre></div>
<p>The output “edge.length” tells us the shortest branch lengths are 1. This looks right (“edge” is the math term for branches in a tree).</p>
<p>Next we combine Humans and chimps into a clade, then Re-calculate the distance matrix with a new combined human-chimp grouping. The distance from this human-chimp group to gorillas is the average distance from humans to gorillas and chimps to gorillas.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#distance humans to gorillas</span>
<span class="va">Dh_g</span> <span class="op">&lt;-</span> <span class="fl">6</span>

<span class="co">#distance chimps to gorillas</span>
<span class="va">Dc_g</span> <span class="op">&lt;-</span> <span class="fl">4</span>

<span class="co">#mean of the two distances</span>
<span class="va">Dhc_g</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">Dh_g</span><span class="op">*</span><span class="fl">1</span> <span class="op">+</span> <span class="va">Dc_g</span><span class="op">*</span><span class="fl">1</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fl">1</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>The result is distance between our human-chimp clade and gorillas.</p>
<p>This new distance allows us to calculate the branch lengths for the rest of the tree</p>
<p>The branch length from the h-c clade to the gorillas g is 1/2 half the distance we just calculated:</p>
<p>Dhc_g/2 = 5/2 = 2.5.</p>
<p>The remaining branch length is 2.5 - 1 = 1.5. (Next week we’ll draw this out to show exactly how this works; see if you can figure out why this is done by sketching out the tree.)</p>
<p>These values match what is in the upgma object</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">seq.upgma0</span><span class="op">)</span>
<span class="co">#&gt; List of 4</span>
<span class="co">#&gt;  $ edge       : int [1:4, 1:2] 5 5 4 4 1 2 3 5</span>
<span class="co">#&gt;  $ edge.length: num [1:4] 1 1 2.5 1.5</span>
<span class="co">#&gt;  $ tip.label  : chr [1:3] "human" "chimp" "gorilla"</span>
<span class="co">#&gt;  $ Nnode      : int 2</span>
<span class="co">#&gt;  - attr(*, "class")= chr "phylo"</span>
<span class="co">#&gt;  - attr(*, "order")= chr "postorder"</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
