<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Simulating molecular evolution • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Simulating molecular evolution">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="lll-jc_sim_for_render_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Simulating molecular evolution</h1>
            
      
      
      <div class="hidden name"><code>lll-jc_sim_for_render.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="r-markdown" class="section level2">
<h2 class="hasAnchor">
<a href="#r-markdown" class="anchor"></a>R Markdown</h2>
<p>Human cytochrome C</p>
<p><a href="https://www.ncbi.nlm.nih.gov/gene/4512" class="uri">https://www.ncbi.nlm.nih.gov/gene/4512</a> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1698023/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1698023/</a></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cyo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">[1564 chars quoted with '"']</span><span class="op">)</span></code></pre></div>
<p>Use regular expression gsub() to remove the newline character</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cyo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\n"</span>,<span class="st">""</span>,<span class="va">cyo</span><span class="op">)</span></code></pre></div>
<p>Turn character string into vector using strsplit()</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cyo.vect.temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">cyo</span>,<span class="st">""</span><span class="op">)</span>
<span class="va">cyo.vect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">cyo.vect.temp</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cyo.vect</span><span class="op">)</span>
<span class="co">#&gt; cyo.vect</span>
<span class="co">#&gt;   A   C   G   T </span>
<span class="co">#&gt; 419 463 250 410</span></code></pre></div>
<p>Compare formats</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># original</span>
<span class="fu">is</span><span class="op">(</span><span class="va">cyo</span><span class="op">)</span>
<span class="co">#&gt; [1] "character"           "vector"              "data.frameRowLabels"</span>
<span class="co">#&gt; [4] "SuperClassMethod"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">cyo</span><span class="op">)</span>
<span class="co">#&gt;  chr "ATGTTCGCCGACCGTTGACTATTCTCTACAAACCACAAAGACATTGGAACACTATACCTATTATTCGGCGCATGAGCTGGAGTCCTAGGCACAGCTCTAAGCCTCCTTATT"| __truncated__</span>
<span class="va">cyo</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; [1] "ATGTTCGCCGACCGTTGACTATTCTCTACAAACCACAAAGACATTGGAACACTATACCTATTATTCGGCGCATGAGCTGGAGTCCTAGGCACAGCTCTAAGCCTCCTTATTCGAGCCGAGCTGGGCCAGCCAGGCAACCTTCTAGGTAACGACCACATCTACAACGTTATCGTCACAGCCCATGCATTTGTAATAATCTTCTTCATAGTAATACCCATCATAATCGGAGGCTTTGGCAACTGACTAGTTCCCCTAATAATCGGTGCCCCCGATATGGCGTTTCCCCGCATAAACAACATAAGCTTCTGACTCTTACCTCCCTCTCTCCTACTCCTGCTCGCATCTGCTATAGTGGAGGCCGGAGCAGGAACAGGTTGAACAGTCTACCCTCCCTTAGCAGGGAACTACTCCCACCCTGGAGCCTCCGTAGACCTAACCATCTTCTCCTTACACCTAGCAGGTGTCTCCTCTATCTTAGGGGCCATCAATTTCATCACAACAATTATCAATATAAAACCCCCTGCCATAACCCAATACCAAACGCCCCTCTTCGTCTGATCCGTCCTAATCACAGCAGTCCTACTTCTCCTATCTCTCCCAGTCCTAGCTGCTGGCATCACTATACTACTAACAGACCGCAACCTCAACACCACCTTCTTCGACCCCGCCGGAGGAGGAGACCCCATTCTATACCAACACCTATTCTGATTTTTCGGTCACCCTGAAGTTTATATTCTTATCCTACCAGGCTTCGGAATAATCTCCCATATTGTAACTTACTACTCCGGAAAAAAAGAACCATTTGGATACATAGGTATGGTCTGAGCTATGATATCAATTGGCTTCCTAGGGTTTATCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACGAGCATATTTCACCTCCGCTACCATAATCATCGCTATCCCCACCGGCGTCAAAGTATTTAGCTGACTCGCCACACTCCACGGAAGCAATATGAAATGATCTGCTGCAGTGCTCTGAGCCCTAGGATTCATCTTTCTTTTCACCGTAGGTGGCCTGACTGGCATTGTATTAGCAAACTCATCACTAGACATCGTACTACACGACACGTACTACGTTGTAGCCCACTTCCACTATGTCCTATCAATAGGAGCTGTATTTGCCATCATAGGAGGCTTCATTCACTGATTTCCCCTATTCTCAGGCTACACCCTAGACCAAACCTACGCCAAAATCCATTTCACTATCATATTCATCGGCGTAAATCTAACTTTCTTCCCACAACACTTTCTCGGCCTATCCGGAATGCCCCGACGTTACTCGGACTACCCCGATGCATACACCACATGAAACATCCTATCATCTGTAGGCTCATTCATTTCTCTAACAGCAGTAATATTAATAATTTTCATGATTTGAGAAGCCTTCGCTTCGAAGCGAAAAGTCCTAATAGTAGAAGAACCCTCCATAAACCTGGAGTGACTATATGGATGCCCCCCACCCTACCACACATTCGAAGAACCCGTATACATAAAATCTAGA"</span>

<span class="co"># after strin spli</span>
<span class="fu">is</span><span class="op">(</span><span class="va">cyo.vect.temp</span><span class="op">)</span>
<span class="co">#&gt; [1] "list"   "vector"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">cyo.vect.temp</span><span class="op">)</span>
<span class="co">#&gt; List of 1</span>
<span class="co">#&gt;  $ : chr [1:1542] "A" "T" "G" "T" ...</span>
<span class="va">cyo.vect.temp</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt;    [1] "A" "T" "G" "T" "T" "C" "G" "C" "C" "G" "A" "C" "C" "G" "T" "T" "G" "A"</span>
<span class="co">#&gt;   [19] "C" "T" "A" "T" "T" "C" "T" "C" "T" "A" "C" "A" "A" "A" "C" "C" "A" "C"</span>
<span class="co">#&gt;   [37] "A" "A" "A" "G" "A" "C" "A" "T" "T" "G" "G" "A" "A" "C" "A" "C" "T" "A"</span>
<span class="co">#&gt;   [55] "T" "A" "C" "C" "T" "A" "T" "T" "A" "T" "T" "C" "G" "G" "C" "G" "C" "A"</span>
<span class="co">#&gt;   [73] "T" "G" "A" "G" "C" "T" "G" "G" "A" "G" "T" "C" "C" "T" "A" "G" "G" "C"</span>
<span class="co">#&gt;   [91] "A" "C" "A" "G" "C" "T" "C" "T" "A" "A" "G" "C" "C" "T" "C" "C" "T" "T"</span>
<span class="co">#&gt;  [109] "A" "T" "T" "C" "G" "A" "G" "C" "C" "G" "A" "G" "C" "T" "G" "G" "G" "C"</span>
<span class="co">#&gt;  [127] "C" "A" "G" "C" "C" "A" "G" "G" "C" "A" "A" "C" "C" "T" "T" "C" "T" "A"</span>
<span class="co">#&gt;  [145] "G" "G" "T" "A" "A" "C" "G" "A" "C" "C" "A" "C" "A" "T" "C" "T" "A" "C"</span>
<span class="co">#&gt;  [163] "A" "A" "C" "G" "T" "T" "A" "T" "C" "G" "T" "C" "A" "C" "A" "G" "C" "C"</span>
<span class="co">#&gt;  [181] "C" "A" "T" "G" "C" "A" "T" "T" "T" "G" "T" "A" "A" "T" "A" "A" "T" "C"</span>
<span class="co">#&gt;  [199] "T" "T" "C" "T" "T" "C" "A" "T" "A" "G" "T" "A" "A" "T" "A" "C" "C" "C"</span>
<span class="co">#&gt;  [217] "A" "T" "C" "A" "T" "A" "A" "T" "C" "G" "G" "A" "G" "G" "C" "T" "T" "T"</span>
<span class="co">#&gt;  [235] "G" "G" "C" "A" "A" "C" "T" "G" "A" "C" "T" "A" "G" "T" "T" "C" "C" "C"</span>
<span class="co">#&gt;  [253] "C" "T" "A" "A" "T" "A" "A" "T" "C" "G" "G" "T" "G" "C" "C" "C" "C" "C"</span>
<span class="co">#&gt;  [271] "G" "A" "T" "A" "T" "G" "G" "C" "G" "T" "T" "T" "C" "C" "C" "C" "G" "C"</span>
<span class="co">#&gt;  [289] "A" "T" "A" "A" "A" "C" "A" "A" "C" "A" "T" "A" "A" "G" "C" "T" "T" "C"</span>
<span class="co">#&gt;  [307] "T" "G" "A" "C" "T" "C" "T" "T" "A" "C" "C" "T" "C" "C" "C" "T" "C" "T"</span>
<span class="co">#&gt;  [325] "C" "T" "C" "C" "T" "A" "C" "T" "C" "C" "T" "G" "C" "T" "C" "G" "C" "A"</span>
<span class="co">#&gt;  [343] "T" "C" "T" "G" "C" "T" "A" "T" "A" "G" "T" "G" "G" "A" "G" "G" "C" "C"</span>
<span class="co">#&gt;  [361] "G" "G" "A" "G" "C" "A" "G" "G" "A" "A" "C" "A" "G" "G" "T" "T" "G" "A"</span>
<span class="co">#&gt;  [379] "A" "C" "A" "G" "T" "C" "T" "A" "C" "C" "C" "T" "C" "C" "C" "T" "T" "A"</span>
<span class="co">#&gt;  [397] "G" "C" "A" "G" "G" "G" "A" "A" "C" "T" "A" "C" "T" "C" "C" "C" "A" "C"</span>
<span class="co">#&gt;  [415] "C" "C" "T" "G" "G" "A" "G" "C" "C" "T" "C" "C" "G" "T" "A" "G" "A" "C"</span>
<span class="co">#&gt;  [433] "C" "T" "A" "A" "C" "C" "A" "T" "C" "T" "T" "C" "T" "C" "C" "T" "T" "A"</span>
<span class="co">#&gt;  [451] "C" "A" "C" "C" "T" "A" "G" "C" "A" "G" "G" "T" "G" "T" "C" "T" "C" "C"</span>
<span class="co">#&gt;  [469] "T" "C" "T" "A" "T" "C" "T" "T" "A" "G" "G" "G" "G" "C" "C" "A" "T" "C"</span>
<span class="co">#&gt;  [487] "A" "A" "T" "T" "T" "C" "A" "T" "C" "A" "C" "A" "A" "C" "A" "A" "T" "T"</span>
<span class="co">#&gt;  [505] "A" "T" "C" "A" "A" "T" "A" "T" "A" "A" "A" "A" "C" "C" "C" "C" "C" "T"</span>
<span class="co">#&gt;  [523] "G" "C" "C" "A" "T" "A" "A" "C" "C" "C" "A" "A" "T" "A" "C" "C" "A" "A"</span>
<span class="co">#&gt;  [541] "A" "C" "G" "C" "C" "C" "C" "T" "C" "T" "T" "C" "G" "T" "C" "T" "G" "A"</span>
<span class="co">#&gt;  [559] "T" "C" "C" "G" "T" "C" "C" "T" "A" "A" "T" "C" "A" "C" "A" "G" "C" "A"</span>
<span class="co">#&gt;  [577] "G" "T" "C" "C" "T" "A" "C" "T" "T" "C" "T" "C" "C" "T" "A" "T" "C" "T"</span>
<span class="co">#&gt;  [595] "C" "T" "C" "C" "C" "A" "G" "T" "C" "C" "T" "A" "G" "C" "T" "G" "C" "T"</span>
<span class="co">#&gt;  [613] "G" "G" "C" "A" "T" "C" "A" "C" "T" "A" "T" "A" "C" "T" "A" "C" "T" "A"</span>
<span class="co">#&gt;  [631] "A" "C" "A" "G" "A" "C" "C" "G" "C" "A" "A" "C" "C" "T" "C" "A" "A" "C"</span>
<span class="co">#&gt;  [649] "A" "C" "C" "A" "C" "C" "T" "T" "C" "T" "T" "C" "G" "A" "C" "C" "C" "C"</span>
<span class="co">#&gt;  [667] "G" "C" "C" "G" "G" "A" "G" "G" "A" "G" "G" "A" "G" "A" "C" "C" "C" "C"</span>
<span class="co">#&gt;  [685] "A" "T" "T" "C" "T" "A" "T" "A" "C" "C" "A" "A" "C" "A" "C" "C" "T" "A"</span>
<span class="co">#&gt;  [703] "T" "T" "C" "T" "G" "A" "T" "T" "T" "T" "T" "C" "G" "G" "T" "C" "A" "C"</span>
<span class="co">#&gt;  [721] "C" "C" "T" "G" "A" "A" "G" "T" "T" "T" "A" "T" "A" "T" "T" "C" "T" "T"</span>
<span class="co">#&gt;  [739] "A" "T" "C" "C" "T" "A" "C" "C" "A" "G" "G" "C" "T" "T" "C" "G" "G" "A"</span>
<span class="co">#&gt;  [757] "A" "T" "A" "A" "T" "C" "T" "C" "C" "C" "A" "T" "A" "T" "T" "G" "T" "A"</span>
<span class="co">#&gt;  [775] "A" "C" "T" "T" "A" "C" "T" "A" "C" "T" "C" "C" "G" "G" "A" "A" "A" "A"</span>
<span class="co">#&gt;  [793] "A" "A" "A" "G" "A" "A" "C" "C" "A" "T" "T" "T" "G" "G" "A" "T" "A" "C"</span>
<span class="co">#&gt;  [811] "A" "T" "A" "G" "G" "T" "A" "T" "G" "G" "T" "C" "T" "G" "A" "G" "C" "T"</span>
<span class="co">#&gt;  [829] "A" "T" "G" "A" "T" "A" "T" "C" "A" "A" "T" "T" "G" "G" "C" "T" "T" "C"</span>
<span class="co">#&gt;  [847] "C" "T" "A" "G" "G" "G" "T" "T" "T" "A" "T" "C" "G" "T" "G" "T" "G" "A"</span>
<span class="co">#&gt;  [865] "G" "C" "A" "C" "A" "C" "C" "A" "T" "A" "T" "A" "T" "T" "T" "A" "C" "A"</span>
<span class="co">#&gt;  [883] "G" "T" "A" "G" "G" "A" "A" "T" "A" "G" "A" "C" "G" "T" "A" "G" "A" "C"</span>
<span class="co">#&gt;  [901] "A" "C" "A" "C" "G" "A" "G" "C" "A" "T" "A" "T" "T" "T" "C" "A" "C" "C"</span>
<span class="co">#&gt;  [919] "T" "C" "C" "G" "C" "T" "A" "C" "C" "A" "T" "A" "A" "T" "C" "A" "T" "C"</span>
<span class="co">#&gt;  [937] "G" "C" "T" "A" "T" "C" "C" "C" "C" "A" "C" "C" "G" "G" "C" "G" "T" "C"</span>
<span class="co">#&gt;  [955] "A" "A" "A" "G" "T" "A" "T" "T" "T" "A" "G" "C" "T" "G" "A" "C" "T" "C"</span>
<span class="co">#&gt;  [973] "G" "C" "C" "A" "C" "A" "C" "T" "C" "C" "A" "C" "G" "G" "A" "A" "G" "C"</span>
<span class="co">#&gt;  [991] "A" "A" "T" "A" "T" "G" "A" "A" "A" "T" "G" "A" "T" "C" "T" "G" "C" "T"</span>
<span class="co">#&gt; [1009] "G" "C" "A" "G" "T" "G" "C" "T" "C" "T" "G" "A" "G" "C" "C" "C" "T" "A"</span>
<span class="co">#&gt; [1027] "G" "G" "A" "T" "T" "C" "A" "T" "C" "T" "T" "T" "C" "T" "T" "T" "T" "C"</span>
<span class="co">#&gt; [1045] "A" "C" "C" "G" "T" "A" "G" "G" "T" "G" "G" "C" "C" "T" "G" "A" "C" "T"</span>
<span class="co">#&gt; [1063] "G" "G" "C" "A" "T" "T" "G" "T" "A" "T" "T" "A" "G" "C" "A" "A" "A" "C"</span>
<span class="co">#&gt; [1081] "T" "C" "A" "T" "C" "A" "C" "T" "A" "G" "A" "C" "A" "T" "C" "G" "T" "A"</span>
<span class="co">#&gt; [1099] "C" "T" "A" "C" "A" "C" "G" "A" "C" "A" "C" "G" "T" "A" "C" "T" "A" "C"</span>
<span class="co">#&gt; [1117] "G" "T" "T" "G" "T" "A" "G" "C" "C" "C" "A" "C" "T" "T" "C" "C" "A" "C"</span>
<span class="co">#&gt; [1135] "T" "A" "T" "G" "T" "C" "C" "T" "A" "T" "C" "A" "A" "T" "A" "G" "G" "A"</span>
<span class="co">#&gt; [1153] "G" "C" "T" "G" "T" "A" "T" "T" "T" "G" "C" "C" "A" "T" "C" "A" "T" "A"</span>
<span class="co">#&gt; [1171] "G" "G" "A" "G" "G" "C" "T" "T" "C" "A" "T" "T" "C" "A" "C" "T" "G" "A"</span>
<span class="co">#&gt; [1189] "T" "T" "T" "C" "C" "C" "C" "T" "A" "T" "T" "C" "T" "C" "A" "G" "G" "C"</span>
<span class="co">#&gt; [1207] "T" "A" "C" "A" "C" "C" "C" "T" "A" "G" "A" "C" "C" "A" "A" "A" "C" "C"</span>
<span class="co">#&gt; [1225] "T" "A" "C" "G" "C" "C" "A" "A" "A" "A" "T" "C" "C" "A" "T" "T" "T" "C"</span>
<span class="co">#&gt; [1243] "A" "C" "T" "A" "T" "C" "A" "T" "A" "T" "T" "C" "A" "T" "C" "G" "G" "C"</span>
<span class="co">#&gt; [1261] "G" "T" "A" "A" "A" "T" "C" "T" "A" "A" "C" "T" "T" "T" "C" "T" "T" "C"</span>
<span class="co">#&gt; [1279] "C" "C" "A" "C" "A" "A" "C" "A" "C" "T" "T" "T" "C" "T" "C" "G" "G" "C"</span>
<span class="co">#&gt; [1297] "C" "T" "A" "T" "C" "C" "G" "G" "A" "A" "T" "G" "C" "C" "C" "C" "G" "A"</span>
<span class="co">#&gt; [1315] "C" "G" "T" "T" "A" "C" "T" "C" "G" "G" "A" "C" "T" "A" "C" "C" "C" "C"</span>
<span class="co">#&gt; [1333] "G" "A" "T" "G" "C" "A" "T" "A" "C" "A" "C" "C" "A" "C" "A" "T" "G" "A"</span>
<span class="co">#&gt; [1351] "A" "A" "C" "A" "T" "C" "C" "T" "A" "T" "C" "A" "T" "C" "T" "G" "T" "A"</span>
<span class="co">#&gt; [1369] "G" "G" "C" "T" "C" "A" "T" "T" "C" "A" "T" "T" "T" "C" "T" "C" "T" "A"</span>
<span class="co">#&gt; [1387] "A" "C" "A" "G" "C" "A" "G" "T" "A" "A" "T" "A" "T" "T" "A" "A" "T" "A"</span>
<span class="co">#&gt; [1405] "A" "T" "T" "T" "T" "C" "A" "T" "G" "A" "T" "T" "T" "G" "A" "G" "A" "A"</span>
<span class="co">#&gt; [1423] "G" "C" "C" "T" "T" "C" "G" "C" "T" "T" "C" "G" "A" "A" "G" "C" "G" "A"</span>
<span class="co">#&gt; [1441] "A" "A" "A" "G" "T" "C" "C" "T" "A" "A" "T" "A" "G" "T" "A" "G" "A" "A"</span>
<span class="co">#&gt; [1459] "G" "A" "A" "C" "C" "C" "T" "C" "C" "A" "T" "A" "A" "A" "C" "C" "T" "G"</span>
<span class="co">#&gt; [1477] "G" "A" "G" "T" "G" "A" "C" "T" "A" "T" "A" "T" "G" "G" "A" "T" "G" "C"</span>
<span class="co">#&gt; [1495] "C" "C" "C" "C" "C" "A" "C" "C" "C" "T" "A" "C" "C" "A" "C" "A" "C" "A"</span>
<span class="co">#&gt; [1513] "T" "T" "C" "G" "A" "A" "G" "A" "A" "C" "C" "C" "G" "T" "A" "T" "A" "C"</span>
<span class="co">#&gt; [1531] "A" "T" "A" "A" "A" "A" "T" "C" "T" "A" "G" "A"</span>

<span class="co">#final</span>
<span class="fu">is</span><span class="op">(</span><span class="va">cyo.vect</span><span class="op">)</span>
<span class="co">#&gt; [1] "character"           "vector"              "data.frameRowLabels"</span>
<span class="co">#&gt; [4] "SuperClassMethod"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">cyo.vect</span><span class="op">)</span>
<span class="co">#&gt;  chr [1:1542] "A" "T" "G" "T" "T" "C" "G" "C" "C" "G" "A" "C" "C" "G" "T" ...</span>
<span class="va">cyo.vect</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; [1] "A"</span>
<span class="va">cyo.vect</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="co">#&gt;  [1] "A" "T" "G" "T" "T" "C" "G" "C" "C" "G"</span></code></pre></div>
</div>
<div id="simulate-mutation-at-a-single-locus" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-mutation-at-a-single-locus" class="anchor"></a>Simulate mutation at a single locus</h2>
<p>Function to mutate a single base</p>
<p>Function has 3 arguments</p>
<ul>
<li>original; where we started</li>
</ul>
<p>This will take on different values as we cycle through a sequence. It has a default but normally will change as a loop progresses through a sequence.</p>
<ul>
<li>possible; what we can become</li>
</ul>
<p>This could be set to amino acid</p>
<ul>
<li>probs; probility of each thing in “possible” happening</li>
</ul>
<p>These have really high default values for illustration purposes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mutate_position</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">original</span> <span class="op">=</span> <span class="st">"A"</span>,
                            <span class="va">possible</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"T"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>,
                            <span class="va">probs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">/</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">/</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">/</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  
  <span class="co"># First, if "original"is an NA, throw an erro</span>
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">original</span><span class="op">)</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/warning.html">warning</a></span><span class="op">(</span><span class="st">"Thing to mutate is not listed in possible nucleotides or amino acids.  Check the data"</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">original</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/base/browser.html">browser</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">}</span>

  <span class="co"># determine what the current thing to mutate is</span>
  <span class="va">i.original</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">possible</span> <span class="op">==</span> <span class="va">original</span><span class="op">)</span>
  
  <span class="co"># remove the current status from the possible mutations</span>
  <span class="va">possible.working</span> <span class="op">&lt;-</span> <span class="va">possible</span><span class="op">[</span><span class="op">-</span><span class="va">i.original</span><span class="op">]</span>
  
  <span class="co"># assign a new base</span>
  <span class="va">new.base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">possible.working</span>, 
                size <span class="op">=</span> <span class="fl">1</span>, 
                prob <span class="op">=</span> <span class="va">probs</span><span class="op">)</span>
  
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">new.base</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>Test the function</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mutate_position</span><span class="op">(</span><span class="op">)</span>
<span class="fu">mutate_position</span><span class="op">(</span>original <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span>
<span class="fu">mutate_position</span><span class="op">(</span>original <span class="op">=</span> <span class="st">"T"</span><span class="op">)</span>
<span class="fu">mutate_position</span><span class="op">(</span>original <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span>
<span class="fu">mutate_position</span><span class="op">(</span>original <span class="op">=</span> <span class="st">"G"</span><span class="op">)</span>
<span class="fu">mutate_position</span><span class="op">(</span>original <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
</div>
<div id="function-to-mutate-an-entire-sequence" class="section level2">
<h2 class="hasAnchor">
<a href="#function-to-mutate-an-entire-sequence" class="anchor"></a>Function to mutate an entire sequence</h2>
<p>This calls the function mutation_position()</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mutate_sequence</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">u</span> <span class="op">=</span> <span class="fl">2e-01</span>,
                            <span class="va">seq.t1</span> <span class="op">=</span> <span class="va">cyo.vect</span><span class="op">)</span><span class="op">{</span>
  
  <span class="co"># length of sequence</span>
  <span class="va">seq.len</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">seq.t1</span><span class="op">)</span>
  
  <span class="co"># create copy of seq to make changes to</span>
  <span class="va">seq.t2</span> <span class="op">&lt;-</span> <span class="va">seq.t1</span>
  
  <span class="co"># counter of total muatations</span>
  <span class="va">tot.mutations</span> <span class="op">&lt;-</span> <span class="fl">0</span>
  
  <span class="co"># generate locations of mutations</span>
  <span class="co">## for each position, generate 0 or 1, with probablity of 1 = u</span>
  <span class="co">## returns a vector with 0 or 1 for each position</span>
  <span class="va">mutated.positions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Binomial.html">rbinom</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">seq.len</span>, <span class="co"># number of positions</span>
                              size <span class="op">=</span> <span class="fl">1</span>,
                              prob <span class="op">=</span> <span class="va">u</span><span class="op">)</span>
  
  <span class="co"># apply mutations</span>
  
  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">mutated.positions</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">{</span>
    <span class="va">mutation.locations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">mutated.positions</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span>
  
    <span class="kw">for</span><span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">mutation.locations</span><span class="op">)</span><span class="op">{</span>
     <span class="va">base.j</span> <span class="op">&lt;-</span> <span class="va">seq.t1</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>
     
     <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">base.j</span><span class="op">)</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">{</span>
       <span class="fu"><a href="https://rdrr.io/r/base/browser.html">browser</a></span><span class="op">(</span><span class="op">)</span>
     <span class="op">}</span>
     
     <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">is.character</a></span><span class="op">(</span><span class="va">base.j</span><span class="op">)</span> <span class="op">==</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span>
       <span class="fu"><a href="https://rdrr.io/r/base/browser.html">browser</a></span><span class="op">(</span><span class="op">)</span>
     <span class="op">}</span>
     
     <span class="va">new.base.j</span> <span class="op">&lt;-</span> <span class="fu">mutate_position</span><span class="op">(</span>original <span class="op">=</span> <span class="va">base.j</span><span class="op">)</span>
      
     <span class="va">seq.t2</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">new.base.j</span>
      
    <span class="co"># determine the total number of </span>
    <span class="va">tot.mutations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">mutated.positions</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="op">}</span>
  
  <span class="co"># output</span>
  <span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>tot.mutations <span class="op">=</span> <span class="va">tot.mutations</span>,
                 seq.t1 <span class="op">=</span> <span class="va">seq.t1</span>,
                 seq.t2 <span class="op">=</span> <span class="va">seq.t2</span> <span class="op">)</span>
  
   <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span>
  
  <span class="op">}</span>
 
    </code></pre></div>
<p>Test the function</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cyo.vect.short</span> <span class="op">&lt;-</span> <span class="va">cyo.vect</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">25</span><span class="op">]</span>

<span class="fu">mutate_sequence</span><span class="op">(</span>u <span class="op">=</span> <span class="fl">0.05</span>,seq.t1 <span class="op">=</span> <span class="va">cyo.vect.short</span> <span class="op">)</span>
<span class="co">#&gt; $tot.mutations</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $seq.t1</span>
<span class="co">#&gt;  [1] "A" "T" "G" "T" "T" "C" "G" "C" "C" "G" "A" "C" "C" "G" "T" "T" "G" "A" "C"</span>
<span class="co">#&gt; [20] "T" "A" "T" "T" "C" "T"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $seq.t2</span>
<span class="co">#&gt;  [1] "A" "T" "G" "T" "T" "C" "G" "C" "C" "G" "A" "C" "C" "G" "T" "T" "G" "A" "C"</span>
<span class="co">#&gt; [20] "T" "A" "T" "T" "C" "T"</span></code></pre></div>
</div>
<div id="simulate-evolution-of-2-sequences" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-evolution-of-2-sequences" class="anchor"></a>Simulate evolution of 2 sequences</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">diverge_two_sequences</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">seq1</span>,
                                  <span class="va">seq2</span>,
                                  <span class="va">u</span> <span class="op">=</span> <span class="fl">2e-03</span>,
                                  <span class="va">time</span> <span class="op">=</span> <span class="fl">1e4</span><span class="op">)</span><span class="op">{</span>
  
  <span class="co"># create data storage dataframe</span>
  <span class="va">time.series</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">time</span>,
                            tot.mutations.s1 <span class="op">=</span> <span class="cn">NA</span>,
                            tot.mutations.s2 <span class="op">=</span> <span class="cn">NA</span>,
                            vis.mutations.s1 <span class="op">=</span> <span class="cn">NA</span>,
                            vis.mutations.s2 <span class="op">=</span> <span class="cn">NA</span>,
                            pairwise.diff <span class="op">=</span> <span class="cn">NA</span>,
                            seq1 <span class="op">=</span> <span class="cn">NA</span>,
                            seq2 <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>
  
  <span class="co"># Store starting sequences (not implemented)</span>
  <span class="co"># time.series[i, "seq1"] &lt;- seq1</span>
  <span class="co"># time.series[i, "seq2"] &lt;- seq2</span>
  
  <span class="co"># Save original sequence</span>
  <span class="va">seq1.orig</span> <span class="op">&lt;-</span> <span class="va">seq1</span>
  <span class="va">seq2.orig</span> <span class="op">&lt;-</span> <span class="va">seq2</span>
  
  <span class="co"># set initial njmber of mutations to 0</span>
  <span class="va">time.series</span><span class="op">[</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tot.mutations.s1"</span>,<span class="st">"tot.mutations.s2"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>
  <span class="va">time.series</span><span class="op">[</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"vis.mutations.s1"</span>,<span class="st">"vis.mutations.s2"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>
    
  <span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">2</span><span class="op">:</span><span class="va">time</span><span class="op">)</span><span class="op">{</span>
   <span class="co"># print(i)</span>
    <span class="co">#mutate sequence 1</span>
    <span class="co">## run mutation function</span>
    <span class="co">### This function 1) mutates an input sequence </span>
    <span class="co">###               2) compares mutated to input sequence</span>
    <span class="co">###               3) determines number of differences beteween</span>
    <span class="co">###                  input and mutated sequence</span>
    <span class="va">seq1.output.i</span> <span class="op">&lt;-</span> <span class="fu">mutate_sequence</span><span class="op">(</span>u <span class="op">=</span> <span class="va">u</span>, 
                                     seq.t1 <span class="op">=</span> <span class="va">seq1</span><span class="op">)</span>
    
    <span class="co">#mutate sequence 2</span>
    <span class="va">seq2.output.i</span> <span class="op">&lt;-</span> <span class="fu">mutate_sequence</span><span class="op">(</span>u <span class="op">=</span> <span class="va">u</span>, 
                                     seq.t1 <span class="op">=</span> <span class="va">seq2</span><span class="op">)</span>
    

    
    <span class="co">#determine cumulative number of mutations</span>
    <span class="co">## add up previous total of mutations to number that occurred</span>
    <span class="co">## during this time step.  </span>
    <span class="co">## This will include any back mutations since number of mutations</span>
    <span class="co">## is determined by compareing input vs. output sequence of seq1.output.x</span>
    <span class="co">## This is the TRUE genetic distance - ALL mutations of ANY kind</span>
    <span class="va">time.series</span><span class="op">[</span><span class="va">i</span>, <span class="st">"tot.mutations.s1"</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">time.series</span><span class="op">[</span><span class="va">i</span><span class="op">-</span><span class="fl">1</span>, <span class="st">"tot.mutations.s1"</span><span class="op">]</span> <span class="op">+</span>
                                         <span class="va">seq1.output.i</span><span class="op">$</span><span class="va">tot.mutations</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
    
    <span class="va">time.series</span><span class="op">[</span><span class="va">i</span>, <span class="st">"tot.mutations.s2"</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">time.series</span><span class="op">[</span><span class="va">i</span><span class="op">-</span><span class="fl">1</span>, <span class="st">"tot.mutations.s2"</span><span class="op">]</span> <span class="op">+</span>
                                         <span class="va">seq2.output.i</span><span class="op">$</span><span class="va">tot.mutations</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
    
    <span class="co"># time.series[i+1, "seq1"] &lt;- seq1.output.i$seq.t2[1]</span>
    <span class="co"># time.series[i+1, "seq2"] &lt;- seq2.output.i$seq.t2[1]</span>
    
    <span class="co"># Divergence from original sequence</span>
    <span class="co">## Determine where in current sequence there are differrences</span>
    <span class="co">## with the original sequence</span>
    <span class="va">seq1.delta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">seq1.orig</span> <span class="op">!=</span> 
                        <span class="va">seq1.output.i</span><span class="op">$</span><span class="va">seq.t2</span><span class="op">)</span>
    <span class="va">seq2.delta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">seq2.orig</span> <span class="op">!=</span> 
                        <span class="va">seq2.output.i</span><span class="op">$</span><span class="va">seq.t2</span><span class="op">)</span>
    
    
    <span class="co"># Count up number of differences with lenght()</span>
    <span class="co">## This is ONLY differeneces we'd see if we had access to the ancestral sequence</span>
    <span class="co">## A-&gt;T-&gt;A over 3 times steps = 0 mutations.</span>
    <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">seq1.delta</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">{</span>
      <span class="va">time.series</span><span class="op">[</span><span class="va">i</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"vis.mutations.s1"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">seq1.delta</span><span class="op">)</span>
    <span class="op">}</span>
    
    <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">seq2.delta</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">{</span>
      <span class="va">time.series</span><span class="op">[</span><span class="va">i</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"vis.mutations.s2"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">seq2.delta</span><span class="op">)</span>
    <span class="op">}</span>
    
    
    <span class="va">seq1</span> <span class="op">&lt;-</span> <span class="va">seq1.output.i</span><span class="op">$</span><span class="va">seq.t2</span>
    <span class="va">seq2</span> <span class="op">&lt;-</span> <span class="va">seq2.output.i</span><span class="op">$</span><span class="va">seq.t2</span>
    
    <span class="va">pairwise.diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">seq1.output.i</span><span class="op">$</span><span class="va">seq.t2</span><span class="op">!=</span> 
                           <span class="va">seq2.output.i</span><span class="op">$</span><span class="va">seq.t2</span><span class="op">)</span>
  
    <span class="va">time.series</span><span class="op">$</span><span class="va">pairwise.diff</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">pairwise.diff</span><span class="op">)</span>
    
  <span class="op">}</span>
  
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">time.series</span><span class="op">)</span>

<span class="op">}</span></code></pre></div>
<div id="simulate-short-time-frame" class="section level3">
<h3 class="hasAnchor">
<a href="#simulate-short-time-frame" class="anchor"></a>Simulate Short time frame</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seq.length</span> <span class="op">&lt;-</span> <span class="fl">25</span>
<span class="va">cyo.vect.short</span> <span class="op">&lt;-</span> <span class="va">cyo.vect</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">seq.length</span><span class="op">]</span>
<span class="va">years</span> <span class="op">&lt;-</span> <span class="fl">1000</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">diverge_two_sequences</span><span class="op">(</span>seq1 <span class="op">=</span> <span class="va">cyo.vect.short</span>,
                           seq2 <span class="op">=</span> <span class="va">cyo.vect.short</span>,
                                  u <span class="op">=</span> <span class="fl">0.001</span>,
                                  time <span class="op">=</span> <span class="va">years</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot" class="section level3">
<h3 class="hasAnchor">
<a href="#plot" class="anchor"></a>Plot</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">### Plot Cumulative number of mutations</span>


<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>, 
    mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">4</span>,<span class="fl">3</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">tot.mutations.s1</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>,  type <span class="op">=</span> <span class="st">"l"</span>, 
     <span class="co">#ylim = c(0, seq.length*1.01),</span>
       main <span class="op">=</span> <span class="st">"Cummulative number of mutations\n(inc. back mutations)"</span>,
       xlab <span class="op">=</span> <span class="st">"Time"</span>,
       ylab <span class="op">=</span> <span class="st">"Number of mutations"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="va">tot.mutations.s2</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>, col <span class="op">=</span> <span class="fl">2</span>, type <span class="op">=</span> <span class="st">"l"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0</span>, b <span class="op">=</span> <span class="fl">0.001</span><span class="op">*</span><span class="fl">25</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">25</span>, lty <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>,legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"seq1"</span>,<span class="st">"seq2"</span>,<span class="st">"Seq length"</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span>




<span class="co">### Divergence with ancestor</span>

<span class="co">#If we could at least compare with the ancestral sequence.</span>

<span class="co"># Divergence with ancestor can **decrease** due to **back mutations**</span>


<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">vis.mutations.s1</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>,  type <span class="op">=</span> <span class="st">"l"</span>, 
     ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">seq.length</span><span class="op">*</span><span class="fl">1.01</span><span class="op">)</span>,
       main <span class="op">=</span> <span class="st">"Divergence with ancestor"</span>,
       xlab <span class="op">=</span> <span class="st">"Time"</span>,
       ylab <span class="op">=</span> <span class="st">"Divergence"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="va">vis.mutations.s2</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>, col <span class="op">=</span> <span class="fl">2</span>, type <span class="op">=</span> <span class="st">"l"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0</span>, b <span class="op">=</span> <span class="fl">0.001</span><span class="op">*</span><span class="fl">25</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">25</span>, lty <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">25</span><span class="op">*</span><span class="fl">0.75</span>, lty <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>





<span class="co">### Pairwise differences</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">pairwise.diff</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>,  type <span class="op">=</span> <span class="st">"l"</span>, 
     ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">seq.length</span><span class="op">*</span><span class="fl">1.01</span><span class="op">)</span>,
       main <span class="op">=</span> <span class="st">"Current pairwise differences"</span>,
       xlab <span class="op">=</span> <span class="st">"Time"</span>,
       ylab <span class="op">=</span> <span class="st">"pairwise.diff"</span>,
     lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0</span>, b <span class="op">=</span> <span class="fl">0.001</span><span class="op">*</span><span class="fl">25</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">25</span><span class="op">*</span><span class="fl">0.75</span>, lty <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>




<span class="co">### Jukes Cantor Model</span>


<span class="co">## JC distance function</span>


<span class="va">jc</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">{</span>
  <span class="va">K</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fl">3</span><span class="op">/</span><span class="fl">4</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="fl">4</span><span class="op">/</span><span class="fl">3</span><span class="op">*</span><span class="va">p</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">K</span><span class="op">)</span>
<span class="op">}</span>



<span class="co">##Add JC to plot</span>


<span class="va">x</span><span class="op">$</span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">pairwise.diff</span><span class="op">/</span><span class="va">seq.length</span>
<span class="va">x</span><span class="op">$</span><span class="va">JC</span> <span class="op">&lt;-</span> <span class="fu">jc</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">p</span><span class="op">)</span>  
<span class="va">x</span><span class="op">$</span><span class="va">JC.total</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">JC</span><span class="op">*</span><span class="va">seq.length</span>


<span class="va">x</span><span class="op">$</span><span class="va">distance.total</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">tot.mutations.s1</span> <span class="op">+</span> <span class="va">x</span><span class="op">$</span><span class="va">tot.mutations.s2</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">distance.total</span><span class="op">/</span><span class="fl">25</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>,  type <span class="op">=</span> <span class="st">"l"</span>, 
    <span class="co"># ylim = c(0, seq.length*1.01),</span>
    <span class="co"># main = "Cummulative number of mutations\n(inc. back mutations)",</span>
       xlab <span class="op">=</span> <span class="st">"Time"</span>,
       ylab <span class="op">=</span> <span class="st">"Cumulative number of mutations"</span><span class="co">#,</span>
  <span class="co">#  ylim = c(0,3)</span>
    <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="va">p</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>, col <span class="op">=</span> <span class="fl">2</span>, type <span class="op">=</span> <span class="st">"l"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="va">JC</span> <span class="op">~</span> <span class="va">t</span>, data <span class="op">=</span> <span class="va">x</span>, col <span class="op">=</span> <span class="fl">3</span>, type <span class="op">=</span> <span class="st">"l"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>,legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Real dist"</span>,<span class="st">"p dist"</span>,<span class="st">"JC"</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="lll-jc_sim_for_render_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
