<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Displaying Data - boxplot() &amp; hist() • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Displaying Data - boxplot() &amp; hist()">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Displaying Data - boxplot() &amp; hist()</h1>
            
      
      
      <div class="hidden name"><code>plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="lab-2-displaying-describing-data" class="section level1">
<h1 class="hasAnchor">
<a href="#lab-2-displaying-describing-data" class="anchor"></a>Lab 2: Displaying &amp; Describing Data</h1>
<div id="lab-2-outline" class="section level2">
<h2 class="hasAnchor">
<a href="#lab-2-outline" class="anchor"></a>Lab 2 outline</h2>
<ul>
<li>Intro</li>
<li>The Iris data</li>
<li>Loading data: iris
<ul>
<li>command: data(iris)</li>
</ul>
</li>
<li>Loading packages in base R: MASS
<ul>
<li>command: library(MASS)</li>
</ul>
</li>
<li>Loading packages from CRAN: doBy</li>
<li>Boxplots w/ iris data
<ul>
<li>anatomy of an R function:boxplot(…)</li>
<li>plotting arguments: xlab=, ylab=</li>
<li>arguement: cex =</li>
<li>arguement: lw =</li>
</ul>
</li>
<li>Histograms w/Iris data
<ul>
<li>coomand: hist(…)</li>
</ul>
</li>
<li>Scatterplots w/iris data</li>
</ul>
<p>Based generally on Chapters 2 &amp; 3, Whitlock and Schulter, 2nd Ed</p>
</div>
</div>
<div id="example-data-for-lab-2-fishers-irises" class="section level1">
<h1 class="hasAnchor">
<a href="#example-data-for-lab-2-fishers-irises" class="anchor"></a>Example data for Lab 2: Fisher’s Irises</h1>
<div id="dataset-made-popular-by-r-a--fisher" class="section level2">
<h2 class="hasAnchor">
<a href="#dataset-made-popular-by-r-a--fisher" class="anchor"></a>Dataset made popular by R.A. Fisher</h2>
</div>
<div id="frequently-used-to-explaintest-stats-procedures" class="section level2">
<h2 class="hasAnchor">
<a href="#frequently-used-to-explaintest-stats-procedures" class="anchor"></a>Frequently used to explain/test stats procedures</h2>
<div class="figure">
<img src="../../../../Box%20Sync/0_R/compbio4all/vignettes/irises.png" alt=""><p class="caption">Fisher’s irises</p>
</div>
<p>See <a href="https://en.wikipedia.org/wiki/Iris_flower_data_set" class="uri">https://en.wikipedia.org/wiki/Iris_flower_data_set</a> for more info.</p>
</div>
</div>
<div id="loading-data-into-r-the-easy-way-pre-made-data-in-an-r-package" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-data-into-r-the-easy-way-pre-made-data-in-an-r-package" class="anchor"></a>Loading data into R the easy way: pre-made data in an R “Package”</h1>
<div id="getting-data-into-r-or-sas-or-arcgis-can-be-a-pain" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-data-into-r-or-sas-or-arcgis-can-be-a-pain" class="anchor"></a>Getting data into R (or SAS, or ArcGIS…) can be a pain</h2>
</div>
<div id="r-comes-with-many-datasets-that-are-pre-loaded-into-it" class="section level2">
<h2 class="hasAnchor">
<a href="#r-comes-with-many-datasets-that-are-pre-loaded-into-it" class="anchor"></a>R comes with many datasets that are pre-loaded into it</h2>
</div>
<div id="there-are-also-many-stat--techniques-that-can-easily-be-added-to-r" class="section level2">
<h2 class="hasAnchor">
<a href="#there-are-also-many-stat--techniques-that-can-easily-be-added-to-r" class="anchor"></a>There are also many stat. techniques that can easily be added to R</h2>
</div>
<div id="these-are-contained-in-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#these-are-contained-in-packages" class="anchor"></a>These are contained in “packages”</h2>
</div>
</div>
<div id="load-data-that-is-already-in-the-base-distribution-of-r" class="section level1">
<h1 class="hasAnchor">
<a href="#load-data-that-is-already-in-the-base-distribution-of-r" class="anchor"></a>Load data that is already in the “base” distribution of R</h1>
<p>Fisher’s iris data comes automaically with R. You can load it into R’s memory using the command “data()”</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Load the iris data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span></code></pre></div>
<div id="look-at-the-iris-data" class="section level2">
<h2 class="hasAnchor">
<a href="#look-at-the-iris-data" class="anchor"></a>Look at the iris data</h2>
<p>We’ll look at the iris data using some commands like ls(), dim(), and names().</p>
<p>You can check that it was loaded using the <strong>ls()</strong> command (“list”).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] "iris"</span></code></pre></div>
<p>You can get info about the nature of the daframe using commands like <strong>dim()</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt; [1] 150   5</span></code></pre></div>
<p>This tells us that the iris data is essentially a spreadhshett that has 150 rows and 5 columns.</p>
<p>We can get the column names with names()</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt; [1] "Sepal.Length" "Sepal.Width"  "Petal.Length" "Petal.Width"  "Species"</span></code></pre></div>
<p>Note that the first letter of each word is capitalized. What are the implications of this?</p>
<p>The top of the data and the bottom of the data can be checked with head() and tail() commands</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#top of dataframe</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt;   Sepal.Length Sepal.Width Petal.Length Petal.Width Species</span>
<span class="co">#&gt; 1          5.1         3.5          1.4         0.2  setosa</span>
<span class="co">#&gt; 2          4.9         3.0          1.4         0.2  setosa</span>
<span class="co">#&gt; 3          4.7         3.2          1.3         0.2  setosa</span>
<span class="co">#&gt; 4          4.6         3.1          1.5         0.2  setosa</span>
<span class="co">#&gt; 5          5.0         3.6          1.4         0.2  setosa</span>
<span class="co">#&gt; 6          5.4         3.9          1.7         0.4  setosa</span>

<span class="co">#bottom of dataframe</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt;     Sepal.Length Sepal.Width Petal.Length Petal.Width   Species</span>
<span class="co">#&gt; 145          6.7         3.3          5.7         2.5 virginica</span>
<span class="co">#&gt; 146          6.7         3.0          5.2         2.3 virginica</span>
<span class="co">#&gt; 147          6.3         2.5          5.0         1.9 virginica</span>
<span class="co">#&gt; 148          6.5         3.0          5.2         2.0 virginica</span>
<span class="co">#&gt; 149          6.2         3.4          5.4         2.3 virginica</span>
<span class="co">#&gt; 150          5.9         3.0          5.1         1.8 virginica</span></code></pre></div>
<p>Another common R command is <strong>is()</strong>, which tells you what something is in R land.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">is</span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt; [1] "data.frame" "list"       "oldClass"   "vector"</span></code></pre></div>
<p>R might spew a lot of things out at you when you use <strong>is()</strong>; usually the first item is most important. Here, it tells us that the “object” called “iris” in your workspace is 1st and foremost a “data.frame”, which is esseentially a spreadsheet of data loaded into R.</p>
<p>You can get basic info about the data themseleves using commands like <strong>summary()</strong>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt;   Sepal.Length    Sepal.Width     Petal.Length    Petal.Width   </span>
<span class="co">#&gt;  Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  </span>
<span class="co">#&gt;  1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  </span>
<span class="co">#&gt;  Median :5.800   Median :3.000   Median :4.350   Median :1.300  </span>
<span class="co">#&gt;  Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  </span>
<span class="co">#&gt;  3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  </span>
<span class="co">#&gt;  Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  </span>
<span class="co">#&gt;        Species  </span>
<span class="co">#&gt;  setosa    :50  </span>
<span class="co">#&gt;  versicolor:50  </span>
<span class="co">#&gt;  virginica :50  </span>
<span class="co">#&gt;                 </span>
<span class="co">#&gt;                 </span>
<span class="co">#&gt; </span></code></pre></div>
<p>If you wanted information on just a single column, you would tell R to isolate that column like this, using a dollar sign ($).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Sepal.Width</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;   2.000   2.800   3.000   3.057   3.300   4.400</span></code></pre></div>
<p>That is, that name of the dataframe, a dollar sign ($), and the name of the column.</p>
<p>What happens when you don’t capitalize something? Try this intentioanl mistakes:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#all lower case</span>
<span class="co"># summary(iris$sepal.width) # this won't work</span>

<span class="co">#just "s" in "sepal" lower case</span>
<span class="co"># summary(iris$sepal.Width)  #this won't work either</span>

<span class="co">#or what if you capitalize "i" in "Iris"?</span>
<span class="co"># summary(Iris$Sepal.Width) #won't work either</span></code></pre></div>
<p>The 1st two error messages are not very informative; the 3rd one (“Error in summary(Iris$Sepal.Width) : object ‘Iris’ not found”) does make a little sense.</p>
</div>
<div id="load-data-that-is-in-another-r-package-part-1" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data-that-is-in-another-r-package-part-1" class="anchor"></a>Load data that is in another R package part 1</h2>
<div id="packages-that-come-w-r" class="section level3">
<h3 class="hasAnchor">
<a href="#packages-that-come-w-r" class="anchor"></a>Packages that come w/ R</h3>
<p>Many scientists develop software for R, and they often include datasets to demonstrate how the software works. Some of this software, called a “package” comes with R already and just needs to be loaded. This is done with the <strong>library()</strong> command.</p>
<p>The <strong>MASS</strong> package comes with R when you download it and has many useful functions and interesting datasets.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Load the MASS package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.stats.ox.ac.uk/pub/MASS4/">MASS</a></span><span class="op">)</span></code></pre></div>
<p>MASS contains a dataset called called “mammals”</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">mammals</span><span class="op">)</span></code></pre></div>
<p>You can confirm that the mammals data is in your workspace using <strong>ls()</strong></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] "iris"    "mammals"</span></code></pre></div>
<p>You should now have the iris and the mammals data in your R “workspace.”"</p>
<p>What is in the mammals dataset? Datasets actually usually have useful help files. Access help using the <strong>?</strong> function.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">mammals</span></code></pre></div>
<p>The help screen you pop up. It tells us that mammals is</p>
<blockquote>
<p>“A data frame with average brain and body weights for 62 species of land mammals.”</p>
</blockquote>
<p>Since this is someone elses data, the authors of the MASS package need to provide proper citation. At the bottom we can see that these data come from the paper:</p>
<blockquote>
<p>Allison, T. and Cicchetti, D. V. (1976) Sleep in mammals: ecological and constitutional correlates. Science 194: 732-734.</p>
</blockquote>
<p>We can learn about the mammals data usig the usual commands</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">mammals</span><span class="op">)</span>
<span class="co">#&gt; [1] 62  2</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">mammals</span><span class="op">)</span>
<span class="co">#&gt; [1] "body"  "brain"</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mammals</span><span class="op">)</span>
<span class="co">#&gt;                    body brain</span>
<span class="co">#&gt; Arctic fox        3.385  44.5</span>
<span class="co">#&gt; Owl monkey        0.480  15.5</span>
<span class="co">#&gt; Mountain beaver   1.350   8.1</span>
<span class="co">#&gt; Cow             465.000 423.0</span>
<span class="co">#&gt; Grey wolf        36.330 119.5</span>
<span class="co">#&gt; Goat             27.660 115.0</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">mammals</span><span class="op">)</span>
<span class="co">#&gt;                    body brain</span>
<span class="co">#&gt; Echidna           3.000  25.0</span>
<span class="co">#&gt; Brazilian tapir 160.000 169.0</span>
<span class="co">#&gt; Tenrec            0.900   2.6</span>
<span class="co">#&gt; Phalanger         1.620  11.4</span>
<span class="co">#&gt; Tree shrew        0.104   2.5</span>
<span class="co">#&gt; Red fox           4.235  50.4</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mammals</span><span class="op">)</span>
<span class="co">#&gt;       body              brain        </span>
<span class="co">#&gt;  Min.   :   0.005   Min.   :   0.14  </span>
<span class="co">#&gt;  1st Qu.:   0.600   1st Qu.:   4.25  </span>
<span class="co">#&gt;  Median :   3.342   Median :  17.25  </span>
<span class="co">#&gt;  Mean   : 198.790   Mean   : 283.13  </span>
<span class="co">#&gt;  3rd Qu.:  48.202   3rd Qu.: 166.00  </span>
<span class="co">#&gt;  Max.   :6654.000   Max.   :5712.00</span></code></pre></div>
</div>
</div>
<div id="load-data-that-is-in-another-r-package-part-2" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data-that-is-in-another-r-package-part-2" class="anchor"></a>Load data that is in another R package part 2</h2>
<div id="packages-from-the-cran-website" class="section level3">
<h3 class="hasAnchor">
<a href="#packages-from-the-cran-website" class="anchor"></a>Packages from the CRAN website</h3>
<p>Most packages don’t come with R when you download it but are stored in a central site called CRAN. We’ll load the <strong>doBy</strong> package.</p>
</div>
<div id="loading-packages-using-r-studio" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-packages-using-r-studio" class="anchor"></a>Loading packages using R-Studio</h3>
<p>RStudio makes it easy to find and load packages. Follow these instructions. A visual walkthrough is also avaialbel on D2L</p>
<ul>
<li>In the panel of RStudio that has the tabs “Plots”, “Packages”,“Help”, “Viewer” click on “Packages”"</li>
<li>On the next line it says “Install” and “Update”. Click on “Install”</li>
<li>A window will pop up. In the white field in the middle of the window under “Packages” type the name of the package you want.</li>
<li>RStudio will automatically bring up potential packages as you type.</li>
<li>Finish typing “doBy” or click on the name.</li>
<li>Click on the “Install” button.</li>
<li>In the source vieewr some misc. test should show up. Most of the time this works. If it doesn’t, talk to the professor!</li>
</ul>
<p>If an R packages doesn’t load properly, it could be for several reasons.<br>
* 1st, your internet connection might be having problems.<br>
* 2nd, the website where the package is stored might be down for mainteance.<br>
* 3rd, the version of are you are using is probably newer than the version of R used to make the package. This is a real pain - aks for help from an expert R user if think you have this problem.</p>
</div>
</div>
<div id="loading-packages-using-code" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-packages-using-code" class="anchor"></a>Loading packages using code</h2>
<p>You can also use the <strong>install.packages()</strong> command to try to load the package. I already have the <strong>doBy</strong> packaged downloaded to my computer so I have “commented out” the code with a “#”. To run the code, remove the “#”. If you already followed the instructions above you don’t need to run the code.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#install.packages("doBy")</span></code></pre></div>
</div>
</div>
<div id="what-if-you-tell-r-to-install-a-package-you-already-have-downloaded" class="section level1">
<h1 class="hasAnchor">
<a href="#what-if-you-tell-r-to-install-a-package-you-already-have-downloaded" class="anchor"></a>What if you tell R to install a package you already have downloaded?</h1>
<p>If you already have the package downloaded to your computer then a window will pop up asking you if you want to restart your computer. Normally this isn’t necessary; just click “no”. You might see a “warning” message pop up in the console such as “Warning in install.packages: package ‘doBy’ is in use and will not be installed”. This isn’t a problem for basic R work. If you are doing serious work (e.g. for a publication) you should restart R.</p>
</div>
<div id="what-if-i-cant-get-a-package-i-need-loaded" class="section level1">
<h1 class="hasAnchor">
<a href="#what-if-i-cant-get-a-package-i-need-loaded" class="anchor"></a>What if I can’t get a package I need loaded?</h1>
<ul>
<li>Talk to someone who is good w/R (eg, your professor)</li>
<li>Google something like “how to install R package” for general info</li>
<li>Google something like “problem loading R package”</li>
<li>Copy and paste any error message you might be getting into Google and see if anyone has written about this problem</li>
</ul>
<p>See above for reasons why a package might not load properly the 1st time you try.</p>
</div>
<div id="boxplots" class="section level1">
<h1 class="hasAnchor">
<a href="#boxplots" class="anchor"></a>Boxplots</h1>
<p>Now we’re going to finally plot some data using the <strong>boxplot()</strong> command</p>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
</div>
<div id="r-function-boxplot" class="section level1">
<h1 class="hasAnchor">
<a href="#r-function-boxplot" class="anchor"></a>R function: boxplot(…)</h1>
<div id="its-very-very-easy-to-make-boxplots-in-r" class="section level2">
<h2 class="hasAnchor">
<a href="#its-very-very-easy-to-make-boxplots-in-r" class="anchor"></a>Its Very very easy to make boxplots in R</h2>
</div>
<div id="very-very-very-very-hard-to-make-in-excel" class="section level2">
<h2 class="hasAnchor">
<a href="#very-very-very-very-hard-to-make-in-excel" class="anchor"></a>Very very very very hard to make in Excel</h2>
</div>
<div id="you-should-be-making-these-all-the-time-to-explore-your-data" class="section level2">
<h2 class="hasAnchor">
<a href="#you-should-be-making-these-all-the-time-to-explore-your-data" class="anchor"></a>You should be making these all the time to explore your data</h2>
</div>
<div id="also-good-way-to-present-data-in-a-final-paper" class="section level2">
<h2 class="hasAnchor">
<a href="#also-good-way-to-present-data-in-a-final-paper" class="anchor"></a>Also good way to present data in a final paper</h2>
<p>Making a basic plot is faily easy. This basic formulas-like structure is used frequently in R so we will discuss it in some detial.</p>
</div>
</div>
<div id="anatomy-of-an-r-command" class="section level1">
<h1 class="hasAnchor">
<a href="#anatomy-of-an-r-command" class="anchor"></a>Anatomy of an R command</h1>
</div>
<div id="key-features-of-the-command-are" class="section level1">
<h1 class="hasAnchor">
<a href="#key-features-of-the-command-are" class="anchor"></a>Key features of the command are</h1>
<div id="the" class="section level2">
<h2 class="hasAnchor">
<a href="#the" class="anchor"></a>the “~”"</h2>
</div>
<div id="the-comma" class="section level2">
<h2 class="hasAnchor">
<a href="#the-comma" class="anchor"></a>the comma</h2>
</div>
<div id="the-data-iris" class="section level2">
<h2 class="hasAnchor">
<a href="#the-data-iris" class="anchor"></a>the “data = iris”</h2>
</div>
<div id="capitalization" class="section level2">
<h2 class="hasAnchor">
<a href="#capitalization" class="anchor"></a>capitalization</h2>
<div class="figure">
<img src="../../../../Box%20Sync/0_R/compbio4all/vignettes/anatomy_of_an_R_function_1_basic_boxplot.png" alt=""><p class="caption">Anatomy of the boxplot command</p>
</div>
</div>
</div>
<div id="box-plot-with-labels" class="section level1">
<h1 class="hasAnchor">
<a href="#box-plot-with-labels" class="anchor"></a>Box plot with labels</h1>
<p>R will usually generate labels for the x and y axes based on the command. These need to be changed.</p>
<p>The arguement <strong>“xlab =”</strong> sets the labels for the x-axis, <strong>“ylab”</strong> for the y axis.</p>
<p>Note that these both occur inside the paretheses, and that the text for the labels goes in quotes (ie, “Iris species”). Forgetting the quotes will cause the code to fail. Note that units (mm) are included for the y axis.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#boxplot w/ axes labels</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length (mm)"</span> <span class="op">)</span> <span class="co">#Label for y axis, w/ units</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
</div>
<div id="changing-colors-in-r-plots-part-1" class="section level1">
<h1 class="hasAnchor">
<a href="#changing-colors-in-r-plots-part-1" class="anchor"></a>Changing colors in R plots part 1</h1>
<p>If we wanted we could change the color of the boxplots using the universal argument <strong>“col =”</strong>. This code can be used to change the color of most types of plots in R.<br>
This doesn’t increase the information content of the figure but maybe makes it nicer to look at.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#boxplot w/color changes</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length"</span>,
        col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> </code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
</div>
<div id="changing-colors-in-r-plots---part-2" class="section level1">
<h1 class="hasAnchor">
<a href="#changing-colors-in-r-plots---part-2" class="anchor"></a>Changing colors in R plots - part 2</h1>
<p>We can set the color of each box to somethingdifferent. This accents that each box is a differnet Iris species. The code <strong>“col = 2:4”</strong> tells R to use the sequence of colors “2, 3, 4”. (I skip “1” b/c its black and will obscure the bar that indicates the median).</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#boxplot w/ each color different</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length"</span>,
        col <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span> </code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
</div>
<div id="changing-colors-in-r-plots---part-3" class="section level1">
<h1 class="hasAnchor">
<a href="#changing-colors-in-r-plots---part-3" class="anchor"></a>Changing colors in R plots - part 3</h1>
<p>This next code uses a slight variant of the argument ** col = …* (I’ve reversed the order of the colors, just so you can see the difference.) I’ve used <strong>“col = c(4,3,2)”</strong> which is just longhand for “col = 4:2”. As is common in R, there is more than 1 way to do the exact same thing.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#col arguement w/explicit vector of colors</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length"</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">3</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
</div>
<div id="changing-colors-in-r-plots-part---4" class="section level1">
<h1 class="hasAnchor">
<a href="#changing-colors-in-r-plots-part---4" class="anchor"></a>Changing colors in R plots part - 4</h1>
<p>We could also do this, naming the colors explicitly.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#col arguement w/character vector of colors</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length"</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-26-1.png" width="700"></p>
</div>
<div id="more-on-the-c-function" class="section level1">
<h1 class="hasAnchor">
<a href="#more-on-the-c-function" class="anchor"></a>More on The “c(…)” Function</h1>
<p>This stands for “concatenate” ## This shows up a lot in R code. ## can have numbers inside it, separated by a colon ### c(2:4) ## can have numbers in it, sep. by commas ### c(4,3,2) ## can have text in it, sep by commas AND w/ " " around each word ### c(“red”,“green”,“blue”)</p>
<div id="tweaking-plots-the-cex-arguement" class="section level3">
<h3 class="hasAnchor">
<a href="#tweaking-plots-the-cex-arguement" class="anchor"></a>Tweaking plots: the cex = arguement</h3>
<p>Run the code below, which includes the new arguement <strong>“cex =2”</strong> and see what happens. Can you see what changed?</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#boxplot w/cex = 2</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length"</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span><span class="op">)</span>,
        cex <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>  <span class="co">#cex set to 2</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
</div>
<div id="cex-increases-the-size-of-elements-within-plots-usually-points" class="section level3">
<h3 class="hasAnchor">
<a href="#cex-increases-the-size-of-elements-within-plots-usually-points" class="anchor"></a>“cex =” increases the size of elements within plots (usually points)</h3>
<p>It might be hard to see, but <strong>“cex”</strong> stands for “character expansions” and it has “expanded” the size of the outlier dot in the <em>virginica</em> data on the far right of the plot.</p>
</div>
<div id="tweaking-plots-lwd-arguement" class="section level3">
<h3 class="hasAnchor">
<a href="#tweaking-plots-lwd-arguement" class="anchor"></a>Tweaking plots: lwd arguement</h3>
<p>Run the code below, which includes the new command “cex =2” AND “lwd = 2”. This stands for “line width” and essentially makes all of the lines in the plot “bold”, including the line for the outlier circle.</p>
<p>This is often a good thing to do to make plot elements easier to see (especially if you are showing them on Powerpoint!)</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#cex and lwd</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Species</span>,    <span class="co">#the plotting formula of y ~ x</span>
        data <span class="op">=</span> <span class="va">iris</span>,           <span class="co">#the data to plot</span>
        xlab <span class="op">=</span> <span class="st">"Iris Species"</span>, <span class="co">#Label for x axis</span>
        ylab <span class="op">=</span> <span class="st">"Sepal Length"</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span><span class="op">)</span>,
        lwd <span class="op">=</span> <span class="fl">2</span>,
        cex <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>  <span class="co">#cex set to 2</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
</div>
</div>
<div id="histograms" class="section level1">
<h1 class="hasAnchor">
<a href="#histograms" class="anchor"></a>Histograms</h1>
<div id="made-with-hist-command" class="section level2">
<h2 class="hasAnchor">
<a href="#made-with-hist-command" class="anchor"></a>made with hist(…) command</h2>
</div>
<div id="very-very-easy-to-make-in-r" class="section level2">
<h2 class="hasAnchor">
<a href="#very-very-easy-to-make-in-r" class="anchor"></a>very very easy to make in R</h2>
</div>
<div id="very-very-very-hard-to-make-in-excel" class="section level2">
<h2 class="hasAnchor">
<a href="#very-very-very-hard-to-make-in-excel" class="anchor"></a>very very very hard to make in Excel</h2>
</div>
<div id="you-should-make-them-all-the-time-for-you-data" class="section level2">
<h2 class="hasAnchor">
<a href="#you-should-make-them-all-the-time-for-you-data" class="anchor"></a>You should make them all the time for you data!</h2>
<p>This code makes a histogram of one of the iris species’ Petal.Length.</p>
<p>Note that the <strong>$</strong> appears between the data, iris, and the column of data you want to plot, Petal.Length. NOte that the “P” and “L” have to be capitalized.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-29-1.png" width="700"></p>
</div>
</div>
<div id="modifying-histograms-titles-with-the-main-argument" class="section level1">
<h1 class="hasAnchor">
<a href="#modifying-histograms-titles-with-the-main-argument" class="anchor"></a>Modifying histograms: titles with the main = argument</h1>
<p>For some reason the hist() command automatically puts a title above the plot. Titles are good for your own use but acutally are almost never used in figures published in papers and books. We can get rid of the title using the arguement “main =”" “. This is telling R to turn the”main title" into nothing.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span>, main <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
</div>
<div id="modifying-histograms-cex" class="section level1">
<h1 class="hasAnchor">
<a href="#modifying-histograms-cex" class="anchor"></a>Modifying histograms: cex</h1>
<p>What happens when we add <strong>cex</strong> to the histogram?…</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#histogram with cex=2</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span>, main <span class="op">=</span> <span class="st">""</span>, cex <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-31-1.png" width="700"></p>
<p>…nothing.</p>
<p>Note that R does nothing, and doesn’t tell you anything. The <strong>cex = </strong> arguement does not apply to histograms, but R will still run the code.</p>
</div>
<div id="modifying-histograms-lwd" class="section level1">
<h1 class="hasAnchor">
<a href="#modifying-histograms-lwd" class="anchor"></a>Modifying histograms: lwd</h1>
<p>What about lwd = … ?</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#hist() with lwd =</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span>, main <span class="op">=</span> <span class="st">""</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-32-1.png" width="700"></p>
<p>For some reason this only makes the axes of the histogram thicker, not the boxes themselves. This is a quirk of R.</p>
<p>It makes sense (to me) that if “lwd = 2” made the boxes in a boxplot bigger, it would make the bars in a histogram bigger.</p>
</div>
<div id="modifying-plots-par" class="section level1">
<h1 class="hasAnchor">
<a href="#modifying-plots-par" class="anchor"></a>Modifying plots: par()</h1>
<p>The way to make the lines that make up a boxplot thicker requires another command, called, <strong>par()</strong> that is run <em>separately</em> from <strong>hist()</strong>. I don’t want to get into the details right now, but this code will make the bars bigger in the boxplot. Note that all your plots from now on will <em>also</em> have thicker bars. This is b/c <strong>par</strong> sets a “global” feature of your R session.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#par to change hist line widths</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span>, main <span class="op">=</span> <span class="st">""</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-33-1.png" width="700"></p>
</div>
<div id="finding-r-help-with-google" class="section level1">
<h1 class="hasAnchor">
<a href="#finding-r-help-with-google" class="anchor"></a>Finding R help with Google</h1>
<p>There’s lots of info about R on the web, and if you have a problem, then someone else has probably had it before and perhaps written somethign about it.</p>
<div id="google-this-r-histogram-make-boxes-thicker" class="section level2">
<h2 class="hasAnchor">
<a href="#google-this-r-histogram-make-boxes-thicker" class="anchor"></a>Google this: “R histogram make boxes thicker”</h2>
<p>In my browser the 1st thing to come up is</p>
<blockquote>
<p>R - histogram rectangles line thickness - Stack Overflow <a href="http://stackoverflow.com/questions/24715055/r-histogram-rectangles-line-thickness" class="uri">http://stackoverflow.com/questions/24715055/r-histogram-rectangles-line-thickness</a></p>
</blockquote>
<p>The website <strong>stackoverflow.com</strong> has lots of info about R. However, many people who use it are hard-core programmers, who can come accross as jerks sometimes when they answer questions if you don’t follow the rules and protocols of stackoverflow.</p>
<p>This post on histograms happens to provide a clue as to how to use <strong>par()</strong> to fix our problem, though its not 110% straightforward</p>
</div>
</div>
<div id="modifying-histograms-color" class="section level1">
<h1 class="hasAnchor">
<a href="#modifying-histograms-color" class="anchor"></a>Modifying histograms: color</h1>
<p>We can change the color of the fill of the histogram</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span>,
     main <span class="op">=</span> <span class="st">""</span>, 
     lwd <span class="op">=</span> <span class="fl">2</span>, 
     col <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-34-1.png" width="700"></p>
</div>
<div id="modifying-histograms-axis-labels" class="section level1">
<h1 class="hasAnchor">
<a href="#modifying-histograms-axis-labels" class="anchor"></a>Modifying histograms: axis labels</h1>
<p>And we can add an x axis label. (The y axis is already labeled)</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">iris</span><span class="op">$</span><span class="va">Petal.Length</span>, 
     main <span class="op">=</span> <span class="st">""</span>, 
     lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">3</span>,
     xlab <span class="op">=</span> <span class="st">"Petal Length"</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-35-1.png" width="700"></p>
</div>
<div id="scatter-plots-2-continuous-variables" class="section level1">
<h1 class="hasAnchor">
<a href="#scatter-plots-2-continuous-variables" class="anchor"></a>Scatter plots: 2 continuous variables</h1>
<p>Let’s make a scatter plot, where we plot two continous, numeric variables against each other (both x and y variables are numbers; not categories)</p>
<p>I’ve forgotten the names of all the iris variables, so I’ll use the <strong>names()</strong> command to see what they are</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>
<span class="co">#&gt; [1] "Sepal.Length" "Sepal.Width"  "Petal.Length" "Petal.Width"  "Species"</span></code></pre></div>
<p>I’ll plot the sepals against the petals</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Petal.Length</span>, data <span class="op">=</span> <span class="va">iris</span><span class="op">)</span></code></pre></div>
<p><img src="plot-displaying_data_box_hist_base_R-NEEDS-IMAGES_files/figure-html/unnamed-chunk-37-1.png" width="700"></p>
</div>
<div id="exercise-modify-code-to-do-the-following-things" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise-modify-code-to-do-the-following-things" class="anchor"></a>Exercise: Modify code to do the following things</h1>
<div id="plot-different-variables-in-the-iris-data-against-each-other" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-different-variables-in-the-iris-data-against-each-other" class="anchor"></a>Plot different variables in the iris data against each other</h2>
</div>
<div id="change-the-axes-labels" class="section level2">
<h2 class="hasAnchor">
<a href="#change-the-axes-labels" class="anchor"></a>Change the axes labels</h2>
</div>
<div id="change-the-color-of-the-points-to-something-else-" class="section level2">
<h2 class="hasAnchor">
<a href="#change-the-color-of-the-points-to-something-else-" class="anchor"></a>Change the color of the points to something else.</h2>
<p>Remember to have the proper capitalization! Personally, I find the that uppercase and lowercase Ws in R look similar (W vs w).</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
