<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Types of data • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Types of data">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sss-nlb-types_of_data_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Types of data</h1>
            
      
      
      <div class="hidden name"><code>sss-nlb-types_of_data.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: ggplot2</span></code></pre></div>
<p>Data if the lifeblood of science. Science is science because it is based on empirical observations about the natural world. There’s lots of other essential components - hypotheses, theories, models, math - but empirical observations are either the input for these other activities or the final arbitrater of their value.</p>
<p>We therefore use the word "data’ a lot in science, and use it as adjective or adverb for many scientidic activities. We do <strong>data collection</strong> in our labs, record data in lab notebooks or <strong>data sheets</strong>, make graphs to do <strong>data visualization</strong>, use stats to do <strong>data analysis</strong>, and we read pulished paper to evaluate other scientists’ data.</p>
<p>Data, however, can take on many different forms. Just as “data” can be used as an adjective or adverb to describe many scientific activites, there are many adjecives which specify exactly the nature of the data we are working with.</p>
<p>For example, you might want to know if a particular allele (genetic variant) occurs in a strain of <strong>E. coli</strong>. You therefore do PCR with primers specific to that allele and run out the reaction on a gel. The presence or absence of a DNA band on the gel is a type of <strong>qualitative data</strong>: evidence for whether the strain possess a certain quality.</p>
<p>[transition]</p>
<p>Frequently we carry out repeated observations or measurements, such as recording how quickly an object falls to calculate accelaration due to gravity, how frequently UV radiation prevents the growth of bacteria, or rating how much damage an herbivore caused to a leaf on a scale from 1 to 10. These <strong>numeric data</strong> are usually summarized mathematically using means and percentages. If you’re being thorough - and you should – you should characterize variation between repeated observations using standard deviations (SD), standard errors (SE), or confidence intervals (CI).</p>
<p>For example, you could measure the caloric energy in 10 samples and produce a plot like this</p>
<pre><code>#&gt; `stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="sss-nlb-types_of_data_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div id="weapons-of-math-destruction" class="section level3">
<h3 class="hasAnchor">
<a href="#weapons-of-math-destruction" class="anchor"></a>Weapons of math destruction</h3>
<p>Something that is not always appreciated is that once you start applying math like means and standard deviations to data you need to be mindful of exactly what type of data you are working. Indeed, applying the wrong kind of mathematical operation to specific kinds of data can result in nonsensical output or even erroneous conclusion. This is most usually a problem when percentages are involved or rating scales. In the next section I’ll work through an extended example to show you the kind of problem that can come up. After that we’ll talk about different kinds of data and the right kind of math to deal with them.</p>
</div>
<div id="example-survival-rates-of-bacteria" class="section level3">
<h3 class="hasAnchor">
<a href="#example-survival-rates-of-bacteria" class="anchor"></a>Example: survival rates of bacteria</h3>
<p>Here’s an example of how math can run amok. Let’s say I want to know how frequently a strain of bacteria is killed by UV radiation. I have 3 petri dishes and I dilute out a broth culture of bacteria so that 10 cells end up on each plate. I then put the plates into an incubator the contains a UV light. When I check back on the plates I get the following results:</p>
<p>Plate 1: 2 of the 10 cells grew into colonies Plate 2: 0 of the 10 cells grew into colonies Plate 3: 0 of the 10 cells grew into a colony</p>
<p>So I have 3 data points: 20% survival, 0% survival, and 0% survival. I can calcualte the mean survival rate as</p>
<p>(20%+0%+0%)/3 = 20%/3 = 6.67%</p>
<p>Plugging these numbers into a spreadsheet we can quickly calculate a standard deviation, for example using Excel’s STDEV.S() function like this:</p>
<pre><code>#&gt; [1] 6.666667
#&gt; [1] 11.54701</code></pre>
<table class="table">
<thead><tr class="header">
<th align="right"></th>
<th align="left">A</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left"><strong>bacteria.survival</strong></td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="left">20</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="left">0</td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="left">=STDEV.S(A2:A4</td>
</tr>
</tbody>
</table>
<p>This tells use that the standard deviation of the data is 11.5. You may have learned that a general rule of thumb is that we should expect abou 60% of our data points to fall within plus or minus 1 SD of our mean.</p>
<p>Its always good to plot data; ideally you want to plot the raw data and your mean, and also the standard deviation as error bars around the mean. For our study we get something like this</p>
<pre><code>#&gt; `stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="sss-nlb-types_of_data_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>This plot has 3 elements related to the data:</p>
<ol style="list-style-type: decimal">
<li>The 3 small dots are the 3 data points</li>
<li>The big point is the mean</li>
<li>The error bars extend plus 1 SD from the SD and minus 1 SD from the mean</li>
</ol>
<p>Look carefully at this graph; there’s something peculiar. Can you spot it?</p>
<p>Let’s think about what we plotted. The data points are what they are: 20%, 0% and 0%. Nothing wrong with that. The mean looks ok too; its supposed to represent the “Central tendency” of the data or your best guess about what should happen if you repeated the experiment. For example, if you plated 100 bactiera you expect 6.67% to survive, so about 6 or 7.</p>
<p>What about the error bars? There’s actually something wrong there. Recall that about 60% of our data should between +1SD and -1SD. Another way of thinking about this is that if we ran the experiment many time, we’d expect 60% of data points to be within those error bars, and only 40% higher or lower. Given what error bars represent, there’s something amiss with ours. Let’s add something to the graph to see if you can spot it:</p>
<pre><code>#&gt; `stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="sss-nlb-types_of_data_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>I’ve added a line at 0. Two of our data points are 0%; 0 is a perfectly valid data point; it could even be the mean if all of our data were 0%. Notice, however, that the error bar extends <em>beyond</em> 0. It actually goes down to about -5%. Recall what I just said about standard deviations: if you repeated your experiment many times (say 100) you’d expect 60% to be plotted within those error bars, and 40% to be higher or lower. What our error bars are implying is that data points could fall between 0 and -5%, and some may even be less than that!</p>
<p>In a picture, this is what this would look like, where each dot is one of 100 imaginary replications of our study.</p>
<pre><code>#&gt; `stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="sss-nlb-types_of_data_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>Wow! What’s implied by our initial graph is that many experiments could produce negative values, sometimes with “survival” of -20%!</p>
<p>Our simple math of the standard deviation really went afoul when we plugged percentages into it. The solution to this problem is to use a different formulation of the standard deviation that is appropriate for this type of frequency data. We won’t worry about the math, but if we do things correctly we should get error bars which almost go to zero (2%) and still extend up to about 20%.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">surv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.2</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>
<span class="va">trials</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">10</span>,<span class="fl">10</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span><span class="va">surv</span> <span class="op">~</span> <span class="fl">1</span>, weights <span class="op">=</span> <span class="va">trials</span>, family <span class="op">=</span> <span class="st">"binomial"</span><span class="op">)</span><span class="op">)</span>

<span class="fu">arm</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/arm/man/invlogit.html">invlogit</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2.6391</span><span class="op">)</span>
<span class="fu">arm</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/arm/man/invlogit.html">invlogit</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2.6391</span><span class="op">+</span><span class="fl">0.7319</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>
<span class="fu">arm</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/arm/man/invlogit.html">invlogit</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2.6391</span><span class="op">-</span><span class="fl">0.7319</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span></code></pre></div>
</div>
<div id="types-of-data" class="section level3">
<h3 class="hasAnchor">
<a href="#types-of-data" class="anchor"></a>Types of data</h3>
<p>Statisticians have specific classifications for different types of data. You will see different ways of presenting this these classifications; what I have done here is focused on the most practical features that impact common data analyses.</p>
<p>1)quantitative data (but trying to think of a better name) 2)ordinal 3)frequencies (better name?) 4)counts 5)continuous percentages</p>
<p>skip: nominal (categorical)</p>
<p>for the mathematically inclined :these categories relate to the type of distributions which can be used to describe data</p>
<ul>
<li>
<strong>Quantitative data</strong> measurements such as length, mass, pH, area, current, produced by instruments such as rulers, calipers, meters.</li>
</ul>
<p>“continous quantitative data”</p>
<ul>
<li>
<strong>Ordinal data</strong> or “ordered categories” produced by ranking, scoring, or classifying something. The most familiar ordinal scale is letter grades in school. Common now are also online rankings, such as the number of stars from 1 to 4 used for Amazon products. OMETS In medicine we use pain scale from 1 to 10, and in orthopedics and physical therapy joint mobility scales are used. Ecologists studying herbivory and disease frequently assign a score to plants from 1 (no damage) to 10 (completley destroyed). In principal most ordinal data could be measured with some instrument to yield quantitative data, but that might too expensive or difficult.</li>
</ul>
<p>ranks vs ordinal</p>
<p>“. Ordinal measurements have imprecise differences between consecutive values, but have a meaningful order to those values, and permit any order-preserving transformation.”</p>
<ul>
<li>
<strong>Frequency data</strong>, which comes from determining the resolution of either-or occurrences, such as the survival of bacterial cells or inheritance of a Y chromosome during conception. A key feature is that each “occurrence” happens on its own, like flipping a coin. Frequency data can be summarized using a percentage, where the denominator is the number of occurrences and the numerator is the number of times a particular outcome happens. Don’t confuse frequency data with count data OR continuous percentage data (see below)! A key diagnostic of frequency data is that you can always identify a denominator. For example, we can count up the number of games the Pittsburgh Pirates have won this season; we can also calculate the percentage of wins. Since there is a denominator available (total number of games, this is frequency data. Similarly, there’s always an upper bound to frequencies: you can’t win more than 100% of the games or can’t have more than 100% of your fruit flies be XY male.<br>
Another diagnostic of count data is that the numerator and denominator are both always whole numbers.</li>
</ul>
<p>counted fractions Tukey</p>
<ul>
<li>
<strong>Count data</strong> of the number of items or objects: number of flowers, number of offspring, number of tumors. This count data should not be confused with frequency data. A key diagnostic: frequency data can be summarized using a percentage, while this never makes sense for count data. A key diagnostic of frequency data is that you can always identify a denominator. For example, we can count up the number of games the Pittsburgh Pirates have won this season; we can also calculate the percentage of wins. Since there is a denominator available (total number of games, this is frequency data. Similarly, there’s no upper bound. So while you can’t win more than 100% of your baseball games (frequencies) there’s no limit on the number of runs you can score in a single game (count data)</li>
</ul>
<p>discrete quantitative</p>
<ul>
<li>
<strong>Continuous percentages</strong>, such how much of a petri dish is covered by bacteria, how much of a patient’s body was burned, or the exact percent of a leaf eaten by an herbivore. For this type of data, usually the numerator and the denominator are some form of quantitiatve data.</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
