<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Intro to 1-way ANOVA: impacts of diet on deer antlers • compbio4all</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Intro to 1-way ANOVA: impacts of diet on deer antlers">
<meta property="og:description" content="compbio4all">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sss-stats-ANOVA_deer_antlers-ADD-DATA_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Intro to 1-way ANOVA: impacts of diet on deer antlers</h1>
            
      
      
      <div class="hidden name"><code>sss-stats-ANOVA_deer_antlers-ADD-DATA.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">compbio4all</span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The purpose of this lab is to learn the basics of 1-way ANOVA in R. We will explore data where white-tailed deer where fed different diets over the course of the spring and summer of one year.</p>
<ul>
<li>Diet 1: “Hi.Hi” = deer fed high protein diets during spring and summer</li>
<li>Diet 2: “Hi.Lo” = deer fed high quality diets during spring, but low quality diet during summer</li>
<li>Diet 3: “Lo.Hi” = deer fed low-protein diet during spring but high quality diet during summer.</li>
</ul>
<p>Note: there is no “Lo.Lo” treatment</p>
<p>Analyis with 1-way ANOVA will allow us to determine whether</p>
<ol style="list-style-type: decimal">
<li>All of the means are similar or if at least one differs</li>
<li>Which treatment, if any, is most likely impacting antler growth.</li>
</ol>
<div id="references" class="section level3">
<h3 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h3>
<p>Data were simulated from summary parameters in:</p>
<p>Asleson et al. 1997. Effects of seasonal protein restriction on antlerogenesis and body mass in adult male white-tailed deer. Journal of Wildlife Management 61.</p>
<p>For more information see ?antlers</p>
</div>
</div>
<div id="part-1-summary-stats-plotting-means" class="section level2">
<h2 class="hasAnchor">
<a href="#part-1-summary-stats-plotting-means" class="anchor"></a>Part 1: Summary stats &amp; plotting means</h2>
<div id="summary-stats-on-all-data" class="section level3">
<h3 class="hasAnchor">
<a href="#summary-stats-on-all-data" class="anchor"></a>1) Summary stats on all data</h3>
<p>In this section the grand mean on ALL the data is calculated; data is NOT broken up by treatment! Data is considered by subgroup below</p>
<p>Load data</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brouwern/wildlifeR">wildlifeR</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># total sample size (all observations)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">)</span>

<span class="va">n.total</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">$</span><span class="va">mass</span><span class="op">)</span>

<span class="co">#mean of ALL samples</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">$</span><span class="va">mass</span><span class="op">)</span>

<span class="co">#variance of ALL samples</span>
<span class="fu"><a href="https://rdrr.io/r/stats/cor.html">var</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">$</span><span class="va">mass</span><span class="op">)</span>

<span class="co">#stdev of ALL samples</span>
<span class="va">mass.sd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">$</span><span class="va">mass</span><span class="op">)</span></code></pre></div>
</div>
<div id="standard-error-for-all-data" class="section level3">
<h3 class="hasAnchor">
<a href="#standard-error-for-all-data" class="anchor"></a>2) Standard error for all data</h3>
<p>This ignores treatments. All data are combined / pooled.</p>
<p>We’ll do this a couple different ways to show how R code can vary.</p>
<p>Broken up into steps, saving an object at each stuep</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#square root of N</span>
<span class="va">sqrt.n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">n.total</span><span class="op">)</span>

<span class="co">#the see</span>
<span class="va">mass.se</span> <span class="op">&lt;-</span> <span class="va">mass.sd</span><span class="op">/</span><span class="va">sqrt.n</span>

<span class="va">mass.se</span></code></pre></div>
<p>Doing it a little “on fly”</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mass.se</span> <span class="op">&lt;-</span> <span class="va">mass.sd</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">n.total</span><span class="op">)</span></code></pre></div>
<p>Totally on the fly - nothing pre-saved as an object</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#Using raw data</span>
<span class="va">mass.se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">$</span><span class="va">mass</span><span class="op">)</span><span class="op">/</span>
             <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">antlers</span><span class="op">$</span><span class="va">mass</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="ci-for-all-data" class="section level3">
<h3 class="hasAnchor">
<a href="#ci-for-all-data" class="anchor"></a>3) 95% CI for all data</h3>
<p>This is the 95% confidence interval around the overall/grand mean. W approximate the CI as 1.96*SE</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fl">1.96</span><span class="op">*</span><span class="va">mass.se</span></code></pre></div>
</div>
<div id="calcualte-summary-stats-for-each-group" class="section level3">
<h3 class="hasAnchor">
<a href="#calcualte-summary-stats-for-each-group" class="anchor"></a>4) Calcualte summary stats for each group</h3>
<ul>
<li>Calculae the mean for each diet treatment using dplyr.<br>
</li>
<li>THe group_by() function split it up by treatment</li>
<li>summarize() calcualtes the mean</li>
<li>Recall the “%&gt;%” is called a “pipe”</li>
</ul>
<p>See pages 70-73 in “Getting started with R: An INtroduction for Biologists” for information on summarize() and group_by(). In particular see section 3.7.3 on page 72: “Method 2: Pipe, no nesting”</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="va">antlers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">diet</span><span class="op">)</span> <span class="op">%&gt;%</span>
   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span>mass.mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We can get the SD like this</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">antlers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">diet</span><span class="op">)</span> <span class="op">%&gt;%</span>
   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span>mass.sd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>And the sample size like this</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">antlers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">diet</span><span class="op">)</span> <span class="op">%&gt;%</span>
   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span>mass.N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We can calcualte all of them like below. this time we’ll store the ouptut in an object “mass.means”</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mass.means</span> <span class="op">&lt;-</span> <span class="va">antlers</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">diet</span><span class="op">)</span> <span class="op">%&gt;%</span>
   <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span>mass.mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span>,
             mass.sd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span>,
             mass.N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">mass</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="calcualte-the-3-se-values" class="section level3">
<h3 class="hasAnchor">
<a href="#calcualte-the-3-se-values" class="anchor"></a>5) Calcualte the 3 SE values</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mass.SEs</span> <span class="op">&lt;-</span> <span class="va">mass.means</span><span class="op">$</span><span class="va">mass.sd</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">mass.means</span><span class="op">$</span><span class="va">mass.N</span><span class="op">)</span></code></pre></div>
<p>Add them to the dataframme</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mass.means</span><span class="op">$</span><span class="va">mass.SEs</span> <span class="op">&lt;-</span> <span class="va">mass.SEs</span></code></pre></div>
<p>We can conver these to approximately 95% confidence intervals like this</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mass.means</span><span class="op">$</span><span class="va">mass.CI95</span> <span class="op">&lt;-</span> <span class="va">mass.means</span><span class="op">$</span><span class="va">mass.SEs</span><span class="op">*</span><span class="fl">1.96</span></code></pre></div>
<p>What do you notice about the relationship between the SD, SE, and 95% CI?</p>
</div>
</div>
<div id="plot-the-3-mean-values" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-3-mean-values" class="anchor"></a>6) Plot the 3 mean values</h2>
<p>We’ll plot means using ggpubr. We’ll give the function ggerrorplot() the raw data and it will calculate the means etc to make the plot.</p>
<p>See the link below for more information <a href="http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/79-plot-meansmedians-and-error-bars/" class="uri">http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/79-plot-meansmedians-and-error-bars/</a></p>
<p>mean_se mean_sd mean_range</p>
<p>First, we’ll plot the means and their standard deviations SD</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span>

<span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggerrorplot.html">ggerrorplot</a></span><span class="op">(</span><span class="va">antlers</span>, 
            x <span class="op">=</span> <span class="st">"diet"</span>, 
            y <span class="op">=</span> <span class="st">"mass"</span>, 
            desc_stat <span class="op">=</span> <span class="st">"mean_sd"</span>,
            add <span class="op">=</span> <span class="st">"mean"</span>
            <span class="op">)</span></code></pre></div>
<p>Next we’ll plot the standard errors. What happens to the x-axis?</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggerrorplot.html">ggerrorplot</a></span><span class="op">(</span><span class="va">antlers</span>, 
            x <span class="op">=</span> <span class="st">"diet"</span>, 
            y <span class="op">=</span> <span class="st">"mass"</span>, 
            desc_stat <span class="op">=</span> <span class="st">"mean_se"</span>,
            add <span class="op">=</span> <span class="st">"mean"</span>
            <span class="op">)</span></code></pre></div>
<p>We can make a plot with both if we save the plots to R objects. We’ll use the argument ylim = c(400,1000) to make it so they both have the same y axes</p>
<p>Note: We’ll need to load the gridExtra package</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"gridExtra"</span><span class="op">)</span>
<span class="va">plot1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggerrorplot.html">ggerrorplot</a></span><span class="op">(</span><span class="va">antlers</span>, 
            x <span class="op">=</span> <span class="st">"diet"</span>, 
            y <span class="op">=</span> <span class="st">"mass"</span>, 
            desc_stat <span class="op">=</span> <span class="st">"mean_sd"</span>,
            add <span class="op">=</span> <span class="st">"mean"</span>,
            ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">400</span>,<span class="fl">900</span><span class="op">)</span> <span class="co">#set axes</span>
            <span class="op">)</span>

<span class="va">plot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggerrorplot.html">ggerrorplot</a></span><span class="op">(</span><span class="va">antlers</span>, 
            x <span class="op">=</span> <span class="st">"diet"</span>, 
            y <span class="op">=</span> <span class="st">"mass"</span>, 
            desc_stat <span class="op">=</span> <span class="st">"mean_se"</span>,
            add <span class="op">=</span> <span class="st">"mean"</span>,
            ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">400</span>,<span class="fl">900</span><span class="op">)</span> <span class="co">#set axes</span>
            <span class="op">)</span></code></pre></div>
<p>Plot both plots. What do you notice about the standard errors?</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">plot1</span>,<span class="va">plot2</span><span class="op">)</span></code></pre></div>
<p>We can also easily make a plot using confience intervals.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggerrorplot.html">ggerrorplot</a></span><span class="op">(</span><span class="va">antlers</span>, 
            x <span class="op">=</span> <span class="st">"diet"</span>, 
            y <span class="op">=</span> <span class="st">"mass"</span>, 
            desc_stat <span class="op">=</span> <span class="st">"mean_ci"</span>,
            add <span class="op">=</span> <span class="st">"mean"</span>,
            ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">400</span>,<span class="fl">900</span><span class="op">)</span> <span class="co">#set axes</span>
            <span class="op">)</span></code></pre></div>
<p><strong>We have not calculated any p-values yet, but can you use the “inference by eye” approach to judge whether any of the means are likely to be “significantly” different based on the 95% confidence intervals?</strong></p>
</div>
<div id="part-2-omnibus-anova-test" class="section level2">
<h2 class="hasAnchor">
<a href="#part-2-omnibus-anova-test" class="anchor"></a>Part 2: Omnibus ANOVA test</h2>
<p>An “omnibus ANOVA test” (or “omnibus F test”) is used to gauge whether all the means are approximatley equal or whether at least one is likely to be different than the others. “Omnibus” means “overall”.</p>
<div id="build-the-null-ho-and-alternative-ha-models" class="section level3">
<h3 class="hasAnchor">
<a href="#build-the-null-ho-and-alternative-ha-models" class="anchor"></a>7 &amp; 8) Build the null (Ho) and alternative (Ha) models</h3>
<p>We build a null modle represent the null hypothesis that there is no difference between the any of the groups. Not that we code “mass ~ 1” to represent this null model, which has just a single mean value for all of the data.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">model.null</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="fl">1</span>, 
                 data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
<p>We build an alternative model or “model of interest” to test the hypothesis we are interested in: that diet impacts antler growth in some way. This model will have 3 means: one for each treatment.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R">

<span class="va">model.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="va">diet</span>, 
                 data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="conduct-the-omnibus-test-and-get-the-anova-output" class="section level2">
<h2 class="hasAnchor">
<a href="#conduct-the-omnibus-test-and-get-the-anova-output" class="anchor"></a>9-12) Conduct the Omnibus test and get the ANOVA output</h2>
<p>Produce the ANOVA table using the anova() command. This will give us the F statistics (the test statistics for ANOVA), degrees of freedom, and our p-values. (It will also give us some other info that we’ll ignore for now but that is typically reported when you do an ANOAV)</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">model.null</span>,
      <span class="va">model.alt</span><span class="op">)</span></code></pre></div>
<div id="optional" class="section level3">
<h3 class="hasAnchor">
<a href="#optional" class="anchor"></a>OPTIONAL:</h3>
<p>If you are curious and things are going well, do the optional activities. These are completely optional</p>
<div id="optional-the-summary-command-on-an-lm-object" class="section level4">
<h4 class="hasAnchor">
<a href="#optional-the-summary-command-on-an-lm-object" class="anchor"></a>Optional: The summary command on an lm() object</h4>
<p>Look at the output and how it compares to what we did with anova() on the two models.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">model.null</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">model.alt</span><span class="op">)</span></code></pre></div>
</div>
<div id="optional-the-anova-command-on-a-single-lm-object" class="section level4">
<h4 class="hasAnchor">
<a href="#optional-the-anova-command-on-a-single-lm-object" class="anchor"></a>Optional: The anova command on a single lm() object</h4>
<p>Look at the output and how it compares to what we did with anova() on the two models.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">model.alt</span><span class="op">)</span></code></pre></div>
</div>
<div id="optional-change-how-model-is-defined" class="section level4">
<h4 class="hasAnchor">
<a href="#optional-change-how-model-is-defined" class="anchor"></a>Optional: Change how model is defined</h4>
<p>We’ll add a “-1” to the model statemnt and call summary(). How does this differe from summary(model.alt)? (The difference is subtle…)</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model.alt.2</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span><span class="op">-</span><span class="fl">1</span> <span class="op">+</span> <span class="va">diet</span>, data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">model.alt.2</span><span class="op">)</span></code></pre></div>
</div>
<div id="optional-plotting-model-diagnostics" class="section level4">
<h4 class="hasAnchor">
<a href="#optional-plotting-model-diagnostics" class="anchor"></a>Optional: Plotting model diagnostics</h4>
<p>We can plot the “model diagnostics” using plot(). We will go into further details on this topic in the future.</p>
<p>Note: the par(mfrow = c(2,2)) is a bit of cryptic R code that sets up the plotting window to have 4 panels</p>
<p>Also note: we’ll use the ggplot2 addon library ggfortify for this, which has a fabulous function autoplot()</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load the library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sinhrks/ggfortify">ggfortify</a></span><span class="op">)</span>

<span class="co">#plot the residuals</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html">autoplot</a></span><span class="op">(</span><span class="va">model.alt</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="part-3-pairwise-comparisons" class="section level2">
<h2 class="hasAnchor">
<a href="#part-3-pairwise-comparisons" class="anchor"></a>Part 3: Pairwise comparisons</h2>
<p>The traditional way of doing a 1-way ANOVA is to follow up a “significant” Omnibus test with seperate t-tests that compare pair of means. This is called doing “pairwise” comparison - you go pair by pair doing t-tests.</p>
<p>The function pairwise.t.test() is used. Note that p.adjust.method = “none” line, which means we are given raw p-values that have <em>not</em> been corrected fro muotipel comparisons (see below)</p>
<div id="get-pairwise-p-valeus" class="section level3">
<h3 class="hasAnchor">
<a href="#get-pairwise-p-valeus" class="anchor"></a>13-15) Get pairwise p-valeus</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/pairwise.t.test.html">pairwise.t.test</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">antlers</span><span class="op">$</span><span class="va">mass</span>,
                g <span class="op">=</span> <span class="va">antlers</span><span class="op">$</span><span class="va">diet</span>,
                p.adjust.method <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="part-4-pairwise-comparisons-correcting-for-multiple-comparisons-effect-sizes" class="section level2">
<h2 class="hasAnchor">
<a href="#part-4-pairwise-comparisons-correcting-for-multiple-comparisons-effect-sizes" class="anchor"></a>Part 4: Pairwise comparisons, correcting for multiple comparisons, &amp; effect sizes</h2>
<p>The more t-tests you do, the more likely you are to get a “Significant” (low) p-value just due to chance. Rememeber - the definition of a p-value is such that it always leaves open the possiblity that the patterns you see are just due to random noise. This is called the “problem of multiple comparisons.”</p>
<p>One way that is advocated to to deal with this issue of multiple comparisons is to increase p-values. One way of doing this this is to use a “Bonferroni” correction. This can be done using pairwise.t.test() and setting p.adjust.method = “bonferroni”</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/pairwise.t.test.html">pairwise.t.test</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">antlers</span><span class="op">$</span><span class="va">mass</span>,
                g <span class="op">=</span> <span class="va">antlers</span><span class="op">$</span><span class="va">diet</span>,
                p.adjust.method <span class="op">=</span> <span class="st">"bonferroni"</span><span class="op">)</span></code></pre></div>
<div id="build-model-with-aov-function" class="section level3">
<h3 class="hasAnchor">
<a href="#build-model-with-aov-function" class="anchor"></a>17) Build model with aov() function</h3>
<p>When doing a 1-way ANOVA with three groups, a good option is to do a Tukey Test using the TukeyHSD() function. However, to do this you need first need to jump through a hoop: you need to redo the model model using the aov() function. (aov stands for “analysis of variance”)</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">model.alt.aov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aov.html">aov</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="va">diet</span>, 
                     data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
</div>
<div id="get-p-values-with-tukeys-hsd" class="section level3">
<h3 class="hasAnchor">
<a href="#get-p-values-with-tukeys-hsd" class="anchor"></a>18) Get p-values with Tukey’s HSD</h3>
<p>When the model is fit using aov() you can then use TukeyHSD(). This gives you p-values for each set of pairwise comparisons and, more importantly effect sizes and their confidence intervals for each comparison. Importantly, the p-values and CIs have been increased to deal with the fact that you are making multiple comparisons.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/stats/TukeyHSD.html">TukeyHSD</a></span><span class="op">(</span><span class="va">model.alt.aov</span><span class="op">)</span></code></pre></div>
</div>
<div id="plot-effect-sizes" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-effect-sizes" class="anchor"></a>19) Plot effect sizes</h3>
<p>The wildlifeR package contains a function that makes nice plots (IMHO) of the output of the TukeyHSD function.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="va">tukey.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/TukeyHSD.html">TukeyHSD</a></span><span class="op">(</span><span class="va">model.alt.aov</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/wildlifeR/man/plotTukeysHSD.html">plotTukeysHSD</a></span><span class="op">(</span><span class="va">tukey.out</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span>, col <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="summary-of-modeling-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#summary-of-modeling-workflow" class="anchor"></a>Summary of modeling workflow</h2>
<p>First, we plot the means of the data with error bars</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggerrorplot.html">ggerrorplot</a></span><span class="op">(</span><span class="va">antlers</span>, 
            x <span class="op">=</span> <span class="st">"diet"</span>, 
            y <span class="op">=</span> <span class="st">"mass"</span>, 
            desc_stat <span class="op">=</span> <span class="st">"mean_ci"</span>,
            add <span class="op">=</span> <span class="st">"mean"</span>,
            ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">400</span>,<span class="fl">900</span><span class="op">)</span> <span class="co">#set axes</span>
            <span class="op">)</span></code></pre></div>
<p>Next, we fit a null and alternative model with the lm()</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model.null</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="fl">1</span>, data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span>

<span class="va">model.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="va">diet</span>, data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
<p>THen, we compare these two models with the anova() command</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">model.null</span>,
      <span class="va">model.alt</span><span class="op">)</span></code></pre></div>
<p>We can interpret the P-value for this. We can also carry out a Tukey’s HSD test to calcualte effect sizes with confidence intervals corrected for multiple comparisons</p>
<p>First we have to re-fit the model with the aov() command</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model.alt.aov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aov.html">aov</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="va">diet</span>, 
                     data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
<p>Then we use TukeyHSD() on it (note it is Tukey, NOT Tukeys w/ an “s”)</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/TukeyHSD.html">TukeyHSD</a></span><span class="op">(</span><span class="va">model.alt.aov</span><span class="op">)</span></code></pre></div>
<p>If I were to write this up in a paper I would say something like this:</p>
<p>There was a significant impact of the diet treatment on the mass of deer antlers in the experiment (F = 4.02, P = 0.03). There was no significant difference between the “Hi-lo” and “Hi-Hi” diets (Effect Size = -44, Tukey’s HSD 95% CI for effect size = -206 to 118; p = 0.8) nor was there a difference between the Lo.Hi and the Hi.Hi. treatment (Effect size = 133, 95% CI = -28 to 296, p = 0.13). The only significant difference was between the Hi.Lo and the Lo.Hi diet (Effect Size = 178, 95% CI = 16 to 340, p = 0.029).</p>
</div>
<div id="exploring-regression-with-the-antler-data" class="section level2">
<h2 class="hasAnchor">
<a href="#exploring-regression-with-the-antler-data" class="anchor"></a>Exploring regression with the antler data</h2>
<p>The antler data has several response (y) variables: mass, circumference, beam length and spread. We can use regression to use one variable to predict the others.</p>
<p>Let’s say we want to use the beam length (the straight part of the antler from the base of the skull to where the points branch) to predict the mass.</p>
<p>The word equation would be mass ~ circum</p>
<p>We could make the scatterplot with a regression line like this</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">antlers</span>,
          y <span class="op">=</span> <span class="st">"mass"</span>,
          x <span class="op">=</span> <span class="st">"beam"</span>,
          add <span class="op">=</span> <span class="st">"reg.line"</span><span class="op">)</span></code></pre></div>
<p>HOw do we interpret the confidence interval aroudn this plot? What do we predict the p value to be approxmatley for this regression line?</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">antlers</span>,
          y <span class="op">=</span> <span class="st">"mass"</span>,
          x <span class="op">=</span> <span class="st">"beam"</span>,
          add <span class="op">=</span> <span class="st">"reg.line"</span>,
          conf.int <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>The regression code for a null model would be</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mass.vs.beam.null</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="fl">1</span>, data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
<p>The alternative hypothesis would be represented by the model:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mass.vs.beam.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">mass</span> <span class="op">~</span> <span class="va">beam</span>, data <span class="op">=</span> <span class="va">antlers</span><span class="op">)</span></code></pre></div>
<p>We can compare these models with the anova() command</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">mass.vs.beam.null</span>,
      <span class="va">mass.vs.beam.alt</span><span class="op">)</span></code></pre></div>
<p>We can get the details of the regression line using the summary() command on the alternative model</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mass.vs.beam.alt</span><span class="op">)</span></code></pre></div>
<p>If I were writing up these results I would say something like this:</p>
<p>“THere was a positive relationship between beam length and antler mass (slope = 0.22, SE = 0.49) but the relationship was not significant (F = 0.2, p = 0.6). The model was a very poor fit to the data and explained almost no variation in the data (R^2 = 0.008)”</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
