<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ compbio4all</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates-studnet-vs.html">Higgs and Atwood student version with updates</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG-with-misc-updates.html">Higgs and Atwood student version with updates - copy or alt?</a>
    </li>
    <li>
      <a href="../articles/Higgs_Atwood_AA_analysis-from-unit4-test-ORIG.html">Higgs and Atwood student version with updates - orig?</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_ggpubr.html">Software Checkpoint: ggpubr using Higgs and Atwood amino acids</a>
    </li>
    <li>
      <a href="../articles/Higgs_aa_data_1_load_and_replicate_plot.html">Higgs and Atwood part 1</a>
    </li>
    <li>
      <a href="../articles/PCA_tutorial_iris_data-inc_PCA_by_hand.html">Another PCA tutorial with Fisher's Iris Data</a>
    </li>
    <li>
      <a href="../articles/PCA_with_vegans_DCA.html">vegan PCA: Principal Components Analysis with vegans rda function</a>
    </li>
    <li>
      <a href="../articles/aaa-project_notes.html">a) Project notes</a>
    </li>
    <li>
      <a href="../articles/ac-a-introduction.html">a) Introduction</a>
    </li>
    <li>
      <a href="../articles/ac-k-GC_content_local_variation.html">k) Local variation in GC content</a>
    </li>
    <li>
      <a href="../articles/ac-u-HMM_sequence_evolution.html">u) Hidden Markov Models</a>
    </li>
    <li>
      <a href="../articles/aminio-acids-worked_example_data_visualization-DeprecatedQ.html">Worked example: Data visualization in R</a>
    </li>
    <li>
      <a href="../articles/basic_ac-h-vectors-AC04-01.html">Mini tutorial: Vectors in R</a>
    </li>
    <li>
      <a href="../articles/bbb-motivating_example.html">b) Motivating examples: Species distribution models</a>
    </li>
    <li>
      <a href="../articles/bbb-quickstart_RStudio_cloud.html">Quick start - RStudio Cloud</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_a_summary_stats-FROGARMS.html">Frog-armed: A first encounter with data</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_b_graphs-FROGARMS.html">A first encounter with data: graphs</a>
    </li>
    <li>
      <a href="../articles/casestudy-1st_encounter_w_data_c_t_test-FROGARMS.html">First encounter with data - t-test deprecated?</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-b-downloading_R_AC01-1.html">b) Downloading R</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-c-installing_RStudio_AC01-2.html">c) Downloading the RStudio IDE</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-d-installing_packages_in_RStudio_AC01-3.html">d) Installing R packages</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-e-installing_bioconductor_AC01-4.html">e) Installing Bioconductor</a>
    </li>
    <li>
      <a href="../articles/ccc-ac-f-brief_intro_to_R_AC01-5.html">f) A Brief introduction to R</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-dplyr_filter.html">Data Cleaning: Filtering focal rows with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning-merge_BBS_with_landcover.html">Data cleaning: Merging 2 large data sets with dplyr</a>
    </li>
    <li>
      <a href="../articles/data_cleaning_selecting_different_spp_and_landcovers.html">Selecting different species and landovers from the wildlifeR package</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-g-NCBI_by_GUI1_AC03-1.html">g) Downloading NCBI sequence data by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-DNA_sequence_descriptive_stats-AC03-02-.html">DNA descriptive statics - Part 1</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-NCBI_by_GUI2_AC03-1Q.html">h) Introduction to biological sequences databases</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-h-downloading_DNA_seq_in_R-AC03-01.html">Downloading DNA sequences with R</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-i-uniprot_by_GUI-AC07-01.html">i) Downloading sequences from UniProt by hand</a>
    </li>
    <li>
      <a href="../articles/ddd-ac-j-advanced_NCBI_AC05-01.html">j) Sequence Databases</a>
    </li>
    <li>
      <a href="../articles/eee-PAM_matrices.html">The Dayhoff PAM matrices</a>
    </li>
    <li>
      <a href="../articles/eee-ac-l-alignment_in_R-AC07-02.html">l) Aligments in R</a>
    </li>
    <li>
      <a href="../articles/eee-ac-o-test_alignment_sig_pvalue-AC07-03-NOT_UPATED.html">o) Testing the significance of an alignment</a>
    </li>
    <li>
      <a href="../articles/eee-alignment.html">Aligments and matrices</a>
    </li>
    <li>
      <a href="../articles/eee-nlb-alignment-walkthrough.html">NB-a) Hello R!  Bioinformatics awesomness in a flash</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_12.html">Pevsner Chapter 3: BLAST Algorithm in R (kinda)</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14.html">Pevsner Figure 4.14</a>
    </li>
    <li>
      <a href="../articles/fff-blast-Pevsner_Ch4_figure4_14_extension.html">Simulating sequences to get an EVD</a>
    </li>
    <li>
      <a href="../articles/fff-blast_via_R.html">BLAST via R</a>
    </li>
    <li>
      <a href="../articles/fff-nlb-BLAST_summary.html">BLAST Summary</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m1-retrieving_mult_seqs-AC08-01-NOTUPATED.html">p) Retrieving multiple sequences in R</a>
    </li>
    <li>
      <a href="../articles/ggg-ac-m2-mult_align_CLUSTAL-AC08-02-NEEDS_MAJOR_REWRITE.html">m) Multiple alignment with Clustal</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-q-comp_gene_find-AC10-01-NOTUPATED.html">q) Computational gene finding</a>
    </li>
    <li>
      <a href="../articles/jjj-ac-r-sig_of_ORFs-AC10-02-NOTUPATED.html">r) Significance of ORFs</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA-WPGMA_wikipedia.html">WPGMA or UPGMA?</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_algo_fnxns-not_finished.html">UPGMA functions</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_by_hand-updated.html">UPGMA the hard way</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_cluster_maximization_used_in_effron.html">UPGMA - Cluster max</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_in_R.html">UPGMA in R</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework-ALT-Q.html">UPGMA Wikipedia example - copy or alt Q</a>
    </li>
    <li>
      <a href="../articles/lll-UPGMA_wikipedia_homework.html">UPGMA in R - the hard way - deprecated?</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m3-rooted_tree_NJ-AC08-04-NOTUPATED.html">s) Rooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-ac-m4-unrooted_tree_NJ_tree-AC08-03-NEEDS_MAJOR_REWRITE.html">n) Unrooted Neighbor-joining Phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-jc_distance_fnxn.html">Jukes-Cantor Distance Function</a>
    </li>
    <li>
      <a href="../articles/lll-jc_sim_for_render.html">Simulating molecular evolution</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-newick_format.html">Newick format in R</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees.html">Tutorial: Writing functions to calculate the number of phylogenetic trees</a>
    </li>
    <li>
      <a href="../articles/lll-phylo-number_of_phylogenetic_trees_KEY.html">Tutorial: Writing functions to calculate the number of phylogenetic trees - KEY</a>
    </li>
    <li>
      <a href="../articles/llll-phylo_from_sequence_to_phylo_tree.html">Worked example: from sequence to phylogenetic tree</a>
    </li>
    <li>
      <a href="../articles/multivar_INTRO-high_dimensional_data_in_biology-iris_BCI_etc.html">Multivariate Data Analysis: Making Sense of High-Dimensional Data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_0_allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr-KEY.html">Software checkpoint: ggplot2 and ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_1_scp_ggpubr.html">ggplot2 and ggpubr software checkpoint</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp-KEY.html">Coding checkpoint: Plotting allometric data with ggpubr-KEY</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_2_allometric_data_ccp.html">Coding checkpoint: Plotting allometric data with ggpubr</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_4_vegan_and_pca.html">Overview: The vegan package for PCA</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allometry_5_cluster.html">Checkpoint: Cluster analysis with allometric data</a>
    </li>
    <li>
      <a href="../articles/multivar_data-allomtery_3_scp_scatterplot3d.html">Software checkpoint: scatterplot 3D</a>
    </li>
    <li>
      <a href="../articles/multivar_distances-GC_conent.html">Hiearchical clustering with Euclidean distances</a>
    </li>
    <li>
      <a href="../articles/nnn-ac-t-comparative_genomics.html">t) Comparative Genomics</a>
    </li>
    <li>
      <a href="../articles/plot-displaying_data_box_hist_base_R-NEEDS-IMAGES.html">Displaying Data - boxplot() &amp; hist()</a>
    </li>
    <li>
      <a href="../articles/plot-par_tutorial.html">Graphical parameters in R with par(): With great power comes great responsibility</a>
    </li>
    <li>
      <a href="../articles/plot_nlb-scp-ggplot2-ggpubr-allometric_scaling.html">Software checkpoint: ggplot2 and ggpubr-COPY?</a>
    </li>
    <li>
      <a href="../articles/prog-R_quirks_putting_things_into_vectors.html">R quirks: putting things in vectors</a>
    </li>
    <li>
      <a href="../articles/prog-assignment-for_loops.html">for loops in R</a>
    </li>
    <li>
      <a href="../articles/prog-simulate_random_aa_seqs.html">Simulating random amino acid sequences</a>
    </li>
    <li>
      <a href="../articles/prog_ac-h-programming_in_R-AC04-02.html">Programming in R: for loops and functions</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-function_practice.html">Writing functions - practice problems</a>
    </li>
    <li>
      <a href="../articles/prog_assignment-regex-accessions.html">Computational Biology Portfolio: Practice Working with Strings</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand.html">Building dataframes by hand</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set-KEY.html">Dataframe by hand - problem set - KEY</a>
    </li>
    <li>
      <a href="../articles/prog_data_frame_by_hand_problem_set.html">Dataframe by hand - problem set</a>
    </li>
    <li>
      <a href="../articles/prog_dataframe_by_hand_simple_with_plots.html">Making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/prog_nlb-walkthrough-making_simple_df.html">Walkthrough - making a simple dataframe</a>
    </li>
    <li>
      <a href="../articles/sss-nlb-types_of_data.html">Types of data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-ANOVA_deer_antlers-ADD-DATA.html">Intro to 1-way ANOVA: impacts of diet on deer antlers</a>
    </li>
    <li>
      <a href="../articles/sss-stats-Inf_by_eye.html">Inference-by-eye using RA Fisher's Cat Data</a>
    </li>
    <li>
      <a href="../articles/sss-stats-_means_errorbars_2016_base_R.html">Displaying data - means &amp; error bars</a>
    </li>
    <li>
      <a href="../articles/sss-stats-p_values_defined.html">P-values</a>
    </li>
    <li>
      <a href="../articles/sss-stats-paired_t_test_2_ways_walkthrough.html">paried t tests</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_paired_t_test.html">Running &amp; reporting paired t-tests in R</a>
    </li>
    <li>
      <a href="../articles/sss-stats-tttest-reporting_t_test-ADD-DATA.html">Reporting statistical result from a 2-sample t-test</a>
    </li>
    <li>
      <a href="../articles/zzz0-undergrad_editors.html">z0) Undergraduate editors</a>
    </li>
    <li>
      <a href="../articles/zzz0-vocab.html">z) vocab</a>
    </li>
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz2-TODO.html">TODO</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="AOU_species_codes.html">AOU_species_codes</a></code> </p>
        </td>
        <td><p>American Ornithological Union (AOU) species codes for North-American birds</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA.html">BBS_PA</a></code> </p>
        </td>
        <td><p>Pennsylvania Breeding Survey Data (BBS) from the USGS.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA_SCTA.html">BBS_PA_SCTA</a></code> </p>
        </td>
        <td><p>Number of Scarlet tanagers (SCTA) observed on Pennsylvania BBS routes in 2006 and total forested landcover in 1km buffers around routes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA_combo_cover_1km.html">BBS_PA_combo_cover_1km</a></code> </p>
        </td>
        <td><p>Combinations of NLCD land cover classes into more general groups for Pennsylvania, USA.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA_landcover_1km.html">BBS_PA_landcover_1km</a></code> </p>
        </td>
        <td><p>Land cover buffers from the 1 km surrounding Breeding Bird Survey Routes (BBS) in Pennsylvania</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NA_to_zero.html">NA_to_zero()</a></code> </p>
        </td>
        <td><p>Change all NA values (missing vlaues) in a column to zero</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa_chars.html">aa_chars</a></code> </p>
        </td>
        <td><p>Characteristics of amino acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa_chars_subset.html">aa_chars_subset</a></code> </p>
        </td>
        <td><p>Characteristics of amino acids (subset)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="antlers.html">antlers</a></code> </p>
        </td>
        <td><p>Simulated data on the effects of diet on deer antlers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="blastr.html">blastr()</a></code> </p>
        </td>
        <td><p>Execute BLAST from R.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="brook_trout.html">brook_trout</a></code> </p>
        </td>
        <td><p>Brook trout impacts on Chinook salmon survival - original data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="brook_trout_ABD.html">brook_trout_ABD</a></code> </p>
        </td>
        <td><p>Impact of brook trout on survival of Chinook salmon in Idaho.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chargaff1951.html">chargaff1951</a></code> </p>
        </td>
        <td><p>Data from publications by Chargaff 1951</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="combine_NLCD_classes.html">combine_NLCD_classes()</a></code> </p>
        </td>
        <td><p>Combine multiple landcover classes into a single combined group</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dengueseq_fasta.html">dengueseq_fasta</a></code> </p>
        </td>
        <td><p>dengueseq_fasta</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="diag_show.html">diag_show()</a></code> </p>
        </td>
        <td><p>Print the diagonal of a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="doves_PA.html">doves_PA</a></code> </p>
        </td>
        <td><p>PA Mourning doves</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="doves_PA_3_years.html">doves_PA_3_years</a></code> </p>
        </td>
        <td><p>PA Mourning doves from 3 years, 2005, 2007 and 2010</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eaglesOH.html">eaglesOH</a></code> </p>
        </td>
        <td><p>Eagles in Ohio dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eaglesPA.html">eaglesPA</a></code> </p>
        </td>
        <td><p>Eagles in Pennsylvania dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eaglesWV.html">eaglesWV</a></code> </p>
        </td>
        <td><p>Eagles in West Virginia (WV) dataset#' A dataset containing the number of nesting pairs of eagles found in West Virginia, USA, from 1980 through 2007  Data is not available for all years and is given as NA.
Important dates:
1963	417 pairs in all of continental USA
1966	endangered spp act
1967	listed
1972	DDT banned</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eggs.html">eggs</a></code> </p>
        </td>
        <td><p>Bird eggs datatset.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eggs_and_nests.html">eggs_and_nests</a></code> </p>
        </td>
        <td><p>Egg data combined with data on nests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eggs_and_nests2.html">eggs_and_nests2</a></code> </p>
        </td>
        <td><p>Subset of eggs_and_nests data the contains only Dome, Cup, and ground surface nests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="expand_BBS_routes.html">expand_BBS_routes()</a></code> </p>
        </td>
        <td><p>Expand BBS data to include zeros when a species was not seen.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fasta_cleaner.html">fasta_cleaner()</a></code> </p>
        </td>
        <td><p>Function to convert a FASTA file stored as an object into a vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="findPotentialStartsAndStops.html">findPotentialStartsAndStops()</a></code> </p>
        </td>
        <td><p>Function to identify potential start and stop codons</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="frogarms.html">frogarms</a></code> </p>
        </td>
        <td><p>Data on the body size and arm girth of male and female Australian Quacking Frogs (Crinia georgiana)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generateSeqsWithMultinomialModel.html">generateSeqsWithMultinomialModel()</a></code> </p>
        </td>
        <td><p>generateSeqsWithMultinomialModel</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hShroom2.html">hShroom2</a></code> </p>
        </td>
        <td><p>hShroom2</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hShroom3.html">hShroom3</a></code> </p>
        </td>
        <td><p>hShroom3</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="human_bodytemp.html">human_bodytemp</a></code> </p>
        </td>
        <td><p>Human body temperature data and heartrate</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="human_gene_lengths.html">human_gene_lengths</a></code> </p>
        </td>
        <td><p>Lengths of human genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mShroom3a.html">mShroom3a</a></code> </p>
        </td>
        <td><p>mShroom3a</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_dateset_helpfile.html">make_dateset_helpfile()</a></code> </p>
        </td>
        <td><p>Create template for dataset documentation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_my_data2L.html">make_my_data2L()</a></code> </p>
        </td>
        <td><p>Extract random subset of a dataframe</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="milk.html">milk</a></code> </p>
        </td>
        <td><p>Mammal milk dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pams.html">pams</a></code> </p>
        </td>
        <td><p>Point-accepted mutation (PAM) data matrix from Dayoff et al. 1978</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pikas.html">pikas</a></code> </p>
        </td>
        <td><p>Data on presence of pikas and marmots from Front Range Pika Project</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="printPairwiseAlignment.html">printPairwiseAlignment()</a></code> </p>
        </td>
        <td><p>printPairwiseAlignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sShroom.html">sShroom</a></code> </p>
        </td>
        <td><p>sShroom</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sample_BBS_routes.html">sample_BBS_routes()</a></code> </p>
        </td>
        <td><p>Select random subset of Breeding Bird Survey (BBS) routes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sample_dove_routes.html">sample_dove_routes()</a></code> </p>
        </td>
        <td><p>Select random subset of Breeding Bird Survey (BBS) routes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="score_alignment.html">score_alignment()</a></code> </p>
        </td>
        <td><p>Score a simple alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="shroom_table.html">shroom_table</a></code> </p>
        </td>
        <td><p>shroom_table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tri_print.html">tri_print()</a></code> </p>
        </td>
        <td><p>Print the lower or upper triangular part of a matrix  flextable</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trillium.html">trillium</a></code> </p>
        </td>
        <td><p>Data on the size of Trillium from several sites in Crawford County, PA</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tutor_open.html">tutor_open()</a></code> </p>
        </td>
        <td><p>Open a specified tutorial file contained within the package</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tutor_show.html">tutor_show()</a></code> </p>
        </td>
        <td><p>Show the tutorials currently available for the dayoff package</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="update_dayoff.html">update_dayoff()</a></code> </p>
        </td>
        <td><p>Function to update dayoff package</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="wetlands.html">wetlands</a></code> </p>
        </td>
        <td><p>Data on species richness in constructed wetlands at Pennsylvania Game Lands 297 in south western PA.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="wine.html">wine</a></code> </p>
        </td>
        <td><p>Wine recognition data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="woodcock_MI_3_years.html">woodcock_MI_3_years</a></code> </p>
        </td>
        <td><p>Subset of Woodcock call survey data from Michigan, USA</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


