<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ compbio4all</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio4all</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/zzz1-temp.html">z) vignette template</a>
    </li>
    <li>
      <a href="../articles/zzz3-creating_this_package.html">x) Creating this package</a>
    </li>
    <li>
      <a href="../articles/zzz4-document_datasets.html">y) Documenting datasets</a>
    </li>
    <li>
      <a href="../articles/zzz5-build_and_rebuild_package.html">Building &amp; rebuilding this package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="AOU_species_codes.html">AOU_species_codes</a></code> </p>
        </td>
        <td><p>American Ornithological Union (AOU) species codes for North-American birds</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA.html">BBS_PA</a></code> </p>
        </td>
        <td><p>Pennsylvania Breeding Survey Data (BBS) from the USGS.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA_SCTA.html">BBS_PA_SCTA</a></code> </p>
        </td>
        <td><p>Number of Scarlet tanagers (SCTA) observed on Pennsylvania BBS routes in 2006 and total forested landcover in 1km buffers around routes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA_combo_cover_1km.html">BBS_PA_combo_cover_1km</a></code> </p>
        </td>
        <td><p>Combinations of NLCD land cover classes into more general groups for Pennsylvania, USA.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BBS_PA_landcover_1km.html">BBS_PA_landcover_1km</a></code> </p>
        </td>
        <td><p>Land cover buffers from the 1 km surrounding Breeding Bird Survey Routes (BBS) in Pennsylvania</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NA_to_zero.html">NA_to_zero()</a></code> </p>
        </td>
        <td><p>Change all NA values (missing vlaues) in a column to zero</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa_chars.html">aa_chars</a></code> </p>
        </td>
        <td><p>Characteristics of amino acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa_chars_subset.html">aa_chars_subset</a></code> </p>
        </td>
        <td><p>Characteristics of amino acids (subset)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aa_dat.html">aa_dat</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="antlers.html">antlers</a></code> </p>
        </td>
        <td><p>Simulated data on the effects of diet on deer antlers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="blastr.html">blastr()</a></code> </p>
        </td>
        <td><p>Execute BLAST from R.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="brook_trout.html">brook_trout</a></code> </p>
        </td>
        <td><p>Brook trout impacts on Chinook salmon survival - original data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="brook_trout_ABD.html">brook_trout_ABD</a></code> </p>
        </td>
        <td><p>Impact of brook trout on survival of Chinook salmon in Idaho.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="chargaff1951.html">chargaff1951</a></code> </p>
        </td>
        <td><p>Data from publications by Chargaff 1951</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clean_alignment.html">clean_alignment()</a></code> </p>
        </td>
        <td><p>Print multiple sequence alignment (MSA) to R console</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="combine_NLCD_classes.html">combine_NLCD_classes()</a></code> </p>
        </td>
        <td><p>Combine multiple landcover classes into a single combined group</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dengueseq_fasta.html">dengueseq_fasta</a></code> </p>
        </td>
        <td><p>dengueseq_fasta</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dengueseqstartstring.html">dengueseqstartstring</a></code> </p>
        </td>
        <td><p>Data for computational gene finding exercise</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="diag_show.html">diag_show()</a></code> </p>
        </td>
        <td><p>Print the diagonal of a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dmshrm_blast_1.html">dmshrm_blast_1</a></code> </p>
        </td>
        <td><p>BLAST results obtained from bio3d::blast.pdb().
blast.pdb() only queries the Protein Databank (PDB), which only contains high-question sequences.
In contrast, compbio4all::blastr() is a modification of blast.pdb() which  defaults instead to RefSeq.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dmshrm_blast_2.html">dmshrm_blast_2</a></code> </p>
        </td>
        <td><p>BLAST results obtained with the blastr() function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dna_seq_stringset.html">dna_seq_stringset</a></code> </p>
        </td>
        <td><p>Stringset of four mRNA sequences from dengue fever-related viruses in seqinr format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="doves_PA.html">doves_PA</a></code> </p>
        </td>
        <td><p>PA Mourning doves</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="doves_PA_3_years.html">doves_PA_3_years</a></code> </p>
        </td>
        <td><p>PA Mourning doves from 3 years, 2005, 2007 and 2010</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eaglesOH.html">eaglesOH</a></code> </p>
        </td>
        <td><p>Eagles in Ohio dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eaglesPA.html">eaglesPA</a></code> </p>
        </td>
        <td><p>Eagles in Pennsylvania dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eaglesWV.html">eaglesWV</a></code> </p>
        </td>
        <td><p>Eagles in West Virginia (WV) dataset#' A dataset containing the number of nesting pairs of eagles found in West Virginia, USA, from 1980 through 2007  Data is not available for all years and is given as NA.
Important dates:
1963	417 pairs in all of continental USA
1966	endangered spp act
1967	listed
1972	DDT banned</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eggs.html">eggs</a></code> </p>
        </td>
        <td><p>Bird eggs datatset.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eggs_and_nests.html">eggs_and_nests</a></code> </p>
        </td>
        <td><p>Egg data combined with data on nests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eggs_and_nests2.html">eggs_and_nests2</a></code> </p>
        </td>
        <td><p>Subset of eggs_and_nests data the contains only Dome, Cup, and ground surface nests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ensemblleishmania.html">ensemblleishmania</a></code> </p>
        </td>
        <td><p>Leishmania output from ENSEMBL</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ensemblpfalciparum.html">ensemblpfalciparum</a></code> </p>
        </td>
        <td><p>Plasmodium falciparum-related datasets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="entrez_fetch_list.html">entrez_fetch_list()</a></code> </p>
        </td>
        <td><p>Process multiple NCBI accessions and output as list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="expand_BBS_routes.html">expand_BBS_routes()</a></code> </p>
        </td>
        <td><p>Expand BBS data to include zeros when a species was not seen.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fasta_cleaner.html">fasta_cleaner()</a></code> </p>
        </td>
        <td><p>Function to convert a FASTA file stored as an object into a vector</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="find_ORFs_in_seq.html">find_ORFs_in_seq()</a></code> </p>
        </td>
        <td><p>Find potential open reading frames (ORFs) in a DNA sequence.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="find_start_stop_codons.html">find_start_stop_codons()</a></code> </p>
        </td>
        <td><p>Function to identify potential start and stop codons</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="frogarms.html">frogarms</a></code> </p>
        </td>
        <td><p>Data on the body size and arm girth of male and female Australian Quacking Frogs (Crinia georgiana)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate_hmm_seq.html">generate_hmm_seq()</a></code> </p>
        </td>
        <td><p>Function to generate a DNA sequence, given a HMM and the length of the sequence to be generated.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate_markov_seq.html">generate_markov_seq()</a></code> </p>
        </td>
        <td><p>Generate a DNA sequence using a Markov Chain.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genes_dpc.html">genes_dpc</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genes_hpc.html">genes_hpc</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genes_mpc.html">genes_mpc</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hShroom2.html">hShroom2</a></code> </p>
        </td>
        <td><p>hShroom2</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hShroom3.html">hShroom3</a></code> </p>
        </td>
        <td><p>hShroom3</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="human_bodytemp.html">human_bodytemp</a></code> </p>
        </td>
        <td><p>Human body temperature data and heartrate</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="human_gene_lengths.html">human_gene_lengths</a></code> </p>
        </td>
        <td><p>Lengths of human genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="leishmaniagenes.html">leishmaniagenes</a></code> </p>
        </td>
        <td><p>Leishmaniabgenes output from ENSEMBL</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="leishmaniagenes2.html">leishmaniagenes2</a></code> </p>
        </td>
        <td><p>Subset of Leishmania genes output from ENSEMBL</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mShroom3a.html">mShroom3a</a></code> </p>
        </td>
        <td><p>mShroom3a</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_dateset_helpfile.html">make_dateset_helpfile()</a></code> </p>
        </td>
        <td><p>Create template for dataset documentation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_my_data2L.html">make_my_data2L()</a></code> </p>
        </td>
        <td><p>Extract random subset of a dataframe</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_seqs_multinom_mod.html">make_seqs_multinom_mod()</a></code> </p>
        </td>
        <td><p>make_seqs_multinom_mod</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="make_viterbi_mat.html">make_viterbi_mat()</a></code> </p>
        </td>
        <td><p>Function to generate a DNA sequence, given a HMM and the length of the sequence to be generated.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="martin1995.html">martin1995</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="milk.html">milk</a></code> </p>
        </td>
        <td><p>Mammal milk dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pams.html">pams</a></code> </p>
        </td>
        <td><p>Point-accepted mutation (PAM) data matrix from Dayoff et al. 1978</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pfalciparumgenes.html">pfalciparumgenes</a></code> </p>
        </td>
        <td><p>Plasmodium falciparum-related datasets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pikas.html">pikas</a></code> </p>
        </td>
        <td><p>Data on presence of pikas and marmots from Front Range Pika Project</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_ORFs_in_seq.html">plot_ORFs_in_seq()</a></code> </p>
        </td>
        <td><p>Find and plot potential ORFs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_sliding_window.html">plot_sliding_window()</a></code> </p>
        </td>
        <td><p>plot_sliding_window</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_start_stop_codons.html">plot_start_stop_codons()</a></code> </p>
        </td>
        <td><p>plot_potential_starts_and_stops</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print_msa.html">print_msa()</a></code> </p>
        </td>
        <td><p>Print multiple sequence alignment (MSA) to R console</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print_pairwise_alignment.html">print_pairwise_alignment()</a></code> </p>
        </td>
        <td><p>printPairwiseAlignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="robinson_aafreq.html">robinson_aafreq</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rooted_NJ_tree.html">rooted_NJ_tree()</a></code> </p>
        </td>
        <td><p>Wrapper function for creating rooted neighbor-joining trees</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sShroom.html">sShroom</a></code> </p>
        </td>
        <td><p>sShroom</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sample_BBS_routes.html">sample_BBS_routes()</a></code> </p>
        </td>
        <td><p>Select random subset of Breeding Bird Survey (BBS) routes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sample_dove_routes.html">sample_dove_routes()</a></code> </p>
        </td>
        <td><p>Select random subset of Breeding Bird Survey (BBS) routes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="score_alignment.html">score_alignment()</a></code> </p>
        </td>
        <td><p>Score a simple alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seq_1_2_3_4.html">seq_1_2_3_4</a></code> </p>
        </td>
        <td><p>seq_1_2_3_4</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seq_1_2_3_4_stringset.html">seq_1_2_3_4_stringset</a></code> </p>
        </td>
        <td><p>seq_1_2_3_4_stringset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="shroom_table.html">shroom_table</a></code> </p>
        </td>
        <td><p>shroom_table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tri_print.html">tri_print()</a></code> </p>
        </td>
        <td><p>Print the lower or upper triangular part of a matrix  flextable</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="trillium.html">trillium</a></code> </p>
        </td>
        <td><p>Data on the size of Trillium from several sites in Crawford County, PA</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virus_mrna_aln.html">virus_mrna_aln</a></code> </p>
        </td>
        <td><p>Alignment of four mRNA sequences from dengue fever-related viruses</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virus_mrna_aln_bin.html">virus_mrna_aln_bin</a></code> </p>
        </td>
        <td><p>Alignment of four proteins from dengue fever-related viruses in bin format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virus_mrna_dist.html">virus_mrna_dist</a></code> </p>
        </td>
        <td><p>Distance matrix of Dengue virus mRNA sequences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virus_snc.html">virus_snc</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virus_snr.html">virus_snr</a></code> </p>
        </td>
        <td><p>Dataset helpfile header . . .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virusaln.html">virusaln</a></code> </p>
        </td>
        <td><p>Alignment of four proteins from dengue fever-related viruses</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virusaln_seqinr.html">virusaln_seqinr</a></code> </p>
        </td>
        <td><p>Alignment of four proteins from dengue fever-related viruses in seqinr format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="virusaln_seqinr_clean.html">virusaln_seqinr_clean</a></code> </p>
        </td>
        <td><p>Alignment of four proteins from dengue fever-related viruses with poorly aligned portions removed</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="viterbi.html">viterbi()</a></code> </p>
        </td>
        <td><p>By Coghlan (2011), adapted from "Applied Statistics for Bioinformatics using R" by Wim P. Krijnen, page 209</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="wetlands.html">wetlands</a></code> </p>
        </td>
        <td><p>Data on species richness in constructed wetlands at Pennsylvania Game Lands 297 in south western PA.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="wine.html">wine</a></code> </p>
        </td>
        <td><p>Wine recognition data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="woodcock_MI_3_years.html">woodcock_MI_3_years</a></code> </p>
        </td>
        <td><p>Subset of Woodcock call survey data from Michigan, USA</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nathan Brouwer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


